--- /usr/portage/sys-auth/policykit/policykit-0.9.ebuild	2009-04-23 18:10:21.000000000 +0200
+++ ./sys-auth/policykit/policykit-0.9.ebuild	2010-03-28 19:03:42.000000000 +0200
@@ -35,10 +35,7 @@
 	enewuser polkituser -1 "-1" /dev/null polkituser
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Add zsh/bash completion
 	epatch "${FILESDIR}/${PN}-0.7-completions.patch"
 
@@ -54,7 +51,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	local authdb=
 
 	if use pam ; then
@@ -75,7 +72,6 @@
 		--localstatedir=/var
 	# won't install with tests
 	#	$(use_enable test tests) \
-	emake || die "emake failed"
 }
 
 src_install() {
