--- /usr/portage/sys-auth/policykit/policykit-0.9-r1.ebuild	2009-11-29 21:06:40.000000000 +0100
+++ ./sys-auth/policykit/policykit-0.9-r1.ebuild	2010-03-28 19:05:26.000000000 +0200
@@ -36,10 +36,7 @@
 	enewuser polkituser -1 "-1" /dev/null polkituser
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Add zsh/bash completion
 	epatch "${FILESDIR}/${PN}-0.7-completions.patch"
 
@@ -57,7 +54,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	local authdb=
 
 	if use pam ; then
@@ -78,7 +75,6 @@
 		--localstatedir=/var
 	# won't install with tests
 	#	$(use_enable test tests) \
-	emake || die "emake failed"
 }
 
 src_install() {
