--- /usr/portage/app-text/texlive-core/texlive-core-2009-r1.ebuild	2010-05-15 17:05:47.000000000 +0200
+++ ./app-text/texlive-core/texlive-core-2009-r1.ebuild	2011-01-23 15:27:57.000000000 +0100
@@ -97,13 +97,15 @@
 	done
 	cd "${S}"
 	mv "${WORKDIR}"/texmf* "${S}" || die "failed to move texmf files"
+}
 
+src_prepare() {
 	EPATCH_SUFFIX="patch" epatch "${WORKDIR}/patches"
 
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	# Too many regexps use A-Z a-z constructs, what causes problems with locales
 	# that don't have the same alphabetical order than ascii. Bug #242430
 	# So we set LC_ALL to C in order to avoid problems.
@@ -152,7 +154,9 @@
 		--enable-shared \
 		--disable-native-texlive-build \
 		$(use_with X x)
+}
 
+src_compile() {
 	emake SHELL=/bin/sh texmf=${TEXMF_PATH:-/usr/share/texmf} || die "emake failed"
 
 	# Mimic updmap --syncwithtrees to enable only fonts installed
