--- m4/package_utilies.m4
+++ m4/package_utilies.m4
@@ -32,15 +32,7 @@ AC_DEFUN([AC_PACKAGE_UTILITIES],
     AC_SUBST(make)
     AC_PACKAGE_NEED_UTILITY($1, "$make", make, [GNU make])
 
-    if test -z "$LIBTOOL"; then
-	AC_PATH_PROG(LIBTOOL, glibtool,, /usr/bin)
-    fi
-    if test -z "$LIBTOOL"; then
-	AC_PATH_PROG(LIBTOOL, libtool,, /usr/bin:/usr/local/bin:/usr/freeware/bin)
-    fi
-    libtool=$LIBTOOL
-    AC_SUBST(libtool)
-    AC_PACKAGE_NEED_UTILITY($1, "$libtool", libtool, [GNU libtool])
+    AC_PROG_LIBTOOL
 
     if test -z "$TAR"; then
         AC_PATH_PROG(TAR, tar,, /usr/freeware/bin:/bin:/usr/local/bin:/usr/bin)
--- include/builddefs.in
+++ include/builddefs.in
@@ -17,6 +17,7 @@ LIBMISC = $(TOPDIR)/libmisc/libmisc.la
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
+top_builddir = @top_builddir@
 datarootdir = @datarootdir@
 
 PKG_NAME	= @pkg_name@
--- m4/package_globals.m4
+++ m4/package_globals.m4
@@ -8,6 +8,8 @@ AC_DEFUN([AC_PACKAGE_GLOBALS],
   [ pkg_name="$1"
     AC_SUBST(pkg_name)
 
+    AC_PROG_CC
+
     . ./VERSION
     pkg_version=${PKG_MAJOR}.${PKG_MINOR}.${PKG_REVISION}
     AC_SUBST(pkg_version)
