diff -ur dbus-1.3.0.orig/dbus/Makefile.am dbus-1.3.0/dbus/Makefile.am
--- dbus-1.3.0.orig/dbus/Makefile.am	2008-04-04 19:32:04.000000000 +0300
+++ dbus-1.3.0/dbus/Makefile.am	2009-08-05 20:15:21.000000000 +0300
@@ -176,6 +176,7 @@
 libdbus_1_la_LDFLAGS= -export-symbols-regex "^[^_].*" -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined @R_DYNAMIC_LDFLAG@ @PIC_LDFLAGS@
 
 libdbus_convenience_la_LDFLAGS=@R_DYNAMIC_LDFLAG@
+libdbus_convenience_la_LIBADD=$(DBUS_TEST_LIBS)
 
 ## note that TESTS has special meaning (stuff to use in make check)
 ## so if adding tests not to be run in make check, don't add them to 
@@ -194,7 +195,7 @@
 dbus_test_SOURCES=				\
 	dbus-test-main.c
 
-dbus_test_LDADD=libdbus-convenience.la $(DBUS_TEST_LIBS)
+dbus_test_LDADD=$(DBUS_TEST_LIBS) libdbus-convenience.la
 dbus_test_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 ## mop up the gcov files
diff -ur dbus-1.3.0.orig/test/name-test/Makefile.am dbus-1.3.0/test/name-test/Makefile.am
--- dbus-1.3.0.orig/test/name-test/Makefile.am	2009-07-16 17:01:01.000000000 +0300
+++ dbus-1.3.0/test/name-test/Makefile.am	2009-08-05 20:12:15.000000000 +0300
@@ -21,52 +21,52 @@
 test_names_SOURCES=				\
 	test-names.c
 
-test_names_LDADD=$(top_builddir)/dbus/libdbus-convenience.la $(DBUS_TEST_LIBS)
+test_names_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la
 test_names_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_pending_call_dispatch_SOURCES =		\
 	test-pending-call-dispatch.c
 
-test_pending_call_dispatch_LDADD=$(top_builddir)/dbus/libdbus-convenience.la $(DBUS_TEST_LIBS)
+test_pending_call_dispatch_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la
 test_pending_call_dispatch_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_pending_call_timeout_SOURCES =		\
 	test-pending-call-timeout.c
 
-test_pending_call_timeout_LDADD=$(top_builddir)/dbus/libdbus-convenience.la $(DBUS_TEST_LIBS)
+test_pending_call_timeout_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la
 test_pending_call_timeout_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_threads_init_SOURCES =            \
 	test-threads-init.c
 
-test_threads_init_LDADD=$(top_builddir)/dbus/libdbus-convenience.la $(DBUS_TEST_LIBS)
+test_threads_init_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la
 test_threads_init_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_ids_SOURCES =            \
 	test-ids.c
 
-test_ids_LDADD=$(top_builddir)/dbus/libdbus-convenience.la $(DBUS_TEST_LIBS)
+test_ids_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la
 test_ids_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_shutdown_SOURCES =            \
 	test-shutdown.c
 
 test_shutdown_CFLAGS=
-test_shutdown_LDADD=$(top_builddir)/dbus/libdbus-convenience.la ../libdbus-testutils.la $(DBUS_TEST_LIBS)
+test_shutdown_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la ../libdbus-testutils.la
 test_shutdown_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_privserver_SOURCES =            \
 	test-privserver.c
 
 test_privserver_CFLAGS=
-test_privserver_LDADD=$(top_builddir)/dbus/libdbus-convenience.la ../libdbus-testutils.la $(DBUS_TEST_LIBS)
+test_privserver_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la ../libdbus-testutils.la
 test_privserver_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_privserver_client_SOURCES =            \
 	test-privserver-client.c
 
 test_privserver_client_CFLAGS=
-test_privserver_client_LDADD=$(top_builddir)/dbus/libdbus-convenience.la ../libdbus-testutils.la $(DBUS_TEST_LIBS)
+test_privserver_client_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-convenience.la ../libdbus-testutils.la
 test_privserver_client_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 endif
