--- /usr/portage/media-libs/gst-plugins-base/gst-plugins-base-0.10.24.ebuild	2010-01-05 19:35:51.000000000 +0100
+++ ./media-libs/gst-plugins-base/gst-plugins-base-0.10.24.ebuild	2010-03-27 23:31:33.000000000 +0100
@@ -24,14 +24,13 @@
 
 DOCS="AUTHORS README RELEASE"
 
-src_compile() {
+src_configure() {
 	# gst doesnt handle opts well, last tested with 0.10.15
 	strip-flags
 	replace-flags "-O3" "-O2"
 
 	gst-plugins-base_src_configure \
 		$(use_enable nls)
-	emake || die "emake failed."
 }
 
 src_install() {
