--- /usr/portage/dev-db/postgresql-base/postgresql-base-8.4.2.ebuild	2010-01-07 21:35:27.000000000 +0100
+++ ./dev-db/postgresql-base/postgresql-base-8.4.2.ebuild	2010-06-06 02:23:30.000000000 +0200
@@ -50,10 +50,7 @@
 
 S="${WORKDIR}/postgresql-${PV}"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}/postgresql-${SLOT}-common.patch" \
 		"${FILESDIR}/postgresql-${SLOT}-base.patch"
 
@@ -66,7 +63,7 @@
 	eautoconf
 }
 
-src_compile() {
+src_configure() {
 	econf --prefix=/usr/$(get_libdir)/postgresql-${SLOT} \
 		--datadir=/usr/share/postgresql-${SLOT} \
 		--docdir=/usr/share/doc/postgresql-${SLOT} \
@@ -90,7 +87,9 @@
 		$(use_with ldap) \
 		${myconf} \
 		|| die "configure failed"
+}
 
+src_compile() {
 	emake LD="$(tc-getLD) $(get_abi_LDFLAGS)" || die "emake failed"
 
 	cd "${S}/contrib"
