--- /usr/portage/dev-db/postgresql-base/postgresql-base-8.4.1.ebuild	2010-06-15 23:05:38.000000000 +0200
+++ ./dev-db/postgresql-base/postgresql-base-8.4.1.ebuild	2010-06-19 01:43:57.000000000 +0200
@@ -49,10 +49,7 @@
 
 S="${WORKDIR}/postgresql-${PV}"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}/postgresql-${SLOT}-common.patch" \
 		"${FILESDIR}/postgresql-${SLOT}-base.patch"
 
@@ -65,7 +62,7 @@
 	eautoconf
 }
 
-src_compile() {
+src_configure() {
 	econf --prefix=/usr/$(get_libdir)/postgresql-${SLOT} \
 		--datadir=/usr/share/postgresql-${SLOT} \
 		--docdir=/usr/share/doc/postgresql-${SLOT} \
@@ -89,7 +86,9 @@
 		$(use_with ldap) \
 		${myconf} \
 		|| die "configure failed"
+}
 
+src_compile() {
 	emake LD="$(tc-getLD) $(get_abi_LDFLAGS)" || die "emake failed"
 
 	cd "${S}/contrib"
