--- /usr/portage/net-dns/avahi/avahi-0.6.25.ebuild	2010-05-19 21:36:43.000000000 +0200
+++ ./net-dns/avahi/avahi-0.6.25.ebuild	2010-05-23 18:32:05.000000000 +0200
@@ -103,10 +103,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	use ipv6 && sed -i -e s/use-ipv6=no/use-ipv6=yes/ avahi-daemon/avahi-daemon.conf
 
 	sed -i -e "s:\\.\\./\\.\\./\\.\\./doc/avahi-docs/html/:../../../doc/${PF}/html/:" doxygen_to_devhelp.xsl
@@ -119,7 +116,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	use sh && replace-flags -O? -O0
 
 	local myconf=""
@@ -159,6 +156,9 @@
 		$(use_enable qt4) \
 		$(use_enable gdbm) \
 		${myconf}
+}
+
+src_compile() {
 	emake || die "emake failed"
 
 	use doc && emake avahi.devhelp
