--- /usr/portage/net-dns/avahi/avahi-0.6.25.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./net-dns/avahi/avahi-0.6.25.ebuild	2010-04-04 03:10:58.000000000 +0200
@@ -98,10 +98,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	use ipv6 && sed -i -e s/use-ipv6=no/use-ipv6=yes/ avahi-daemon/avahi-daemon.conf
 
 	sed -i -e "s:\\.\\./\\.\\./\\.\\./doc/avahi-docs/html/:../../../doc/${PF}/html/:" doxygen_to_devhelp.xsl
@@ -114,7 +111,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	use sh && replace-flags -O? -O0
 
 	local myconf=""
@@ -155,6 +152,9 @@
 		$(use_enable gdbm) \
 		${myconf} \
 		|| die "econf failed"
+}
+
+src_compile() {
 	emake || die "emake failed"
 
 	use doc && emake avahi.devhelp
