diff --git a/src/glx/x11/Makefile b/src/glx/x11/Makefile
index 419cb28..4d69a59 100644
--- a/src/glx/x11/Makefile
+++ b/src/glx/x11/Makefile
@@ -70,7 +70,7 @@ INCLUDES = -I. \
 default: depend $(TOP)/$(LIB_DIR)/$(GL_LIB_NAME)
 
 # Make libGL
-$(TOP)/$(LIB_DIR)/$(GL_LIB_NAME):  $(OBJECTS) Makefile
+$(TOP)/$(LIB_DIR)/$(GL_LIB_NAME):  $(OBJECTS) $(TOP)/$(LIB_DIR) Makefile
 	$(MKLIB) -o $(GL_LIB) -linker '$(CC)' -ldflags '$(LDFLAGS)' \
 		-major 1 -minor 2 $(MKLIB_OPTIONS) \
 		-install $(TOP)/$(LIB_DIR) -id $(INSTALL_LIB_DIR)/lib$(GL_LIB).1.dylib \
