--- /usr/portage/gnome-extra/evolution-data-server/evolution-data-server-2.30.3.ebuild	2010-08-22 22:05:59.000000000 +0200
+++ ./gnome-extra/evolution-data-server/evolution-data-server-2.30.3.ebuild	2010-08-26 00:47:55.000000000 +0200
@@ -47,6 +47,7 @@
 pkg_setup() {
 	G2CONF="${G2CONF}
 		$(use_with kerberos krb5 /usr)
+		$(use_with kerberos krb5-libs /usr/$(get_libdir) )
 		$(use_with ldap openldap)
 		$(use_enable gnome-keyring)
 		$(use_enable ipv6)
