--- /usr/portage/gnome-extra/evolution-data-server/evolution-data-server-2.26.3.ebuild	2010-08-05 19:12:01.000000000 +0200
+++ ./gnome-extra/evolution-data-server/evolution-data-server-2.26.3.ebuild	2010-08-07 02:54:20.000000000 +0200
@@ -49,6 +49,7 @@
 	G2CONF="${G2CONF}
 		$(use_with ldap openldap)
 		$(use_with kerberos krb5 /usr)
+		$(use_with kerberos krb5-libs /usr/$(get_libdir) )
 		$(use_enable ssl nss)
 		$(use_enable ssl smime)
 		$(use_enable ipv6)
