--- /usr/portage/gnome-extra/evolution-data-server/evolution-data-server-2.30.2.ebuild	2010-06-23 14:56:30.000000000 +0200
+++ ./gnome-extra/evolution-data-server/evolution-data-server-2.30.2.ebuild	2010-06-23 22:49:19.000000000 +0200
@@ -48,6 +48,7 @@
 pkg_setup() {
 	G2CONF="${G2CONF}
 		$(use_with kerberos krb5 /usr)
+		$(use_with kerberos krb5-libs /usr/$(get_libdir) )
 		$(use_with ldap openldap)
 		$(use_enable gnome-keyring)
 		$(use_enable ipv6)
