--- /usr/portage/./dev-util/pkgconfig/pkgconfig-0.23.ebuild	2009-04-05 12:37:41.000000000 +0200
+++ ./dev-util/pkgconfig/pkgconfig-0.23.ebuild	2010-06-05 21:41:45.000000000 +0200
@@ -29,6 +29,19 @@
 	# lines are inherited between libraries
 	use elibc_FreeBSD && myconf="${myconf} --enable-indirect-deps"
 
+	# adjust the default pc search path
+	if [[ -n EMULTILIB_PKG ]]; then
+		local pc_path="/usr/$(get_libdir)/pkgconfig"
+		local abi
+		for abi in ${MULTILIB_ABIS}; do
+			if [[ "$(get_libdir)" != "$(get_abi_LIBDIR ${abi})" ]]; then
+				pc_path="${pc_path}:/usr/$(get_abi_LIBDIR ${abi})/pkgconfig"
+			fi
+		done
+		pc_path="${pc_path}:/usr/share/pkgconfig"
+		myconf="${myconf} --with-pc-path=${pc_path}"
+	fi
+
 	econf ${myconf} || die "econf failed"
 	emake || die "emake failed"
 }
