Index: trunk/avahi-ui-sharp/Makefile.am
===================================================================
--- trunk/avahi-ui-sharp/Makefile.am	(revision 1478)
+++ trunk/avahi-ui-sharp/Makefile.am	(revision 1480)
@@ -39,8 +39,8 @@
 $(ASSEMBLY).config: $(ASSEMBLY).config.in
 	$(srcdir)/gencfg.sh $(top_builddir)/avahi-common/libavahi-common.la < $< > $@
 
-zssh.exe: $(srcdir)/zssh.cs
-	mcs -out:$@ $^ -r:./avahi-ui-sharp.dll -pkg:gtk-sharp-2.0 -r:Mono.Posix
+zssh.exe: $(srcdir)/zssh.cs $(ASSEMBLY)
+	mcs -out:$@ $(srcdir)/zssh.cs -r:./avahi-ui-sharp.dll -r:../avahi-sharp/avahi-sharp.dll -pkg:gtk-sharp-2.0 -r:Mono.Posix
 
 if HAVE_MONO
 if HAVE_DBUS
