diff -Nur avahi-0.6.19.orig/avahi-ui-sharp/Makefile.am avahi-0.6.19/avahi-ui-sharp/Makefile.am
--- avahi-0.6.19.orig/avahi-ui-sharp/Makefile.am	2007-06-29 17:46:50.068591000 +0000
+++ avahi-0.6.19/avahi-ui-sharp/Makefile.am	2007-06-29 17:48:32.023145178 +0000
@@ -44,6 +44,7 @@
 
 if HAVE_MONO
 if HAVE_DBUS
+if HAVE_GTK
 all: $(ASSEMBLY) $(ASSEMBLY).config zssh.exe
 
 if HAVE_MONODOC
@@ -68,3 +69,4 @@
 
 endif
 endif
+endif
