Index: trunk/avahi-daemon/dbus-util.c
===================================================================
--- trunk/avahi-daemon/dbus-util.c (revision 1357)
+++ trunk/avahi-daemon/dbus-util.c (revision 1482)
@@ -345,4 +345,7 @@
         n = 0;
         dbus_message_iter_get_fixed_array(&sub2, &k, &n);
+
+        if (!k)
+            k = (const uint8_t*) "";
             
         strlst = avahi_string_list_add_arbitrary(strlst, k, n);
Index: trunk/avahi-common/strlst.c
===================================================================
--- trunk/avahi-common/strlst.c (revision 1050)
+++ trunk/avahi-common/strlst.c (revision 1482)
@@ -52,5 +52,5 @@
     AvahiStringList *n;
 
-    assert(text);
+    assert(size == 0 || text);
 
     if (!(n = avahi_string_list_add_anonymous(l, size)))
