Index: trunk/avahi-autoipd/main.c
===================================================================
--- trunk/avahi-autoipd/main.c	(revision 1502)
+++ trunk/avahi-autoipd/main.c	(revision 1503)
@@ -510,7 +510,7 @@
     sa.sll_halen = ETHER_ADDRLEN;
     memset(sa.sll_addr, 0xFF, ETHER_ADDRLEN);
 
-    if (sendto(fd, packet, packet_len, 0, (struct sockaddr*) &sa, sizeof(sa)) < 0) {
+    if (sendto(fd, packet->ether_payload, packet_len, 0, (struct sockaddr*) &sa, sizeof(sa)) < 0) {
         daemon_log(LOG_ERR, "sendto() failed: %s", strerror(errno));
         return -1;
     }
