OPENSSL_VERSION_NUMBER is obviously wrong in the release

--- openssl-0.9.8f/crypto/opensslv.h
+++ openssl-0.9.8f/crypto/opensslv.h
@@ -25,7 +25,7 @@
  * (Prior to 0.9.5a beta1, a different scheme was used: MMNNFFRBB for
  *  major minor fix final patch/beta)
  */
-#define OPENSSL_VERSION_NUMBER	0x00908070L
+#define OPENSSL_VERSION_NUMBER	0x0090806fL
 #ifdef OPENSSL_FIPS
 #define OPENSSL_VERSION_TEXT	"OpenSSL 0.9.8f-fips 11 Oct 2007"
 #else
