fix from upstream

http://bugs.gentoo.org/224843

Index: crypto/x509/x509_att.c
===================================================================
RCS file: /usr/local/src/openssl/CVSROOT/openssl/crypto/x509/x509_att.c,v
retrieving revision 1.8.2.2
retrieving revision 1.8.2.3
diff -u -p -r1.8.2.2 -r1.8.2.3
--- crypto/x509/x509_att.c	2 Apr 2008 11:11:51 -0000	1.8.2.2
+++ crypto/x509/x509_att.c	30 May 2008 10:57:13 -0000	1.8.2.3
@@ -303,7 +303,7 @@ int X509_ATTRIBUTE_set1_data(X509_ATTRIB
 	}
 	if(!(attr->value.set = sk_ASN1_TYPE_new_null())) goto err;
 	if(!(ttmp = ASN1_TYPE_new())) goto err;
-	if (len == -1)
+	if ((len == -1) && !(attrtype & MBSTRING_FLAG))
 		{
 		if (!ASN1_TYPE_set1(ttmp, attrtype, data))
 			goto err;
