diff -ur pilot-link-0.11.8/bindings/Java/Makefile.in pilot-link-0.11.8.patched/bindings/Java/Makefile.in
--- pilot-link-0.11.8/bindings/Java/Makefile.in	2003-07-01 20:09:06.000000000 +0100
+++ pilot-link-0.11.8.patched/bindings/Java/Makefile.in	2003-08-30 13:30:00.000000000 +0100
@@ -241,7 +241,8 @@
 java_pisock: java_compile java_lib java_jar
 
 install: all
-	cp libjpisock.so ${JAVABASE}/jre/lib/i386/
+	mkdir -p $(DESTDIR)${JAVABASE}/jre/lib/i386/
+	cp libjpisock.so $(DESTDIR)${JAVABASE}/jre/lib/i386/
 
 test: install
 	${JAVABASE}/bin/java test ${PORT}
