--- pilot-link-0.11.8/bindings/Java/Makefile.in	2005-06-30 13:15:07.000000000 +0200
+++ pilot-link-0.11.8-r2/bindings/Java/Makefile.in	2005-06-30 13:16:45.000000000 +0200
@@ -233,7 +233,7 @@
 	$(JAVAC) -d . $(java_files)
 
 java_lib: libjpisock.c
-	$(CXX) $(INCLUDES) $(DEFS) -L $(top_builddir)/libpisock/.libs -lpisock -shared -o libjpisock.so $(srcdir)/libjpisock.c
+	$(CXX) $(CXXFLAGS) -fPIC $(INCLUDES) $(DEFS) -L $(top_builddir)/libpisock/.libs -lpisock -shared -o libjpisock.so $(srcdir)/libjpisock.c
 
 java_jar: $(java_classes)
 	$(JAR) cvf pilot-link.jar $(java_classes)
