--- pilot-link-0.11.8/bindings/Java/Makefile.am	2003-06-17 17:08:51.000000000 -0400
+++ pilot-link-0.11.8.new/bindings/Java/Makefile.am	2004-10-10 06:06:14.014193728 -0400
@@ -38,7 +38,7 @@
 	$(JAVAC) -d . $(java_files)
 
 java_lib: libjpisock.c
-	$(CXX) $(INCLUDES) $(DEFS) -L $(top_builddir)/libpisock/.libs -lpisock -shared -o libjpisock.so $(srcdir)/libjpisock.c
+	$(CXX) $(CXXFLAGS) -fPIC $(INCLUDES) $(DEFS) -L $(top_builddir)/libpisock/.libs -lpisock -shared -o libjpisock.so $(srcdir)/libjpisock.c
 
 java_jar: $(java_classes)
 	$(JAR) cvf pilot-link.jar $(java_classes)
