diff -urN ./pilot-link-0.12.3.orig/src/pilot-schlep.c ./pilot-link-0.12.3/src/pilot-schlep.c
--- ./pilot-link-0.12.3.orig/src/pilot-schlep.c	2006-10-12 16:21:21.000000000 +0200
+++ ./pilot-link-0.12.3/src/pilot-schlep.c	2009-02-16 16:09:25.321851803 +0100
@@ -21,6 +21,9 @@
  *
  */
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
 #include <stdio.h>
 
 #include "pi-source.h"
@@ -38,7 +38,7 @@
 		fd;
 	pi_buffer_t *buffer;
 
-	fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC);
+	fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
 	if (fd < 0)
 		return -1;
 
