--- glib/poppler-features.h.cmake.orig	2009-09-09 23:22:31.000000000 +0200
+++ glib/poppler-features.h.cmake	2010-01-26 23:36:17.518737903 +0100
@@ -20,6 +20,7 @@
 #define __POPPLER_FEATURES_H__
 
 @CAIRO_FEATURE@
+@GDK_FEATURE@
 
 #define POPPLER_MAJOR_VERSION (@POPPLER_MAJOR_VERSION@)
 #define POPPLER_MINOR_VERSION (@POPPLER_MINOR_VERSION@)
--- glib/CMakeLists.txt.orig	2010-01-26 23:40:40.719242931 +0100
+++ glib/CMakeLists.txt	2010-01-26 23:48:30.107752041 +0100
@@ -13,6 +13,7 @@
 endif (CAIRO_FOUND)
 
 if (GDK_FOUND)
+  set (GDK_FEATURE "#define POPPLER_WITH_GDK 1")
   add_definitions(${GDK2_CFLAGS})
 endif (GDK_FOUND)
 
