diff -ru ../poppler-0.12.3/cmake/modules/PopplerMacros.cmake ./cmake/modules/PopplerMacros.cmake
--- ../poppler-0.12.3/cmake/modules/PopplerMacros.cmake	2009-10-05 00:20:41.000000000 +0200
+++ ./cmake/modules/PopplerMacros.cmake	2010-04-10 03:57:46.763786638 +0200
@@ -86,12 +86,9 @@
 endif(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-  # set the default compile warnings
-  set(DEFAULT_COMPILE_WARNINGS_NO)
-  set(DEFAULT_COMPILE_WARNINGS_YES "-Wall -Wno-write-strings")
-  set(DEFAULT_COMPILE_WARNINGS_KDE "-Wnon-virtual-dtor -Wno-long-long -Wundef -ansi -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -Wcast-align -Wconversion -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -O2 -Wformat-security -Wmissing-format-attribute -fno-exceptions -fno-check-new -fno-common")
-
-  set(CMAKE_CXX_FLAGS                "-Woverloaded-virtual")
+  set(CMAKE_C_FLAGS "-Wall -Wno-write-strings -Wno-long-long -Wundef -Wcast-align -Wconversion -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common ${CMAKE_C_FLAGS}")
+  set(CMAKE_CXX_FLAGS "-Wall -Woverloaded-virtual -Wno-write-strings -Wnon-virtual-dtor -Wno-long-long -Wundef -ansi -Wcast-align -Wconversion -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-check-new -fno-common ${CMAKE_CXX_FLAGS}")
+  add_definitions(-D_XOPEN_SOURCE=500 -D_BSD_SOURCE)
   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
   set(CMAKE_CXX_FLAGS_RELEASE        "-O2 -DNDEBUG")
   set(CMAKE_CXX_FLAGS_DEBUG          "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
diff -ru ../poppler-0.12.3/CMakeLists.txt ./CMakeLists.txt
--- ../poppler-0.12.3/CMakeLists.txt	2009-12-24 11:50:12.000000000 +0100
+++ ./CMakeLists.txt	2010-04-10 03:29:23.799102761 +0200
@@ -115,21 +115,6 @@
   include_directories(${LCMS_INCLUDE_DIR})
 endif(LCMS_FOUND)
 
-if(DEFINED COMPILE_WARNINGS)
-else(DEFINED COMPILE_WARNINGS)
-  set(COMPILE_WARNINGS "yes")
-endif(DEFINED COMPILE_WARNINGS)
-string(TOLOWER "${COMPILE_WARNINGS}" _comp_warnings)
-if(_comp_warnings STREQUAL "no")
-  add_definitions(${DEFAULT_COMPILE_WARNINGS_NO})
-endif(_comp_warnings STREQUAL "no")
-if(_comp_warnings STREQUAL "yes")
-  add_definitions(${DEFAULT_COMPILE_WARNINGS_YES})
-endif(_comp_warnings STREQUAL "yes")
-if(_comp_warnings STREQUAL "kde")
-  set(CMAKE_CXX_FLAGS "${DEFAULT_COMPILE_WARNINGS_KDE} ${CMAKE_CXX_FLAGS}")
-endif(_comp_warnings STREQUAL "kde")
-
 
 include(ConfigureChecks.cmake)
 configure_file(config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h)
