https://bugs.freedesktop.org/show_bug.cgi?id=26442

--- glib/CMakeLists.txt
+++ glib/CMakeLists.txt
@@ -100,6 +100,9 @@
 if (CAIRO_FOUND)
   target_link_libraries(poppler-glib ${CAIRO_LIBRARIES})
 endif (CAIRO_FOUND)
+if (GDK_FOUND)
+  target_link_libraries(poppler-glib ${GDK2_LIBRARIES})
+endif (GDK_FOUND)
 install(TARGETS poppler-glib RUNTIME DESTINATION bin LIBRARY DESTINATION lib${LIB_SUFFIX} ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 install(FILES
