--- ./media-libs/libtheora/libtheora-1.0.ebuild.orig	2010-02-27 00:35:24.000000000 +0100
+++ ./media-libs/libtheora/libtheora-1.0.ebuild	2010-02-27 00:36:05.000000000 +0100
@@ -23,15 +23,13 @@
 
 S=${WORKDIR}/${P/_}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.0_beta2-flags.patch
 
 	AT_M4DIR="m4" eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	use x86 && filter-flags -fforce-addr -frename-registers #200549
 	use doc || export ac_cv_prog_HAVE_DOXYGEN="false"
 
@@ -41,8 +39,6 @@
 
 	econf --disable-dependency-tracking --disable-examples \
 		--disable-sdltest $(use_enable encode)
-
-	emake || die "emake failed."
 }
 
 src_install() {
