commit e02e4ccafcaf3eb8993152dfcbfbee0240ea2db2
Author: Adam Jackson <ajax@redhat.com>
Date:   Sun Feb 24 20:00:43 2008 -0500

    Bug #14029: Don't LockDisplay() recursively.
    
    See also Red Hat bugzilla #326461.

diff --git a/src/xkb/XKBBind.c b/src/xkb/XKBBind.c
index e190d00..4dc9265 100644
--- a/src/xkb/XKBBind.c
+++ b/src/xkb/XKBBind.c
@@ -368,7 +368,6 @@ XkbRefreshKeyboardMapping(register XkbMapNotifyEvent *event)
 	     changes= xkbi->changes;
 	else bzero(&changes,sizeof(changes));
 	XkbNoteMapChanges(&changes,event,XKB_XLIB_MAP_MASK);
-	LockDisplay(dpy);
 	if ((rtrn=XkbGetMapChanges(dpy,xkbi->desc,&changes))!=Success) {
 #ifdef DEBUG
 	    fprintf(stderr,"Internal Error! XkbGetMapChanges failed:\n");
@@ -379,7 +378,6 @@ XkbRefreshKeyboardMapping(register XkbMapNotifyEvent *event)
 	    xkbi->flags&= ~XkbMapPending;
 	    bzero(&xkbi->changes,sizeof(XkbMapChangesRec));
 	}
-	UnlockDisplay(dpy);
 	return rtrn;
     }
     return BadMatch;