diff -ur audiofile-0.2.6.orig/test/printmarkers.c audiofile-0.2.6/test/printmarkers.c
--- audiofile-0.2.6.orig/test/printmarkers.c	2004-03-06 08:39:23.000000000 +0200
+++ audiofile-0.2.6/test/printmarkers.c	2007-07-28 12:23:04.000000000 +0300
@@ -33,6 +33,7 @@
 #include <stdlib.h>
 #include <assert.h>
 #include <audiofile.h>
+#include <math.h>
 
 void printtime (AFframecount position, double rate)
 {
