--- /usr/portage/app-crypt/heimdal/heimdal-1.2.1-r4.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./app-crypt/heimdal/heimdal-1.2.1-r4.ebuild	2010-06-23 23:26:19.000000000 +0200
@@ -42,10 +42,7 @@
 GENTOODIR=${WORKDIR}/gentoo
 S=${WORKDIR}/${P}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	EPATCH_SUFFIX="patch" epatch "${GENTOODIR}"/patches
 
 	epatch "${FILESDIR}"/${PN}-r23238-kb5_locl_h-wind_h.patch
@@ -60,7 +57,7 @@
 	AT_M4DIR="cf" eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	# needed to work with sys-libs/e2fsprogs-libs <- should be removed!!
 	append-flags "-I/usr/include/et"
 	econf \
@@ -87,8 +84,6 @@
 		ewarn "Using old libtool with a quick hack."
 		sed -i -e's/ECHO=/echo=/' libtool
 	fi
-
-	emake || die "emake failed"
 }
 
 src_test() {
