--- setup.py.orig	2005-01-06 23:16:13.000000000 +0100
+++ setup.py	2005-01-21 17:36:43.000000000 +0100
@@ -33,18 +33,18 @@
 
 # delete all but the first one in this list if using your own LAPACK/BLAS
 sourcelist = [os.path.join('Src', 'lapack_litemodule.c'),
-              #os.path.join('Src', 'blas_lite.c'), 
-              #os.path.join('Src', 'f2c_lite.c'), 
-              #os.path.join('Src', 'zlapack_lite.c'),
-              #os.path.join('Src', 'dlapack_lite.c')
+              os.path.join('Src', 'blas_lite.c'), 
+              os.path.join('Src', 'f2c_lite.c'), 
+              os.path.join('Src', 'zlapack_lite.c'),
+              os.path.join('Src', 'dlapack_lite.c')
              ]
 # set these to use your own BLAS;
 
-library_dirs_list = ['/usr/lib/atlas']
-libraries_list = ['lapack', 'cblas', 'f77blas', 'atlas', 'g2c'] 
+#library_dirs_list = ['/usr/lib/atlas']
+#libraries_list = ['lapack', 'cblas', 'f77blas', 'atlas', 'g2c'] 
 
 # set to true (1), if you also want BLAS optimized matrixmultiply/dot/innerproduct
-use_dotblas = 1 
+use_dotblas = 0 
 include_dirs = ['/usr/include/atlas']  
                    # You may need to set this to find cblas.h
                    #  e.g. on UNIX using ATLAS this should be ['/usr/include/atlas']
@@ -86,8 +86,8 @@
                os.path.join('Src', 'linpack.c')],
               extra_compile_args = extra_compile_args),
     Extension('lapack_lite', sourcelist,
-              library_dirs = library_dirs_list,
-              libraries = libraries_list,
+#              library_dirs = library_dirs_list,
+#              libraries = libraries_list,
               extra_link_args = extra_link_args,
               extra_compile_args = extra_compile_args) 
     ]
@@ -120,8 +120,8 @@
     package_dir['dotblas'] = os.path.join('Packages', 'dotblas', 'dotblas')
     ext_modules.append(Extension('_dotblas',
                                  [os.path.join('Packages', 'dotblas', 'dotblas', '_dotblas.c')],
-                                 library_dirs = library_dirs_list,
-                                 libraries = libraries_list,
+#                                 library_dirs = library_dirs_list,
+#                                 libraries = libraries_list,
                                  extra_compile_args=extra_compile_args))
 
 
