--- /usr/portage/app-crypt/mit-krb5/mit-krb5-1.6.3-r6.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./app-crypt/mit-krb5/mit-krb5-1.6.3-r6.ebuild	2010-04-04 01:40:23.000000000 +0200
@@ -29,7 +29,9 @@
 src_unpack() {
 	unpack ${A}
 	unpack ./${MY_P}.tar.gz
-	cd "${S}"
+}
+
+src_prepare() {
 	EPATCH_SOURCE="${WORKDIR}/patch" EPATCH_SUFFIX="patch" epatch
 	epatch "${FILESDIR}/CVE-2009-0844+CVE-2009-0847.patch"
 	epatch "${FILESDIR}/CVE-2009-0846.patch"
@@ -45,7 +47,7 @@
 	done
 }
 
-src_compile() {
+src_configure() {
 	# needed to work with sys-libs/e2fsprogs-libs <- should be removed!!
 	append-flags "-I/usr/include/et"
 	econf \
@@ -54,7 +56,9 @@
 		--with-system-et --with-system-ss \
 		--enable-dns-for-realm \
 		--enable-kdc-replay-cache || die
+}
 
+src_compile() {
 	emake -j1 || die
 
 	if use doc ; then
