diff -Naur check-0.9.5-orig/check.m4 check-0.9.5/check.m4
--- check-0.9.5-orig/check.m4	2006-11-21 17:19:12.000000000 -0600
+++ check-0.9.5/check.m4	2008-12-04 21:45:13.000000000 -0600
@@ -4,8 +4,8 @@
 
 AC_DEFUN([AM_PATH_CHECK],
 [
-  AC_MSG_WARN([[AM_PATH_CHECK() is deprecated]])
-  AC_MSG_WARN([[use PKG_CHECK_MODULES([CHECK], [check >= 0.9.4]) instead]])
+  AC_MSG_WARN([A@&t@M_PATH_CHECK() is deprecated])
+  AC_MSG_WARN([[use P@&t@KG_CHECK_MODULES([CHECK], [check >= 0.9.4]) instead]])
   AC_ARG_WITH([check],
   [  --with-check=PATH       prefix where check is installed [default=auto]])
  
diff -Naur check-0.9.5-orig/doc/example/configure.ac check-0.9.5/doc/example/configure.ac
--- check-0.9.5-orig/doc/example/configure.ac	2006-11-21 17:19:14.000000000 -0600
+++ check-0.9.5/doc/example/configure.ac	2008-12-04 21:45:13.000000000 -0600
@@ -25,7 +25,7 @@
 # CHECK_LIBS accordingly.  
 #  AM_PATH_CHECK([MINIMUM-VERSION,
 #                [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
-AM_PATH_CHECK()
+PKG_CHECK_MODULES([check], [], [])
 
 # Checks for header files.
 AC_HEADER_STDC
