--- /usr/portage/sys-fs/e2fsprogs/e2fsprogs-1.41.11.ebuild	2009-12-07 12:36:43.000000000 +0100
+++ ./sys-fs/e2fsprogs/e2fsprogs-1.41.11.ebuild	2010-03-14 18:48:03.000000000 +0100
@@ -29,9 +29,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.38-tests-locale.patch #99766
 	epatch "${FILESDIR}"/${PN}-1.41.8-makefile.patch
 	epatch "${FILESDIR}"/${PN}-1.40-fbsd.patch
@@ -57,7 +53,7 @@
 	touch lib/ss/ss_err.h
 }
 
-src_compile() {
+src_configure() {
 	# Keep the package from doing silly things #261411
 	export VARTEXFONTS=${T}/fonts
 
@@ -86,6 +82,9 @@
 		eerror "attachment to http://bugs.gentoo.org/show_bug.cgi?id=81096"
 		die "Preventing included intl cruft from building"
 	fi
+}
+
+src_compile() {
 	emake COMPILE_ET=compile_et MK_CMDS=mk_cmds || die
 
 	# Build the FreeBSD helper
