--- x264.orig/Makefile	2008-04-07 21:55:13.000000000 +0200
+++ x264-snapshot-20080406-2245/Makefile	2008-04-07 21:57:25.000000000 +0200
@@ -69,7 +69,7 @@
 
 .PHONY: all default fprofiled clean distclean install install-gtk uninstall dox test testclean
 
-default: $(DEP) x264$(EXE)
+default: $(DEP)
 
 libx264.a: .depend $(OBJS) $(OBJASM)
 	ar rc libx264.a $(OBJS) $(OBJASM)
@@ -148,13 +148,12 @@
 	rm -rf test/
 	$(MAKE) -C gtk distclean
 
-install: x264$(EXE) $(SONAME)
+install: libx264.a $(SONAME)
 	install -d $(DESTDIR)$(bindir) $(DESTDIR)$(includedir)
 	install -d $(DESTDIR)$(libdir) $(DESTDIR)$(libdir)/pkgconfig
 	install -m 644 x264.h $(DESTDIR)$(includedir)
 	install -m 644 libx264.a $(DESTDIR)$(libdir)
 	install -m 644 x264.pc $(DESTDIR)$(libdir)/pkgconfig
-	install x264$(EXE) $(DESTDIR)$(bindir)
 	ranlib $(DESTDIR)$(libdir)/libx264.a
 ifeq ($(SYS),MINGW)
 	$(if $(SONAME), install -m 755 $(SONAME) $(DESTDIR)$(bindir))
