--- bin/jgrasp.orig	2008-03-24 17:58:02.000000000 -0700
+++ bin/jgrasp	2008-03-24 17:58:02.000000000 -0700
@@ -165,7 +165,13 @@
 # If not a delivered binary, run sys_jgrasp if it is present and executable.
 # Run directly otherwise.
 if [ "$OS" = Linux -a "$PROC" = iX86 ]; then
-   jgraspv_prog="$jgraspv_jgrasp_bin_dir/linux/jgrasp"
+   if [ -x "$jgraspv_jgrasp_bin_dir/linux/jgrasp" ]; then
+      # binary installation.
+      jgraspv_prog="$jgraspv_jgrasp_bin_dir/linux/jgrasp"
+   else
+      # stub build
+      jgraspv_prog="$jgraspv_jgrasp_bin_dir/sys_jgrasp"
+   fi
 elif [ "$OS" = SunOS -a "$PROC" = sparc ]; then
    jgraspv_prog="$jgraspv_jgrasp_bin_dir/solaris/jgrasp"
 elif [ "$OS" = Darwin ]; then
