--- /usr/portage/x11-libs/gtk+/gtk+-2.14.7-r2.ebuild	2009-04-27 15:36:59.000000000 +0200
+++ ./x11-libs/gtk+/gtk+-2.14.7-r2.ebuild	2010-03-13 23:14:47.000000000 +0100
@@ -28,7 +28,7 @@
 	>=dev-libs/glib-2.17.6
 	>=x11-libs/pango-1.20
 	>=dev-libs/atk-1.13
-	>=x11-libs/cairo-1.6
+	>=x11-libs/cairo-1.6[X]
 	media-libs/fontconfig
 	x11-misc/shared-mime-info
 	>=media-libs/libpng-1.2.1
@@ -51,23 +51,13 @@
 		 )"
 PDEPEND="vim-syntax? ( app-vim/gtk-syntax )"
 
-pkg_setup() {
-	if ! built_with_use x11-libs/cairo X; then
-		eerror "Please re-emerge x11-libs/cairo with the X USE flag set"
-		die "cairo needs the X flag set"
-	fi
-}
-
 set_gtk2_confdir() {
 	# An arch specific config directory is used on multilib systems
 	has_multilib_profile && GTK2_CONFDIR="/etc/gtk-2.0/${CHOST}"
 	GTK2_CONFDIR=${GTK2_CONFDIR:=/etc/gtk-2.0}
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# use an arch-specific config directory so that 32bit and 64bit versions
 	# dont clash on multilib systems
 	has_multilib_profile && epatch "${FILESDIR}/${PN}-2.8.0-multilib.patch"
@@ -96,7 +86,7 @@
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	# png always on to display icons (foser)
 	local myconf="$(use_enable doc gtk-doc) \
 		$(use_with jpeg libjpeg) \
@@ -112,7 +102,6 @@
 	use debug && myconf="${myconf} --enable-debug=yes"
 
 	econf ${myconf} || die "configure failed"
-	emake || die "compile failed"
 }
 
 src_test() {
