From 4e7f090d1b2e2bb7c6a47d6f45edc8adb335e9a3 Mon Sep 17 00:00:00 2001
From: Didier Barvaux <didier-bugzilla@barvaux.org>
Date: Sat, 17 Apr 2010 21:25:29 +0000
Subject: Add --enable-tools/--disable-tools options to avoid building miscellaneous tools

Fix for bug #614123.
---
diff --git a/Makefile.am b/Makefile.am
index 67e6411..c562c32 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,9 @@ SUBDIRS = . gdk-pixbuf-loader gtk-engine data tests doc
 
 lib_LTLIBRARIES = librsvg-2.la
 bin_PROGRAMS = rsvg-convert $(target_rsvg_view)
+if BUILD_MISC_TOOLS
 noinst_PROGRAMS = test-performance rsvg-dimensions
+endif
 
 man_MANS = rsvg.1
 
diff --git a/configure.in b/configure.in
index 6e102ae..a1b547a 100644
--- a/configure.in
+++ b/configure.in
@@ -286,6 +286,23 @@ GTK_DOC_CHECK([1.0])
 AM_CONDITIONAL(GTK_DOC_INSTALLED, $gtk_doc_installed)
 
 dnl ===========================================================================
+dnl Do we build the miscellaenous tools provided with librsvg ?
+
+AC_ARG_ENABLE(tools,
+              AC_HELP_STRING([--disable-tools],
+                             [do not build miscellaenous tools [default=no]]),
+              [build_misc_tools=$enableval],
+              [build_misc_tools=yes])
+
+AC_MSG_CHECKING(whether to build miscellaenous tools)
+if test "x$build_misc_tools" = "xyes" ; then
+	AC_MSG_RESULT(yes)
+else
+	AC_MSG_RESULT(no)
+fi
+AM_CONDITIONAL([BUILD_MISC_TOOLS], [test "x$build_misc_tools" = "xyes"])
+
+dnl ===========================================================================
 
 m4_copy([AC_DEFUN],[glib_DEFUN])
 glib_DEFUN([GLIB_LC_MESSAGES],
@@ -334,6 +351,7 @@ librsvg-$LIBRSVG_VERSION
 	
 	Build GdkPixbuf loader:         ${enable_pixbuf_loader}
 	Build theme engine:             ${enable_gtk_theme}
+	Build miscellaenous tools:      ${build_misc_tools}
 	Handle svgz files:              ${test_gsf}
 	Use GIO:                        ${test_gio}
 	Use libcroco for css parsing:   ${test_croco}
diff --git a/tests/pdiff/Makefile.am b/tests/pdiff/Makefile.am
index 47ed8c1..8e77972 100644
--- a/tests/pdiff/Makefile.am
+++ b/tests/pdiff/Makefile.am
@@ -1,3 +1,5 @@
+if BUILD_MISC_TOOLS
+
 noinst_PROGRAMS = perceptualdiff
 
 noinst_LTLIBRARIES = libpdiff.la
@@ -14,3 +16,6 @@ perceptualdiff_SOURCES =	\
 
 INCLUDES = $(LIBRSVG_CFLAGS)
 LDADD = libpdiff.la $(LIBRSVG_LIBS) -lm
+
+endif
+
--
cgit v0.8.3.1
