http://bugzilla.gnome.org/show_bug.cgi?id=518384

Bug 518384 – libsoup-2.2.105 and 2.3.x API documentation does not work in devhelp

diff -urNp libsoup-2.2.105.orig/docs/reference/html/libsoup.devhelp libsoup-2.2.105/docs/reference/html/libsoup.devhelp
--- libsoup-2.2.105.orig/docs/reference/html/libsoup.devhelp	2008-02-11 21:51:09.000000000 +0200
+++ libsoup-2.2.105/docs/reference/html/libsoup.devhelp	2008-02-24 13:54:57.000000000 +0200
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8" standalone="no"?>
 <!DOCTYPE book PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "">
-<book xmlns="http://www.devhelp.net/book" title="libsoup Reference Manual" link="index.html" author="" name="libsoup">
+<book xmlns="http://www.devhelp.net/book" title="libsoup Reference Manual" link="index.html" author="" name="libsoup-2.2">
   <chapters>
     <sub name="libsoup Tutorial" link="ch01.html">
       <sub name="Soup Client Basics" link="libsoup-client-howto.html"/>
diff -urNp libsoup-2.2.105.orig/docs/reference/html/libsoup.devhelp2 libsoup-2.2.105/docs/reference/html/libsoup.devhelp2
--- libsoup-2.2.105.orig/docs/reference/html/libsoup.devhelp2	2008-02-11 21:51:09.000000000 +0200
+++ libsoup-2.2.105/docs/reference/html/libsoup.devhelp2	2008-02-24 13:54:57.000000000 +0200
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8" standalone="no"?>
 <!DOCTYPE book PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "">
-<book xmlns="http://www.devhelp.net/book" title="libsoup Reference Manual" link="index.html" author="" name="libsoup" version="2">
+<book xmlns="http://www.devhelp.net/book" title="libsoup Reference Manual" link="index.html" author="" name="libsoup-2.2" version="2">
   <chapters>
     <sub name="libsoup Tutorial" link="ch01.html">
       <sub name="Soup Client Basics" link="libsoup-client-howto.html"/>
diff -urNp libsoup-2.2.105.orig/docs/reference/Makefile.am libsoup-2.2.105/docs/reference/Makefile.am
--- libsoup-2.2.105.orig/docs/reference/Makefile.am	2008-02-24 12:15:53.000000000 +0200
+++ libsoup-2.2.105/docs/reference/Makefile.am	2008-02-24 13:54:57.000000000 +0200
@@ -2,7 +2,7 @@
 AUTOMAKE_OPTIONS = 1.6
 
 # The name of the module
-DOC_MODULE=libsoup
+DOC_MODULE=libsoup-2.2
 
 # The top-level SGML file.
 DOC_MAIN_SGML_FILE=$(DOC_MODULE)-docs.sgml
@@ -54,7 +54,3 @@ GTKDOC_LIBS = $(top_builddir)/libsoup/li
 
 # include common portion ...
 include $(top_srcdir)/gtk-doc.make
-
-# Override
-override TARGET_DIR=$(HTML_DIR)/$(DOC_MODULE)-$(SOUP_API_VERSION)
-
diff -urNp libsoup-2.2.105.orig/docs/reference/Makefile.in libsoup-2.2.105/docs/reference/Makefile.in
--- libsoup-2.2.105.orig/docs/reference/Makefile.in	2008-02-24 12:15:53.000000000 +0200
+++ libsoup-2.2.105/docs/reference/Makefile.in	2008-02-24 14:06:57.000000000 +0200
@@ -192,7 +192,7 @@ top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.6
 
 # The name of the module
-DOC_MODULE = libsoup
+DOC_MODULE = libsoup-2.2
 
 # The top-level SGML file.
 DOC_MAIN_SGML_FILE = $(DOC_MODULE)-docs.sgml
@@ -552,9 +552,6 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 .PHONY : dist-hook-local docs
 
 # include common portion ...
-
-# Override
-override TARGET_DIR=$(HTML_DIR)/$(DOC_MODULE)-$(SOUP_API_VERSION)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
