--- /usr/portage/dev-lang/python/python-2.6.6-r1.ebuild	2010-12-22 16:05:36.000000000 +0100
+++ ./dev-lang/python/python-2.6.6-r1.ebuild	2010-12-25 00:21:09.000000000 +0100
@@ -46,7 +46,8 @@
 			tk? ( >=dev-lang/tk-8.0 )
 			xml? ( >=dev-libs/expat-2 )
 		)
-		doc? ( dev-python/python-docs:${SLOT} )"
+		doc? ( dev-python/python-docs:${SLOT} )
+		app-arch/bzip2"
 DEPEND="${RDEPEND}
 		dev-util/pkgconfig
 		!sys-devel/gcc[libffi]"
