--- /usr/portage/dev-lang/python/python-2.7.ebuild	2010-07-10 15:07:16.000000000 +0200
+++ ./dev-lang/python/python-2.7.ebuild	2010-07-11 18:28:38.000000000 +0200
@@ -48,7 +48,8 @@
 			tk? ( >=dev-lang/tk-8.0 )
 			xml? ( >=dev-libs/expat-2 )
 		)
-		doc? ( dev-python/python-docs:${SLOT} )"
+		doc? ( dev-python/python-docs:${SLOT} )
+		app-arch/bzip2"
 DEPEND="${RDEPEND}
 		dev-util/pkgconfig
 		!sys-devel/gcc[libffi]"
