--- /usr/portage/dev-lang/python/python-3.1.3.ebuild	2010-12-16 17:35:54.000000000 +0100
+++ ./dev-lang/python/python-3.1.3.ebuild	2010-12-25 00:27:42.000000000 +0100
@@ -49,7 +49,8 @@
 			ssl? ( dev-libs/openssl )
 			tk? ( >=dev-lang/tk-8.0 )
 			xml? ( >=dev-libs/expat-2 )
-		)"
+		)
+		app-arch/bzip2"
 DEPEND="${RDEPEND}
 		$([[ "${PV}" == *_pre* ]] && echo "=${CATEGORY}/${PN}-${PV%%.*}*")
 		dev-util/pkgconfig
