--- cipher/md.c.orig	2007-08-14 02:46:51.000000000 +1000
+++ cipher/md.c	2008-11-06 19:05:37.000000000 +1100
@@ -791,7 +791,7 @@
   if ( !algo )
     return GPG_ERR_DIGEST_ALGO; /* i.e. no algo enabled */
 
-  if ( keylen > 64 ) 
+  if ( keylen > hd->ctx->macpads_Bsize ) 
     {
       helpkey = gcry_malloc_secure ( md_digest_length( algo ) );
       if ( !helpkey )
@@ -799,7 +799,7 @@
       gcry_md_hash_buffer ( algo, helpkey, key, keylen );
       key = helpkey;
       keylen = md_digest_length( algo );
-      assert ( keylen <= 64 );
+      assert ( keylen <= hd->ctx->macpads_Bsize );
     }
 
   memset ( hd->ctx->macpads, 0, 2*(hd->ctx->macpads_Bsize) );
