--- ./media-libs/libmikmod/libmikmod-3.1.11-r5.ebuild.orig	2010-02-27 01:04:36.000000000 +0100
+++ ./media-libs/libmikmod/libmikmod-3.1.11-r5.ebuild	2010-02-27 01:05:17.000000000 +0100
@@ -19,9 +19,7 @@
 	esd? ( >=media-sound/esound-0.2.19 )"
 DEPEND="${RDEPEND}"
 
-src_unpack() {
-	unpack ${A}
-
+src_prepare() {
 	epatch "${DISTDIR}"/${P}-esdm4.patch.bz2
 
 	cd "${S}"
@@ -33,12 +31,11 @@
 	AT_M4DIR="${S}/m4" eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	econf --enable-af \
 		$(use_enable esd) \
 		$(use_enable alsa) \
 		$(use_enable oss)
-	emake || die "emake failed."
 }
 
 src_install() {
