From b9badc5d0f570bbb8353f939146163bdf0bc65c1 Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Mon, 31 May 2010 10:08:07 +0000
Subject: Fix build error due to missing semicolon

---
diff --git a/monitor/afc/afcvolume.c b/monitor/afc/afcvolume.c
index 0adce77..0784716 100644
--- a/monitor/afc/afcvolume.c
+++ b/monitor/afc/afcvolume.c
@@ -71,7 +71,7 @@ _g_vfs_afc_volume_update_metadata (GVfsAfcVolume *self)
   idevice_t dev;
   lockdownd_client_t lockdown_cli = NULL;
   idevice_error_t err;
-  guint retries
+  guint retries;
   plist_t value;
   char *model, *display_name;
 
--
cgit v0.8.3.1
