--- /usr/portage/sys-libs/gdbm/gdbm-1.8.3-r4.ebuild	2010-02-06 07:37:17.000000000 +0100
+++ ./sys-libs/gdbm/gdbm-1.8.3-r4.ebuild	2010-03-14 17:57:29.000000000 +0100
@@ -15,9 +15,7 @@
 
 DEPEND="berkdb? ( sys-libs/db )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-fix-install-ownership.patch #24178
 	epatch "${FILESDIR}"/${P}-compat-linking.patch #165263
 	epatch "${FILESDIR}"/${P}-build.patch #209730
@@ -25,14 +23,13 @@
 	append-lfs-flags
 }
 
-src_compile() {
+src_configure() {
 	use berkdb || export ac_cv_lib_dbm_main=no ac_cv_lib_ndbm_main=no
 	econf \
 		--includedir=/usr/include/gdbm \
 		--disable-dependency-tracking \
 		--enable-fast-install \
 		|| die
-	emake || die
 }
 
 src_install() {
