--- /usr/portage/sys-libs/gdbm/gdbm-1.8.3-r3.ebuild	2009-04-05 10:08:24.000000000 +0200
+++ ./sys-libs/gdbm/gdbm-1.8.3-r3.ebuild	2010-03-14 17:56:37.000000000 +0100
@@ -15,18 +15,15 @@
 
 DEPEND="berkdb? ( sys-libs/db )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-fix-install-ownership.patch #24178
 	epatch "${FILESDIR}"/${P}-compat-linking.patch #165263
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	use berkdb || export ac_cv_lib_dbm_main=no ac_cv_lib_ndbm_main=no
 	econf --includedir=/usr/include/gdbm || die
-	emake || die
 }
 
 src_install() {
