diff -ruN libdc1394-1.2.1.orig/examples/Makefile.in libdc1394-1.2.1/examples/Makefile.in
--- libdc1394-1.2.1.orig/examples/Makefile.in	2006-03-31 06:53:44.000000000 +0200
+++ libdc1394-1.2.1/examples/Makefile.in	2006-05-07 09:59:31.000000000 +0200
@@ -38,7 +38,7 @@
 host_triplet = @host@
 bin_PROGRAMS = dc1394_vloopback$(EXEEXT)
 noinst_PROGRAMS = grab_gray_image$(EXEEXT) grab_partial_image$(EXEEXT) \
-	dc1394_multiview$(EXEEXT) grab_color_image$(EXEEXT)
+	grab_color_image$(EXEEXT)
 subdir = examples
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -51,13 +51,7 @@
 am__installdirs = "$(DESTDIR)$(bindir)"
 binPROGRAMS_INSTALL = $(INSTALL_PROGRAM)
 PROGRAMS = $(bin_PROGRAMS) $(noinst_PROGRAMS)
-am_dc1394_multiview_OBJECTS =  \
-	dc1394_multiview-dc1394_multiview.$(OBJEXT)
-dc1394_multiview_OBJECTS = $(am_dc1394_multiview_OBJECTS)
 am__DEPENDENCIES_1 =
-dc1394_multiview_DEPENDENCIES = ../libdc1394/libdc1394_control.la \
-	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_1)
 am_dc1394_vloopback_OBJECTS = dc1394_vloopback.$(OBJEXT) \
 	affine.$(OBJEXT)
 dc1394_vloopback_OBJECTS = $(am_dc1394_vloopback_OBJECTS)
@@ -82,10 +76,10 @@
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
-SOURCES = $(dc1394_multiview_SOURCES) $(dc1394_vloopback_SOURCES) \
+SOURCES = $(dc1394_vloopback_SOURCES) \
 	$(grab_color_image_SOURCES) $(grab_gray_image_SOURCES) \
 	$(grab_partial_image_SOURCES)
-DIST_SOURCES = $(dc1394_multiview_SOURCES) $(dc1394_vloopback_SOURCES) \
+DIST_SOURCES = $(dc1394_vloopback_SOURCES) \
 	$(grab_color_image_SOURCES) $(grab_gray_image_SOURCES) \
 	$(grab_partial_image_SOURCES)
 ETAGS = etags
@@ -202,8 +196,6 @@
 grab_partial_image_SOURCES = grab_partial_image.c
 grab_partial_image_LDADD = ../libdc1394/libdc1394_control.la -lraw1394
 dc1394_multiview_CFLAGS = $(X_CFLAGS)
-dc1394_multiview_SOURCES = dc1394_multiview.c
-dc1394_multiview_LDADD = ../libdc1394/libdc1394_control.la -lraw1394 $(X_LIBS) $(X_PRE_LIBS) -lXv -lXext -lX11 $(X_EXTRA_LIBS)
 dc1394_vloopback_SOURCES = dc1394_vloopback.c affine.c affine.h
 dc1394_vloopback_LDADD = -lm ../libdc1394/libdc1394_control.la -lraw1394
 all: all-am
@@ -276,7 +268,7 @@
 	done
 dc1394_multiview$(EXEEXT): $(dc1394_multiview_OBJECTS) $(dc1394_multiview_DEPENDENCIES) 
 	@rm -f dc1394_multiview$(EXEEXT)
-	$(LINK) $(dc1394_multiview_LDFLAGS) $(dc1394_multiview_OBJECTS) $(dc1394_multiview_LDADD) $(LIBS)
+	$(LINK) $(dc1394_multiview_OBJECTS) $(dc1394_multiview_LDADD) $(LIBS)
 dc1394_vloopback$(EXEEXT): $(dc1394_vloopback_OBJECTS) $(dc1394_vloopback_DEPENDENCIES) 
 	@rm -f dc1394_vloopback$(EXEEXT)
 	$(LINK) $(dc1394_vloopback_LDFLAGS) $(dc1394_vloopback_OBJECTS) $(dc1394_vloopback_LDADD) $(LIBS)
