diff -urN libdc1394-2.0.0-pre4.orig/examples/Makefile.in libdc1394-2.0.0-pre4/examples/Makefile.in
--- libdc1394-2.0.0-pre4.orig/examples/Makefile.in	2005-08-04 04:32:32.000000000 -0400
+++ libdc1394-2.0.0-pre4/examples/Makefile.in	2005-09-12 12:09:51.000000000 -0400
@@ -90,7 +90,7 @@
 
 MAINTAINERCLEANFILES = Makefile.in
 bin_PROGRAMS = dc1394_vloopback
-noinst_PROGRAMS = grab_gray_image grab_partial_image dc1394_multiview grab_color_image
+noinst_PROGRAMS = grab_gray_image grab_partial_image grab_color_image
 
 grab_gray_image_SOURCES = grab_gray_image.c
 grab_gray_image_LDADD = ../libdc1394/libdc1394_control.la -lraw1394
@@ -101,10 +101,6 @@
 grab_partial_image_SOURCES = grab_partial_image.c
 grab_partial_image_LDADD = ../libdc1394/libdc1394_control.la -lraw1394
 
-dc1394_multiview_CFLAGS = $(X_CFLAGS)
-dc1394_multiview_SOURCES = dc1394_multiview.c
-dc1394_multiview_LDADD = ../libdc1394/libdc1394_control.la -lraw1394 $(X_LIBS) $(X_PRE_LIBS) -lXv -lXext -lX11 $(X_EXTRA_LIBS)
-
 dc1394_vloopback_SOURCES = dc1394_vloopback.c affine.c affine.h
 dc1394_vloopback_LDADD = -lm ../libdc1394/libdc1394_control.la -lraw1394
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
@@ -112,7 +108,7 @@
 CONFIG_CLEAN_FILES = 
 bin_PROGRAMS =  dc1394_vloopback$(EXEEXT)
 noinst_PROGRAMS =  grab_gray_image$(EXEEXT) grab_partial_image$(EXEEXT) \
-dc1394_multiview$(EXEEXT) grab_color_image$(EXEEXT)
+ grab_color_image$(EXEEXT)
 PROGRAMS =  $(bin_PROGRAMS) $(noinst_PROGRAMS)
 
 
@@ -133,9 +129,6 @@
 grab_partial_image_OBJECTS =  grab_partial_image.$(OBJEXT)
 grab_partial_image_DEPENDENCIES =  ../libdc1394/libdc1394_control.la
 grab_partial_image_LDFLAGS = 
-dc1394_multiview_OBJECTS =  dc1394_multiview.$(OBJEXT)
-dc1394_multiview_DEPENDENCIES =  ../libdc1394/libdc1394_control.la
-dc1394_multiview_LDFLAGS = 
 grab_color_image_OBJECTS =  grab_color_image.$(OBJEXT)
 grab_color_image_DEPENDENCIES =  ../libdc1394/libdc1394_control.la
 grab_color_image_LDFLAGS = 
@@ -151,8 +144,8 @@
 
 TAR = tar
 GZIP_ENV = --best
-SOURCES = $(dc1394_vloopback_SOURCES) $(grab_gray_image_SOURCES) $(grab_partial_image_SOURCES) $(dc1394_multiview_SOURCES) $(grab_color_image_SOURCES)
-OBJECTS = $(dc1394_vloopback_OBJECTS) $(grab_gray_image_OBJECTS) $(grab_partial_image_OBJECTS) $(dc1394_multiview_OBJECTS) $(grab_color_image_OBJECTS)
+SOURCES = $(dc1394_vloopback_SOURCES) $(grab_gray_image_SOURCES) $(grab_partial_image_SOURCES) $(grab_color_image_SOURCES)
+OBJECTS = $(dc1394_vloopback_OBJECTS) $(grab_gray_image_OBJECTS) $(grab_partial_image_OBJECTS) $(grab_color_image_OBJECTS)
 
 all: all-redirect
 .SUFFIXES:
@@ -255,10 +248,6 @@
 	@rm -f grab_partial_image$(EXEEXT)
 	$(LINK) $(grab_partial_image_LDFLAGS) $(grab_partial_image_OBJECTS) $(grab_partial_image_LDADD) $(LIBS)
 
-dc1394_multiview$(EXEEXT): $(dc1394_multiview_OBJECTS) $(dc1394_multiview_DEPENDENCIES)
-	@rm -f dc1394_multiview$(EXEEXT)
-	$(LINK) $(dc1394_multiview_LDFLAGS) $(dc1394_multiview_OBJECTS) $(dc1394_multiview_LDADD) $(LIBS)
-
 grab_color_image$(EXEEXT): $(grab_color_image_OBJECTS) $(grab_color_image_DEPENDENCIES)
 	@rm -f grab_color_image$(EXEEXT)
 	$(LINK) $(grab_color_image_LDFLAGS) $(grab_color_image_OBJECTS) $(grab_color_image_LDADD) $(LIBS)
