--- ./app-arch/bzip2/bzip2-1.0.5-r1.ebuild.orig	2010-02-14 01:10:39.000000000 +0100
+++ ./app-arch/bzip2/bzip2-1.0.5-r1.ebuild	2010-02-14 01:11:13.000000000 +0100
@@ -29,7 +29,7 @@
 	# - Generate symlinks instead of hardlinks
 	# - pass custom variables to control libdir
 	sed -i \
-		-e 's:ln -s -f $(PREFIX)/bin/:ln -s :' \
+		-e 's:ln -s -f $(PREFIX)/bin/:ln -s -f :' \
 		-e 's:$(PREFIX)/lib:$(PREFIX)/$(LIBDIR):g' \
 		Makefile || die "sed links"
 
