--- config.tests/x11/xfreetype.test	2008-01-15 20:09:15.000000000 +0100
+++ config.tests/x11/xfreetype.test.new	2009-07-13 00:09:07.300309960 +0200
@@ -119,7 +119,7 @@
     INC="freetype2/freetype/freetype.h"
     XDIRS=`sed -n -e '/^QMAKE_INCDIR_X11[	]*=/ { s/[^=]*=[	 ]*//; s/-I/ /g; p; }' $XCONFIG`
     LDIRS=`sed -n -e '/^QMAKE_INCDIR[	]*=/ { s/[^=]*=[	 ]*//; s/-I/ /g; p; }' $XCONFIG`
-    INCDIRS="$IN_INCDIRS $XDIRS $LDIRS /usr/include /include"
+    INCDIRS="$IN_INCDIRS $XDIRS $LDIRS /usr/include /include /usr/include/gentoo-multilib/amd64"
     F=
     for INCDIR in $INCDIRS; do
 	if [ -f $INCDIR/$INC ]; then
