diff -Naur tcl8.4.11.orig/unix/configure tcl8.4.11/unix/configure
--- tcl8.4.11.orig/unix/configure	2005-06-30 01:40:33.000000000 +0900
+++ tcl8.4.11/unix/configure	2005-07-02 20:23:23.000000000 +0900
@@ -7664,7 +7664,7 @@
 DBGX='${TCL_DBGX}'
 eval "TCL_LIB_FILE=${TCL_LIB_FILE}"
 
-TCL_LIBRARY='$(prefix)/lib/tcl$(VERSION)'
+TCL_LIBRARY='$(libdir)/tcl$(VERSION)'
 PRIVATE_INCLUDE_DIR='$(includedir)'
 HTML_DIR='$(DISTDIR)/html'
 
@@ -7788,7 +7788,7 @@
 
 if test "$FRAMEWORK_BUILD" = "1" ; then
     TCL_PACKAGE_PATH="~/Library/Tcl /Library/Tcl /Network/Library/Tcl /System/Library/Tcl ~/Library/Frameworks /Library/Frameworks /Network/Library/Frameworks /System/Library/Frameworks"
-elif test "$prefix" != "$exec_prefix"; then
+elif test "$prefix" != "$exec_prefix" -o "${libdir}" != "${prefix}/lib"; then
     TCL_PACKAGE_PATH="${libdir} ${prefix}/lib"
 else
     TCL_PACKAGE_PATH="${prefix}/lib"
diff -Naur tcl8.4.11.orig/unix/configure.in tcl8.4.11/unix/configure.in
--- tcl8.4.11.orig/unix/configure.in	2005-06-30 01:40:33.000000000 +0900
+++ tcl8.4.11/unix/configure.in	2005-07-02 20:23:12.000000000 +0900
@@ -459,7 +459,7 @@
 DBGX='${TCL_DBGX}'
 eval "TCL_LIB_FILE=${TCL_LIB_FILE}"
 
-TCL_LIBRARY='$(prefix)/lib/tcl$(VERSION)'
+TCL_LIBRARY='$(libdir)/tcl$(VERSION)'
 PRIVATE_INCLUDE_DIR='$(includedir)'
 HTML_DIR='$(DISTDIR)/html'
 
@@ -552,7 +552,7 @@
 
 if test "$FRAMEWORK_BUILD" = "1" ; then
     TCL_PACKAGE_PATH="~/Library/Tcl /Library/Tcl /Network/Library/Tcl /System/Library/Tcl ~/Library/Frameworks /Library/Frameworks /Network/Library/Frameworks /System/Library/Frameworks"
-elif test "$prefix" != "$exec_prefix"; then
+elif test "$prefix" != "$exec_prefix" -o "${libdir}" != "${prefix}/lib"; then
     TCL_PACKAGE_PATH="${libdir} ${prefix}/lib"
 else
     TCL_PACKAGE_PATH="${prefix}/lib"
