diff -NrU5 OpenJPEG_v1_3.orig/Makefile OpenJPEG_v1_3/Makefile
--- OpenJPEG_v1_3.orig/Makefile	2009-02-09 21:25:42.000000000 +0100
+++ OpenJPEG_v1_3/Makefile	2009-02-09 21:34:35.000000000 +0100
@@ -24,11 +24,11 @@
 MODULES = $(SRCS:.c=.o)
 CFLAGS = $(COMPILERFLAGS) $(INCLUDE)
 
 TARGET  = openjpeg
 STATICLIB = lib$(TARGET).a
-SHAREDLIB = lib$(TARGET)-$(VER_MAJOR).$(VER_MINOR).so
+SHAREDLIB = lib$(TARGET).so.$(VER_MAJOR).$(VER_MINOR)
 LIBNAME = lib$(TARGET).so.$(VER_MAJOR)
 
 
 
 default: all
@@ -52,20 +52,20 @@
 
 $(STATICLIB): $(MODULES)
 	$(AR) r $@ $(MODULES)
 
 $(SHAREDLIB): $(MODULES)
-	$(CC) -s -shared -Wl,-soname,$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
 
 install: OpenJPEG
 	install -d '$(DESTDIR)$(INSTALL_LIBDIR)' '$(DESTDIR)$(INSTALL_INCLUDE)'
-	install -m 644 -o root -g root $(STATICLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
+	install -m 644 $(STATICLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
 	ranlib '$(DESTDIR)$(INSTALL_LIBDIR)/$(STATICLIB)'
-	install -m 755 -o root -g root $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
+	install -m 755 $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
 	ln -sf $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)/$(LIBNAME)'
-	install -m 644 -o root -g root libopenjpeg/openjpeg.h '$(DESTDIR)$(INSTALL_INCLUDE)'
-	-ldconfig
+	ln -sf $(LIBNAME) '$(DESTDIR)$(INSTALL_LIBDIR)/lib$(TARGET).so'
+	install -m 644 libopenjpeg/openjpeg.h '$(DESTDIR)$(INSTALL_INCLUDE)'
 
 clean:
 	rm -rf core dist/ u2dtmp* $(MODULES) $(STATICLIB) $(SHAREDLIB) $(LIBNAME)
 
 osx:
