--- OpenJPEG/Makefile.orig	2007-02-13 09:49:00.000000000 +0100
+++ OpenJPEG/Makefile	2007-02-28 08:28:58.000000000 +0100
@@ -36,7 +36,7 @@
 dist: OpenJPEG
 	mkdir -p dist
 	cp *.a dist
-	mv *.so dist
+	cp *.so dist
 	cp libopenjpeg/openjpeg.h dist
 
 dos2unix:
@@ -51,7 +51,7 @@
 	$(AR) r $@ $(MODULES)
 
 $(SHAREDLIB): $(MODULES)
-	$(CC) -s -shared -Wl,-soname,$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
+	$(CC) $(LDFLAGS) -s -shared -Wl,-soname,$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
 
 install:
 	install -m 644 -o root -g root $(STATICLIB) $(INSTALLDIR)
