--- ./x11-misc/notification-daemon/notification-daemon-0.3.7-r1.ebuild.orig	2010-02-28 22:45:46.000000000 +0100
+++ ./x11-misc/notification-daemon/notification-daemon-0.3.7-r1.ebuild	2010-02-28 22:46:16.000000000 +0100
@@ -26,10 +26,7 @@
 
 DOCS="AUTHORS ChangeLog NEWS"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# fix line wrapping, upstream bug #162
 	epatch "${FILESDIR}"/${PN}-0.3.7-line-wrapping.patch
 }
