--- src/NetworkManager.conf.orig	2008-11-03 23:49:29.772317558 +0000
+++ src/NetworkManager.conf	2008-11-03 23:52:29.316349336 +0000
@@ -11,7 +11,7 @@
                 <allow send_destination="org.freedesktop.NetworkManager.PPP"/>
                 <allow send_interface="org.freedesktop.NetworkManager.PPP"/>
         </policy>
-        <policy at_console="true">
+        <policy group="plugdev">
                 <allow send_destination="org.freedesktop.NetworkManager"/>
                 <allow send_interface="org.freedesktop.NetworkManager"/>
         </policy>
