--- examples/rlfe/pty.c
+++ examples/rlfe/pty.c
@@ -23,6 +23,7 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <unistd.h>
 #include <fcntl.h>
 #include <signal.h>
 
@@ -34,7 +34,7 @@
 #endif
 
 /* for solaris 2.1, Unixware (SVR4.2) and possibly others */
-#ifdef HAVE_SVR4_PTYS
+#if defined(HAVE_SYS_STROPTS_H)
 # include <sys/stropts.h>
 #endif
 
--- examples/rlfe/rlfe.c
+++ examples/rlfe/rlfe.c
@@ -73,6 +73,9 @@
 #include <termios.h>
 
 #include "config.h"
+#include "extern.h"
+
+#include <sys/wait.h>
 
 #ifdef READLINE_LIBRARY
 #  include "readline.h"
--- configure
+++ configure
@@ -4956,7 +4956,7 @@
 
 
 
-for ac_header in sys/pte.h sys/stream.h sys/select.h sys/file.h
+for ac_header in sys/pte.h sys/stream.h sys/select.h sys/file.h sys/stropts.h sys/wait.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if eval "test \"\${$as_ac_Header+set}\" = set"; then

--- configure.in
+++ configure.in
@@ -148,7 +148,7 @@
 
 AC_CHECK_HEADERS(fcntl.h unistd.h stdlib.h varargs.h stdarg.h string.h strings.h \
 		limits.h locale.h pwd.h memory.h termcap.h termios.h termio.h)
-AC_CHECK_HEADERS(sys/pte.h sys/stream.h sys/select.h sys/file.h)
+AC_CHECK_HEADERS(sys/pte.h sys/stream.h sys/select.h sys/file.h sys/stropts.h sys/wait.h)
 
 AC_CHECK_HEADERS(sys/ptem.h,,,
 [[
--- config.h.in
+++ config.h.in
@@ -141,6 +141,12 @@
 /* Define if you have the <sys/file.h> header file.  */
 #undef HAVE_SYS_FILE_H
 
+/* Define if you have the <sys/stropts.h> header file.  */
+#undef HAVE_SYS_STROPTS_H
+
+/* Define if you have the <sys/wait.h> header file.  */
+#undef HAVE_SYS_WAIT_H
+
 /* Define if you have the <sys/ndir.h> header file.  */
 #undef HAVE_SYS_NDIR_H
 
