--- a/examples/rlfe/Makefile.in
+++ b/examples/rlfe/Makefile.in
@@ -55,7 +55,7 @@ rlfe-$(VERSION).tar.gz:
 	tar czf $@ $(CFILES) $(HFILES) $(EXTRA_DIST)
 
 .c.o:
-	$(CC) -c -I. -I$(srcdir) $(M_CFLAGS) $(DEFS) $(OPTIONS) $(CFLAGS) $<
+	$(CC) -c -I. -I$(srcdir) $(CPPFLAGS) $(M_CFLAGS) $(DEFS) $(OPTIONS) $(CFLAGS) $<
 
 install_bin: .version screen
 	-if [ -f $(DESTDIR)$(bindir)/$(SCREEN) ] && [ ! -f $(DESTDIR)$(bindir)/$(SCREEN).old ]; \
@@ -170,6 +170,9 @@ depend.in: $(CFILES) term.h
 	for i in $(CFILES); do echo $$i; $(CPP_DEPEND) $$i >> tmp_make; done 
 	mv tmp_make Makefile.in
 
+Makefile: $(srcdir)/Makefile.in
+	./config.status
+
 ###############################################################################
 
 ### Dependencies:
