--- /usr/portage/dev-libs/libcdio/libcdio-0.80-r1.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./dev-libs/libcdio/libcdio-0.80-r1.ebuild	2010-06-05 22:54:40.000000000 +0200
@@ -21,10 +21,7 @@
 	sys-devel/gettext
 	dev-util/pkgconfig"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-minimal.patch
 	epatch "${FILESDIR}"/${P}-fix-pkgconfig.patch
 	epatch "${FILESDIR}"/${P}-fbsd.patch
@@ -45,7 +42,7 @@
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		$(use_enable cddb) \
 		$(use_with !minimal cd-drive) \
@@ -61,7 +58,6 @@
 		--disable-vcd-info \
 		--disable-dependency-tracking \
 		--disable-maintainer-mode || die "configure failed"
-	emake || die "make failed"
 }
 
 src_install() {
