--- /usr/portage/gnome-extra/bug-buddy/bug-buddy-2.26.0.ebuild	2010-01-18 01:08:52.000000000 +0100
+++ ./gnome-extra/bug-buddy/bug-buddy-2.26.0.ebuild	2010-06-05 21:11:38.000000000 +0200
@@ -25,7 +25,7 @@
 	|| ( dev-libs/elfutils dev-libs/libelf )
 	>=sys-devel/gdb-5.1"
 
-DEPEND=${RDEPEND}"
+DEPEND="${RDEPEND}
 	>=app-text/gnome-doc-utils-0.3.2
 	>=dev-util/pkgconfig-0.9
 	>=dev-util/intltool-0.40"
