--- /usr/portage/dev-lang/ruby/ruby-1.8.7_p249.ebuild	2010-05-06 22:05:38.000000000 +0200
+++ ./dev-lang/ruby/ruby-1.8.7_p249.ebuild	2010-06-06 01:20:49.000000000 +0200
@@ -54,10 +54,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	EPATCH_FORCE="yes" EPATCH_SUFFIX="patch" \
 		epatch "${WORKDIR}/patches-${PVR}"
 
@@ -68,7 +65,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	# -fomit-frame-pointer makes ruby segfault, see bug #150413.
 	filter-flags -fomit-frame-pointer
 	# In many places aliasing rules are broken; play it safe
@@ -105,7 +102,9 @@
 		--with-sitedir=/usr/$(get_libdir)/ruby/site_ruby \
 		--enable-option-checking=no \
 		|| die "econf failed"
+}
 
+src_compile() {
 	emake EXTLDFLAGS="${LDFLAGS}" || die "emake failed"
 }
 
