Index: pulseaudio-0.9.8/configure.ac
===================================================================
--- pulseaudio-0.9.8.orig/configure.ac
+++ pulseaudio-0.9.8/configure.ac
@@ -836,7 +836,7 @@ if test "x$HAVE_HAL" = x1 ; then
    dbus=yes
 fi
 
-if test "x${dbus}" != xno ; then
+if test "x${dbus}" != xno || test "x${bluez}" != xno || "x${hal}" != xno ; then
 
     PKG_CHECK_MODULES(DBUS, [ dbus-1 >= 1.0.0 ],
         [
Index: pulseaudio-0.9.8/src/Makefile.am
===================================================================
--- pulseaudio-0.9.8.orig/src/Makefile.am
+++ pulseaudio-0.9.8/src/Makefile.am
@@ -1424,8 +1424,8 @@ module_jack_source_la_CFLAGS = $(AM_CFLA
 # HAL
 libdbus_util_la_SOURCES = modules/dbus-util.c modules/dbus-util.h
 libdbus_util_la_LDFLAGS = -avoid-version
-libdbus_util_la_LIBADD = $(AM_LIBADD) $(HAL_LIBS) libpulsecore.la
-libdbus_util_la_CFLAGS = $(AM_CFLAGS) $(HAL_CFLAGS)
+libdbus_util_la_LIBADD = $(AM_LIBADD) $(DBUS_LIBS) libpulsecore.la
+libdbus_util_la_CFLAGS = $(AM_CFLAGS) $(DBUS_CFLAGS)
 
 module_hal_detect_la_SOURCES = modules/module-hal-detect.c
 module_hal_detect_la_LDFLAGS = -module -avoid-version
