Index: pulseaudio-0.9.7/src/daemon/main.c
===================================================================
--- pulseaudio-0.9.7.orig/src/daemon/main.c
+++ pulseaudio-0.9.7/src/daemon/main.c
@@ -634,7 +634,6 @@ int main(int argc, char *argv[]) {
     c->default_sample_spec = conf->default_sample_spec;
     c->default_n_fragments = conf->default_n_fragments;
     c->default_fragment_size_msec = conf->default_fragment_size_msec;
-    c->disallow_module_loading = conf->disallow_module_loading;
     c->exit_idle_time = conf->exit_idle_time;
     c->module_idle_time = conf->module_idle_time;
     c->scache_idle_time = conf->scache_idle_time;
@@ -675,6 +674,8 @@ int main(int argc, char *argv[]) {
     pa_log_error("%s", s = pa_strbuf_tostring_free(buf));
     pa_xfree(s);
 
+    c->disallow_module_loading = conf->disallow_module_loading;
+
     if (r < 0 && conf->fail) {
         pa_log("failed to initialize daemon.");
 #ifdef HAVE_FORK
