Index: src/pulse/stream.c
===================================================================
--- src/pulse/stream.c	(revision 2073)
+++ src/pulse/stream.c	(revision 2074)
@@ -586,7 +586,7 @@
         }
     }
 
-    if (s->context->version >= 12) {
+    if (s->context->version >= 12 && s->direction != PA_STREAM_UPLOAD) {
         pa_sample_spec ss;
         pa_channel_map cm;
         const char *dn = NULL;
