commit ff5b7fb222db265dfbc5f44dc3d55ce6396d78ca
Author: Diego Elio 'Flameeyes' Pettenò <flameeyes@gmail.com>
Date:   Fri May 15 23:42:43 2009 +0200

    Add missing headers' include to build on FreeBSD 7.1.

diff --git a/src/modules/rtp/rtsp_client.c b/src/modules/rtp/rtsp_client.c
index 629328a..cb037de 100644
--- a/src/modules/rtp/rtsp_client.c
+++ b/src/modules/rtp/rtsp_client.c
@@ -30,6 +30,7 @@
 #include <arpa/inet.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#include <netinet/in.h>
 
 #ifdef HAVE_SYS_FILIO_H
 #include <sys/filio.h>
