diff --git b/media-libs/mesa/files/lib/libGLU.la a/media-libs/mesa/files/lib/libGLU.la
index a87a7ad..e0d188d 100644
--- b/media-libs/mesa/files/lib/libGLU.la
+++ a/media-libs/mesa/files/lib/libGLU.la
@@ -14,7 +14,7 @@ library_names='libGLU.so.1.3 libGLU.so.1 libGLU.so'
 old_library=''
 
 # Libraries that this one depends upon.
-dependency_libs='-lGL -lSM -lICE -lXmu -lXt -lXext -lXi -lX11 -ldl -lpthread -lstdc++'
+dependency_libs=' -L/usr/${libdir} -lGL -lSM -lICE -lXmu -lXt -lXext -lXi -lX11 -ldl -lpthread'
 
 # Version information for libGLU.
 current=4
@@ -29,4 +29,4 @@ dlopen=''
 dlpreopen=''
 
 # Directory that this library needs to be installed in:
-libdir='/usr/lib'
+libdir='/usr/${libdir}'
\ No newline at end of file
