diff --git b/media-libs/mesa/files/lib/libGL.la a/media-libs/mesa/files/lib/libGL.la
index 5c3f25a..646d544 100644
--- b/media-libs/mesa/files/lib/libGL.la
+++ a/media-libs/mesa/files/lib/libGL.la
@@ -14,7 +14,7 @@ library_names='libGL.so.1.2 libGL.so.1 libGL.so'
 old_library=''
 
 # Libraries that this one depends upon.
-dependency_libs='-lSM -lICE -lXmu -lXt -lXext -lXi -lX11 -ldl -lpthread'
+dependency_libs=' -L/usr/${libdir} -lSM -lICE -lXmu -lXt -lXext -lXi -lX11 -ldl -lpthread'
 
 # Version information for libGL.
 current=3
@@ -29,4 +29,4 @@ dlopen=''
 dlpreopen=''
 
 # Directory that this library needs to be installed in:
-libdir='/usr/${libdir}'
+libdir='/usr/${libdir}'
\ No newline at end of file
