--- Pyrex/Distutils/build_ext.py.old	2007-10-16 00:43:41.035218320 +0200
+++ Pyrex/Distutils/build_ext.py	2007-10-16 00:49:33.564437777 +0200
@@ -81,6 +81,11 @@
             ext.sources = self.pyrex_sources(ext.sources, ext)
             self.build_extension(ext)
 
+    # Workaround, as setuptools wants swig_sources with only two arguments
+    # see Gentoo bugs #195354 and #195466
+    def swig_sources(self, sources, extension=None):
+        return _build_ext.build_ext.swig_sources(self, sources, extension)
+
     def pyrex_sources(self, sources, extension):
 
         """
