--- libcdio-0.80/lib/driver/FreeBSD/freebsd.h.old	2008-07-16 08:26:46 +0000
+++ libcdio-0.80/lib/driver/FreeBSD/freebsd.h	2008-07-16 08:26:54 +0000
@@ -158,7 +158,7 @@
 				cdio_drive_write_cap_t *p_write_cap,
 				cdio_drive_misc_cap_t  *p_misc_cap);
 
-int get_media_changed_freebsd (const void *p_user_data);
+static int get_media_changed_freebsd (const void *p_user_data);
 
 char *get_mcn_freebsd_ioctl (const _img_private_t *p_env);
 
