Index: Makefile.am
===================================================================
RCS file: /sources/libcdio/libcdio/src/Makefile.am,v
retrieving revision 1.45
retrieving revision 1.46
diff -u -B -r1.45 -r1.46
--- src/Makefile.am	20 Mar 2008 19:02:38 -0000	1.45
+++ src/Makefile.am	24 Apr 2008 07:28:00 -0000	1.46
@@ -17,11 +17,6 @@
 
 GETOPT_C = getopt.c getopt1.c
 
-man_MANS = cd-drive.1 cd-info.1 cd-read.1 iso-read.1 iso-info.1 
-EXTRA_DIST = cd-drive.help2man cd-info.help2man cd-read.help2man \
-	     iso-info.help2man iso-read.help2man $(GETOPT_C) getopt.h \
-             $(man_MANS)
-
 noinst_HEADERS = cddb.h getopt.h util.h
 
 ####################################################
@@ -35,10 +30,6 @@
 CDDB_LIBS=@CDDB_LIBS@
 CDDA_PLAYER_LIBS=@CDDA_PLAYER_LIBS@
 
-$(man_MANS): %.1: % %.help2man
-	-$(HELP2MAN) --opt-include=$<.help2man --no-info --output=$@ ./$<
-MOSTLYCLEANFILES = $(man_MANS)
-
 if BUILD_CDDA_PLAYER
 cdda_player_SOURCES = cdda-player.c cddb.c cddb.h $(GETOPT_C)
 cdda_player_LDADD   = $(LIBCDIO_LIBS) $(CDDB_LIBS) $(CDDA_PLAYER_LIBS)
@@ -49,12 +40,14 @@
 cd_drive_SOURCES = cd-drive.c util.c util.h $(GETOPT_C)
 cd_drive_LDADD   =  $(LIBISO9660_LIBS) $(LIBCDIO_LIBS) $(LIBICONV)
 bin_cd_drive     = cd-drive
+man_cd_drive         = cd-drive.1
 endif
 
 if BUILD_CDINFO
 cd_info_SOURCES = cd-info.c cddb.c cddb.h util.c util.h $(GETOPT_C)
 cd_info_LDADD   =  $(LIBISO9660_LIBS) $(LIBCDIO_LIBS) $(CDDB_LIBS) $(VCDINFO_LIBS) $(LIBICONV)
 bin_cd_info     = cd-info 
+man_cd_info     = cd-info.1
 endif
 
 if BUILD_CDINFO_LINUX
@@ -67,18 +60,21 @@
 cd_read_SOURCES = cd-read.c util.c util.h $(GETOPT_C)
 cd_read_LDADD   = $(LIBISO9660_LIBS) $(LIBCDIO_LIBS) $(LIBICONV)
 bin_cd_read     = cd-read 
+man_cd_read     = cd-read.1
 endif
 
 if BUILD_ISO_INFO
 iso_info_SOURCES = iso-info.c util.c util.h $(GETOPT_C)
 iso_info_LDADD   = $(LIBISO9660_LIBS) $(LIBCDIO_LIBS) $(LIBICONV)
 bin_iso_info     = iso-info
+man_iso_info     = iso-info.1
 endif
 
 if BUILD_ISO_READ
 iso_read_SOURCES = iso-read.c util.c util.h $(GETOPT_C)
 iso_read_LDADD   = $(LIBISO9660_LIBS) $(LIBCDIO_LIBS) $(LIBICONV)
 bin_iso_read     = iso-read
+man_iso_read     = iso-read.1
 endif
 
 mmc_tool_SOURCES = mmc-tool.c util.c util.h $(GETOPT_C)
@@ -88,3 +84,12 @@
 bin_PROGRAMS = $(bin_cd_drive) $(bin_cd_info)  $(bin_cdinfo_linux) $(bin_cd_read) $(bin_iso_info) $(bin_iso_read) $(bin_cdda_player) $(bin_mmc_tool)
 
 INCLUDES = -I$(top_srcdir) $(LIBCDIO_CFLAGS) $(VCDINFO_CFLAGS) $(CDDB_CFLAGS)
+
+man_MANS = $(man_cd_drive) $(man_cd_info) $(man_cd_read) $(man_iso_read) $(man_iso_info)
+EXTRA_DIST = cd-drive.help2man cd-info.help2man cd-read.help2man \
+             iso-info.help2man iso-read.help2man $(GETOPT_C) getopt.h \
+             $(man_MANS)
+
+$(man_MANS): %.1: % %.help2man
+        -$(HELP2MAN) --opt-include=$<.help2man --no-info --output=$@ ./$<
+MOSTLYCLEANFILES = $(man_MANS)
