--- camel/providers/imap/camel-imap-store.c.orig	2009-07-26 13:18:22.000000000 +0100
+++ camel/providers/imap/camel-imap-store.c	2009-08-06 17:23:22.815045826 +0100
@@ -959,7 +959,7 @@
 		port = ssl_options[i].port;
 	} else {
 		mode = MODE_CLEAR;
-		serv = (gchar *) "imap";
+		serv = (gchar *) "imap2";
 		port = IMAP_PORT;
 	}
 
diff -u camel.orig/providers/pop3/camel-pop3-store.c camel/providers/pop3/camel-pop3-store.c
--- camel.orig/providers/pop3/camel-pop3-store.c	2005-03-03 10:30:32.779550168 -0600
+++ camel/providers/pop3/camel-pop3-store.c	2005-03-03 10:34:40.812843376 -0600
@@ -308,7 +308,7 @@
 		port = ssl_options[i].port;
 	} else {
 		mode = MODE_CLEAR;
-		serv = (gchar *) "pop3";
+		serv = (gchar *) "pop-3";
 		port = POP3S_PORT;
 	}
 	
