--- openal-0.0.8/src/alc/alc_context.c.ori	2006-09-29 12:36:33.000000000 -0400
+++ openal-0.0.8/src/alc/alc_context.c	2006-09-29 12:38:05.000000000 -0400
@@ -236,8 +236,12 @@
 		/* someone unpaused us */
 		ispaused = AL_FALSE;
 
-		_alcDeviceResume( cc->write_device );
-		_alcDeviceResume( cc->read_device );
+		if (cc->write_device) {
+			_alcDeviceResume( cc->write_device );
+		}
+		if (cc->read_device) {
+			_alcDeviceResume( cc->read_device );
+		}
 
 		_alcUnlockAllContexts();
 		_alUnlockMixerPause();
