diff -urN openal-0.0.8.orig/src/alc/alc_context.h openal-0.0.8/src/alc/alc_context.h
--- openal-0.0.8.orig/src/alc/alc_context.h	2005-11-18 05:31:47.000000000 -0800
+++ openal-0.0.8/src/alc/alc_context.h	2007-09-20 13:16:22.000000000 -0700
@@ -18,7 +18,7 @@
 /*
  * _ALC_DEF_FREQ is the default internal mixing frequency.
  */
-#define _ALC_DEF_FREQ        44100
+#define _ALC_DEF_FREQ        48000
 
 /*
  * _ALC_DEF_BUFSIZ is the default length of chunks mixed and written to the
