diff --git a/Makefile.am b/Makefile.am
index 221e91f..f2a6198 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . libudev/docs
+SUBDIRS = .
 
 udevconfdir = $(sysconfdir)/udev
 dist_udevconf_DATA = \
@@ -447,8 +447,7 @@ libgudev_include_HEADERS = \
 
 pkgconfig_DATA += extras/gudev/gudev-1.0.pc
 
-SUBDIRS += \
-	extras/gudev/docs
+SUBDIRS +=
 endif
 
 if ENABLE_INTROSPECTION
diff --git a/configure.ac b/configure.ac
index 727cba6..bbd3775 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8,7 +8,6 @@ AC_SYS_LARGEFILE
 AC_CONFIG_MACRO_DIR([m4])
 LT_INIT
 AC_PROG_AWK
-GTK_DOC_CHECK(1.10)
 AC_PREFIX_DEFAULT([/usr])
 AC_PATH_PROG([XSLTPROC], [xsltproc])
 
@@ -102,12 +101,8 @@ AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_FILES([
 	Makefile
 	libudev/libudev.pc
-	libudev/docs/Makefile
-	libudev/docs/version.xml
 	udev/udev.pc
 	extras/gudev/gudev-1.0.pc
-	extras/gudev/docs/Makefile
-	extras/gudev/docs/version.xml
 ])
 
 AC_OUTPUT
