--- /usr/portage/media-sound/jack-audio-connection-kit/jack-audio-connection-kit-0.109.2-r1.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./media-sound/jack-audio-connection-kit/jack-audio-connection-kit-0.109.2-r1.ebuild	2010-04-04 16:28:16.000000000 +0200
@@ -40,10 +40,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}/${PN}-transport.patch"
 	epatch "${FILESDIR}/${PN}-0.103.0-riceitdown.patch"
 	epatch "${FILESDIR}/${PN}-0.103.0-ppc64fix.patch"
@@ -52,7 +49,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	local myconf=""
 
 	if [[ ${CHOST} == *-darwin* ]] ; then
@@ -95,6 +92,9 @@
 		--with-html-dir=/usr/share/doc/${PF} \
 		--disable-dependency-tracking \
 		${myconf} || die "configure failed"
+}
+
+src_compile() {
 	emake || die "compilation failed"
 
 	if use caps && kernel_is 2 4 ; then
