--- /usr/portage/www-client/mozilla-firefox/mozilla-firefox-3.6.8.ebuild	2010-06-19 21:07:05.000000000 +0200
+++ ./www-client/mozilla-firefox/mozilla-firefox-3.6.8.ebuild	2010-06-20 23:43:23.000000000 +0200
@@ -116,11 +116,13 @@
 src_unpack() {
 	unpack firefox-${MY_PV}.source.tar.bz2 ${PATCH}.tar.bz2
 
-	linguas
-	for X in ${linguas}; do
-		# FIXME: Add support for unpacking xpis to portage
-		[[ ${X} != "en" ]] && xpi_unpack "${P}-${X}.xpi"
-	done
+	if is_final_abi; then
+		linguas
+		for X in ${linguas}; do
+			# FIXME: Add support for unpacking xpis to portage
+			[[ ${X} != "en" ]] && xpi_unpack "${P}-${X}.xpi"
+		done
+	fi
 }
 
 src_prepare() {
