--- /usr/portage/www-client/mozilla-firefox/mozilla-firefox-3.5.8.ebuild	2010-03-02 21:40:35.000000000 +0100
+++ ./www-client/mozilla-firefox/mozilla-firefox-3.5.8.ebuild	2010-03-14 04:25:28.000000000 +0100
@@ -113,13 +113,15 @@
 src_unpack() {
 	unpack ${A}
 
-	linguas
-	for X in ${linguas}; do
-		# FIXME: Add support for unpacking xpis to portage
-		[[ ${X} != "en" ]] && xpi_unpack "${P}-${X}.xpi"
-	done
-	if [[ ${linguas} != "" && ${linguas} != "en" ]]; then
-		einfo "Selected language packs (first will be default): ${linguas}"
+	if is_final_abi; then
+		linguas
+		for X in ${linguas}; do
+			# FIXME: Add support for unpacking xpis to portage
+			[[ ${X} != "en" ]] && xpi_unpack "${P}-${X}.xpi"
+		done
+		if [[ ${linguas} != "" && ${linguas} != "en" ]]; then
+			einfo "Selected language packs (first will be default): ${linguas}"
+		fi
 	fi
 }
 
