--- /usr/portage/net-misc/curl/curl-7.20.0.ebuild	2010-03-11 19:01:15.000000000 +0100
+++ ./net-misc/curl/curl-7.20.0.ebuild	2010-04-03 22:56:15.000000000 +0200
@@ -37,15 +37,12 @@
 	)"
 # used - but can do without in self test: net-misc/stunnel
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-strip-ldflags.patch
 	epatch "${FILESDIR}"/${PN}-7.19.7-test241.patch
 }
 
-src_compile() {
-
+src_configure() {
 	myconf="$(use_enable ldap)
 		$(use_enable ldap ldaps)
 		$(use_with idn libidn)
@@ -85,8 +82,6 @@
 	fi
 
 	econf ${myconf} || die 'configure failed'
-
-	emake || die "install failed for current version"
 }
 
 src_install() {
@@ -100,4 +95,6 @@
 	dodoc CHANGES README
 	dodoc docs/FEATURES docs/INTERNALS
 	dodoc docs/MANUAL docs/FAQ docs/BUGS docs/CONTRIBUTE
+
+	prep_ml_includes /usr/include/curl
 }
