--- /usr/portage/media-libs/libsamplerate/libsamplerate-0.1.4.ebuild	2009-04-12 10:07:53.000000000 +0200
+++ ./media-libs/libsamplerate/libsamplerate-0.1.4.ebuild	2010-05-02 17:54:28.000000000 +0200
@@ -19,9 +19,7 @@
 DEPEND="${RDEPEND}
 	>=dev-util/pkgconfig-0.14"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.1.3-dontbuild-tests-examples.patch \
 		"${FILESDIR}"/${PN}-0.1.3-pkg_prog_pkg_config.patch
 
@@ -32,13 +30,12 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		--disable-fftw \
 		$(use_enable sndfile) \
 		--disable-dependency-tracking \
 		|| die "econf failed"
-	emake || die "emake failed"
 }
 
 src_install() {
