--- /usr/portage/media-libs/jasper/jasper-1.900.1-r3.ebuild	2010-09-16 19:36:01.000000000 +0200
+++ ./media-libs/jasper/jasper-1.900.1-r3.ebuild	2011-01-22 14:16:20.000000000 +0100
@@ -19,20 +19,22 @@
 DEPEND="${RDEPEND}
 		app-arch/unzip"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${WORKDIR}"/${P}-fixes-20081208.patch
 
-	elibtoolize
+	# when we run eautoreconf instead of elibtoolize we avoid problems with finding libs
+	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		$(use_enable jpeg libjpeg) \
 		$(use_enable opengl) \
 		--enable-shared \
 		|| die
+}
+
+src_compile() {
 	emake || die "If you got undefined references to OpenGL related libraries,please try 'eselect opengl set xorg-x11' before emerging. See bug #133609."
 }
 
