--- murrine-0.53.1.orig/src/support.c
+++ murrine-0.53.1/src/support.c
@@ -274,7 +274,7 @@
 	if (tmp_size)
 	{
 		*indicator_size = *tmp_size;
-		g_free (tmp_size);
+		gtk_requisition_free (tmp_size);
 	}
 	else
 		*indicator_size = default_option_indicator_size;
@@ -282,7 +282,7 @@
 	if (tmp_spacing)
 	{
 		*indicator_spacing = *tmp_spacing;
-		g_free (tmp_spacing);
+		gtk_border_free (tmp_spacing);
 	}
 	else
 		*indicator_spacing = default_option_indicator_spacing;
