--- source4/build/m4/public.m4.org	2009-09-30 14:17:40.000000000 +0200
+++ source4/build/m4/public.m4	2009-10-07 08:42:10.000000000 +0200
@@ -86,10 +86,10 @@
 		if $PKG_CONFIG --atleast-pkgconfig-version $SAMBA_PKG_CONFIG_MIN_VERSION; then
 			AC_MSG_CHECKING(for $2)
 
-			if $PKG_CONFIG --exists '$2' ; then
+			if $PKG_CONFIG --exists "$2" ; then
 				AC_MSG_RESULT(yes)
 
-				$1_CFLAGS="`$PKG_CONFIG --cflags '$2'`"
+				$1_CFLAGS="`$PKG_CONFIG --cflags \"$2\"`"
 				OLD_CFLAGS="$CFLAGS"
 				CFLAGS="$CFLAGS $$1_CFLAGS"
 				AC_MSG_CHECKING([that the C compiler can use the $1_CFLAGS])
@@ -101,15 +101,15 @@
 				CFLAGS="$OLD_CFLAGS"
 
 				SMB_EXT_LIB($1, 
-					[`$PKG_CONFIG --libs-only-l '$2'`], 
-					[`$PKG_CONFIG --cflags-only-other '$2'`],
-					[`$PKG_CONFIG --cflags-only-I '$2'`],
-					[`$PKG_CONFIG --libs-only-other '$2'` `$PKG_CONFIG --libs-only-L '$2'`])
+					[`$PKG_CONFIG --libs-only-l "$2"`], 
+					[`$PKG_CONFIG --cflags-only-other "$2"`],
+					[`$PKG_CONFIG --cflags-only-I "$2"`],
+					[`$PKG_CONFIG --libs-only-other "$2"` `$PKG_CONFIG --libs-only-L "$2"`])
 				ac_cv_$1_found=yes
 
 			else
 				AC_MSG_RESULT(no)
-				$PKG_CONFIG --errors-to-stdout --print-errors '$2'
+				$PKG_CONFIG --errors-to-stdout --print-errors "$2"
 				ac_cv_$1_found=no
 			fi
 		else
