--- configure.ac.orig	2008-09-25 20:34:22.000000000 -0700
+++ configure.ac	2008-09-25 20:43:57.000000000 -0700
@@ -191,10 +191,8 @@
 CFLAGS="$CFLAGS -msse2 $SSE2_CFLAGS"
 
 AC_COMPILE_IFELSE([
-#if defined(__GNUC__) && (__GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 2))
-#   if !defined(__amd64__) && !defined(__x86_64__)
+#if !defined(__GNUC__) && !(__GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 2))
 #      error "Need GCC >= 4.2 for SSE2 intrinsics on x86"
-#   endif
 #endif
 #include <mmintrin.h>
 #include <xmmintrin.h>
