--- configure.ac.orig	2008-07-18 13:15:45.000000000 +0200
+++ configure.ac	2008-07-18 13:15:50.000000000 +0200
@@ -187,9 +187,6 @@
 #if defined(__GNUC__) && (__GNUC__ < 3 || (__GNUC__ == 3 && __GNUC_MINOR__ < 4))
 #error "Need GCC >= 3.4 for SSE intrinsics"
 #endif
-#if !defined(__amd64__) && !defined(__x86_64__)
-#error "Need x86-64 for SSE"
-#endif
 #include <mmintrin.h>
 #include <xmmintrin.h>
 int main () {
