http://bugs.gentoo.org/332857

--- libproxy/modules/config_gnome.cpp
+++ libproxy/modules/config_gnome.cpp
@@ -19,6 +19,7 @@
 
 #include <cstdio>         // For fileno(), fread(), pclose(), popen(), sscanf()
 #include <sys/select.h>   // For select()
+#include <sys/stat.h>     // for stat()
 #include <fcntl.h>        // For fcntl()
 #include <errno.h>        // For errno stuff
 #include <unistd.h>       // For pipe(), close(), vfork(), dup(), execl(), _exit()
