--- fribidi-0.10.4/configure.orig	Sat Feb 12 20:27:25 2005
+++ fribidi-0.10.4/configure	Sat Feb 12 20:28:08 2005
@@ -3539,7 +3539,7 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag=''
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
