--- /usr/portage/gnome-base/orbit/orbit-2.14.16.ebuild	2009-03-05 23:35:41.000000000 +0100
+++ ././gnome-base/orbit/orbit-2.14.16.ebuild	2010-05-05 23:44:47.000000000 +0200
@@ -26,8 +26,8 @@
 
 DOCS="AUTHORS ChangeLog HACKING MAINTAINERS NEWS README* TODO"
 
-src_unpack() {
-	gnome2_src_unpack
+src_prepare() {
+	gnome2_src_prepare
 
 	# Filter out G_DISABLE_DEPRECATED to be future-proof, related to bug 213434
 	sed -i -e '/DISABLE_DEPRECATED/d' \
