--- /usr/portage/net-dns/libidn/libidn-1.16.ebuild	2009-10-31 15:36:20.000000000 +0100
+++ ./net-dns/libidn/libidn-1.16.ebuild	2010-04-04 02:35:24.000000000 +0200
@@ -30,7 +30,7 @@
 	rm "${S}/java/${P}.jar" || die
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		$(use_enable nls) \
 		$(use_enable java) \
@@ -40,7 +40,9 @@
 		--with-packager-version="r${PR}" \
 		--with-packager-bug-reports="https://bugs.gentoo.org" \
 		|| die
+}
 
+src_compile() {
 	emake || die
 
 	if use emacs; then
@@ -69,7 +71,7 @@
 		rm -rf "${D}"/usr/share/java || die
 
 		if use doc ; then
-			java-pkg_dojavadoc doc/java
+			[[ -d "${D}"/usr/share/doc/"${P}"/html ]] || java-pkg_dojavadoc doc/java
 		fi
 	fi
 }
