--- /usr/portage/sys-apps/acl/acl-2.2.47.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./sys-apps/acl/acl-2.2.47.ebuild	2010-04-02 18:42:03.000000000 +0200
@@ -21,9 +21,7 @@
 DEPEND="${RDEPEND}
 	nls? ( sys-devel/gettext )"
 
-src_unpack() {
-	unpack ${MY_P}.tar.gz
-	cd "${S}"
+src_prepare() {
 	if use nfs ; then
 		cp "${DISTDIR}"/acl-2.2.42-CITI_NFS4_ALL-2.dif . || die
 		sed -i '/^diff --git a.debian.changelog b.debian.changelog/,/^diff --git/d' acl-2.2.42-CITI_NFS4_ALL-2.dif || die
@@ -44,7 +42,7 @@
 	strip-linguas po
 }
 
-src_compile() {
+src_configure() {
 	unset PLATFORM #184564
 	export OPTIMIZER=${CFLAGS}
 	export DEBUG=-DNDEBUG
@@ -54,7 +52,6 @@
 		--libexecdir=/usr/$(get_libdir) \
 		--bindir=/bin \
 		|| die
-	emake || die
 }
 
 src_install() {
