--- /usr/portage/dev-libs/nss/nss-3.12.5-r1.ebuild	2010-02-14 15:35:33.000000000 +0100
+++ ./dev-libs/nss/nss-3.12.5-r1.ebuild	2010-04-22 22:01:46.000000000 +0200
@@ -21,9 +21,7 @@
 RDEPEND=">=dev-libs/nspr-${NSPR_VER}
 	>=dev-db/sqlite-3.5"
 
-src_unpack() {
-	unpack ${A}
-
+src_prepare() {
 	# Custom changes for gentoo
 	epatch "${FILESDIR}"/"${PN}"-3.12.5-gentoo-fixups.diff
 
@@ -40,11 +38,11 @@
 	sed -i -e "s:gentoo\/nss:$(get_libdir):" "${S}"/mozilla/security/nss/config/Makefile || die "Failed to fix for multilib"
 }
 
-src_compile() {
+src_configure() {
 	strip-flags
 
 	echo > "${T}"/test.c
-	$(tc-getCC) -c "${T}"/test.c -o "${T}"/test.o
+	$(tc-getCC) ${CFLAGS} -c "${T}"/test.c -o "${T}"/test.o
 	case $(file "${T}"/test.o) in
 	*64-bit*) export USE_64=1;;
 	*32-bit*) ;;
@@ -61,7 +59,9 @@
 	export NSS_ENABLE_ECC=1
 	export XCFLAGS="${CFLAGS}"
 	export FREEBL_NO_DEPEND=1
+}
 
+src_compile() {
 	cd "${S}"/mozilla/security/coreconf
 	emake -j1 CC="$(tc-getCC)" || die "coreconf make failed"
 	cd "${S}"/mozilla/security/dbm
