--- /usr/portage/dev-libs/nss/nss-3.12.6-r1.ebuild	2010-04-20 11:35:38.000000000 +0200
+++ ./dev-libs/nss/nss-3.12.6-r1.ebuild	2010-04-22 22:33:57.000000000 +0200
@@ -19,11 +19,7 @@
 RDEPEND=">=dev-libs/nspr-${NSPR_VER}
 	>=dev-db/sqlite-3.5"
 
-src_unpack() {
-	unpack ${A}
-
-	cd "${S}"
-
+src_prepare() {
 	# Custom changes for gentoo
 	epatch "${FILESDIR}/${PN}-3.12.5-gentoo-fixups.diff"
 	epatch "${FILESDIR}/${PN}-3.12.6-gentoo-fixup-warnings.patch"
@@ -41,7 +37,7 @@
 	sed -i -e "s:gentoo\/nss:$(get_libdir):" "${S}"/mozilla/security/nss/config/Makefile || die "Failed to fix for multilib"
 }
 
-src_compile() {
+src_configure() {
 	strip-flags
 
 	echo > "${T}"/test.c
@@ -60,7 +56,9 @@
 	export NSS_ENABLE_ECC=1
 	export XCFLAGS="${CFLAGS}"
 	export FREEBL_NO_DEPEND=1
+}
 
+src_compile() {
 	cd "${S}"/mozilla/security/coreconf
 	emake -j1 CC="$(tc-getCC)" || die "coreconf make failed"
 	cd "${S}"/mozilla/security/dbm
