--- /usr/portage/sys-devel/gettext/gettext-0.17.ebuild	2010-03-13 04:12:23.000000000 +0100
+++ ./sys-devel/gettext/gettext-0.17.ebuild	2010-03-14 01:17:18.000000000 +0100
@@ -11,7 +11,7 @@
 LICENSE="GPL-3 LGPL-2"
 SLOT="0"
 KEYWORDS="alpha amd64 arm hppa ia64 m68k ~mips ppc ppc64 s390 sh sparc ~sparc-fbsd x86 ~x86-fbsd"
-IUSE="acl doc emacs nls nocxx openmp"
+IUSE="acl doc emacs nls nocxx openmp java"
 
 DEPEND="virtual/libiconv
 	dev-libs/libxml2
@@ -19,11 +19,10 @@
 	dev-libs/expat
 	acl? ( virtual/acl )"
 PDEPEND="emacs? ( app-emacs/po-mode )"
+RDEPEND="${DEPEND}
+	java? ( >=virtual/jdk-1.4 )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epunt_cxx
 
 	epatch "${FILESDIR}"/${PN}-0.14.1-lib-path-tests.patch #81628
@@ -49,7 +48,7 @@
 	fi
 }
 
-src_compile() {
+src_configure() {
 	local myconf=""
 	# Build with --without-included-gettext (on glibc systems)
 	if use elibc_glibc ; then
@@ -65,13 +64,12 @@
 	econf \
 		--docdir="/usr/share/doc/${PF}" \
 		--without-emacs \
-		--disable-java \
+		$(use_enable java) \
 		--with-included-glib \
 		--with-included-libcroco \
 		$(use_enable openmp) \
 		${myconf} \
 		|| die
-	emake || die
 }
 
 src_install() {
@@ -95,6 +93,15 @@
 		gen_usr_ldscript ${libname}
 	fi
 
+	if use java; then
+		java-pkg_newjar "${S}"/gettext-runtime/intl-java/libintl.jar || die
+		
+		if use doc; then
+			rm -rf "${D}"/usr/share/doc/${PF}/{html,javadoc2,javadoc1}
+			java-pkg_dojavadoc gettext-runtime/intl-java/javadoc*
+		fi
+	fi
+
 	if use doc ; then
 		dohtml "${D}"/usr/share/doc/${PF}/*.html
 	else
