--- /usr/portage/sys-devel/gettext/gettext-0.17-r1.ebuild	2010-03-31 01:36:59.000000000 +0200
+++ ./sys-devel/gettext/gettext-0.17-r1.ebuild	2010-07-28 14:10:01.000000000 +0200
@@ -20,10 +20,7 @@
 	acl? ( virtual/acl )"
 PDEPEND="emacs? ( app-emacs/po-mode )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epunt_cxx
 
 	epatch "${FILESDIR}"/${PN}-0.14.1-lib-path-tests.patch #81628
@@ -50,7 +47,7 @@
 	fi
 }
 
-src_compile() {
+src_configure() {
 	local myconf=""
 	# Build with --without-included-gettext (on glibc systems)
 	if use elibc_glibc ; then
@@ -72,7 +69,6 @@
 		$(use_enable openmp) \
 		${myconf} \
 		|| die
-	emake || die
 }
 
 src_install() {
