Index: libffi-3.0.8/src/x86/ffi.c
===================================================================
--- libffi-3.0.8.orig/src/x86/ffi.c
+++ libffi-3.0.8/src/x86/ffi.c
@@ -124,7 +124,7 @@ ffi_status ffi_prep_cif_machdep(ffi_cif 
 #ifdef X86
     case FFI_TYPE_STRUCT:
 #endif
-#if defined(X86) || defined(X86_DARWIN)
+#if defined(X86) || defined(X86_DARWIN) || defined(X86_FREEBSD)
     case FFI_TYPE_UINT8:
     case FFI_TYPE_UINT16:
     case FFI_TYPE_SINT8:
Index: libffi-3.0.8/configure.ac
===================================================================
--- libffi-3.0.8.orig/configure.ac
+++ libffi-3.0.8/configure.ac
@@ -71,7 +71,7 @@ case "$host" in
 	TARGET=PA_HPUX; TARGETDIR=pa
 	;;
 
-  i386-*-freebsd* | i386-*-openbsd*)
+  i?86-*-freebsd* | i?86-*-openbsd*)
 	TARGET=X86_FREEBSD; TARGETDIR=x86
 	;;
   i?86-win32* | i?86-*-cygwin* | i?86-*-mingw*)
