diff -urNp pygtk-2.16.0-orig/gtk/gdkcolor.override pygtk-2.16.0/gtk/gdkcolor.override
--- pygtk-2.16.0-orig/gtk/gdkcolor.override	2009-09-07 11:24:02.000000000 +0530
+++ pygtk-2.16.0/gtk/gdkcolor.override	2009-09-07 11:24:11.000000000 +0530
@@ -139,7 +139,7 @@ _wrap_gdk_color_new(PyGBoxed *self,
 }
 
 %%
-define color_from_hsv
+define color_from_hsv kwargs
 static PyObject *
 _wrap_color_from_hsv (PyObject *ignored, PyObject *args, PyObject*kwargs)
 {
