diff -ur schroedinger-1.0.7.orig/gst/gstschrodec.c schroedinger-1.0.7/gst/gstschrodec.c
--- schroedinger-1.0.7.orig/gst/gstschrodec.c	2009-04-23 04:32:31.000000000 +0300
+++ schroedinger-1.0.7/gst/gstschrodec.c	2009-08-16 15:46:45.000000000 +0300
@@ -547,7 +547,7 @@
 
   n_available -= 3;
 
-  return gst_adapter_masked_scan_uint32 (adapter, 0x42424344, 0xffffffff,
+  return schro_gst_adapter_masked_scan_uint32 (adapter, 0x42424344, 0xffffffff,
       offset, MIN (n, n_available - 3));
 }
 
diff -ur schroedinger-1.0.7.orig/gst/gstschroparse.c schroedinger-1.0.7/gst/gstschroparse.c
--- schroedinger-1.0.7.orig/gst/gstschroparse.c	2009-04-22 21:33:38.000000000 +0300
+++ schroedinger-1.0.7/gst/gstschroparse.c	2009-08-16 15:46:45.000000000 +0300
@@ -305,7 +305,7 @@
 
   n_available -= 3;
 
-  return gst_adapter_masked_scan_uint32 (adapter, 0x42424344, 0xffffffff,
+  return schro_gst_adapter_masked_scan_uint32 (adapter, 0x42424344, 0xffffffff,
       offset, MIN (n, n_available - 3));
 }
 
diff -ur schroedinger-1.0.7.orig/gst-libs/gst/video/gstbasevideoutils.c schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoutils.c
--- schroedinger-1.0.7.orig/gst-libs/gst/video/gstbasevideoutils.c	2009-04-22 20:55:58.000000000 +0300
+++ schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoutils.c	2009-08-16 15:46:45.000000000 +0300
@@ -231,7 +231,7 @@
 
 
 int
-gst_adapter_masked_scan_uint32 (GstAdapter *adapter,
+schro_gst_adapter_masked_scan_uint32 (GstAdapter *adapter,
     guint32 pattern, guint32 mask, int offset, int n)
 {
   GSList *g;
diff -ur schroedinger-1.0.7.orig/gst-libs/gst/video/gstbasevideoutils.h schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoutils.h
--- schroedinger-1.0.7.orig/gst-libs/gst/video/gstbasevideoutils.h	2009-04-22 20:56:06.000000000 +0300
+++ schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoutils.h	2009-08-16 15:46:45.000000000 +0300
@@ -84,7 +84,7 @@
 GstClockTime gst_video_state_get_timestamp (const GstVideoState *state,
     int frame_number);
 
-int gst_adapter_masked_scan_uint32 (GstAdapter *adapter,
+int schro_gst_adapter_masked_scan_uint32 (GstAdapter *adapter,
     guint32 pattern, guint32 mask, int offset, int n);
 GstBuffer *gst_adapter_get_buffer (GstAdapter *adapter);
 
