diff -rdu lua-5.1.3.orig/src/lapi.c lua-5.1.3/src/lapi.c
--- lua-5.1.3.orig/src/lapi.c        2008-02-12 16:17:59.000000000 +0000
+++ lua-5.1.3/src/lapi.c     2008-02-12 16:26:32.000000000 +0000
@@ -93,15 +93,14 @@


 LUA_API int lua_checkstack (lua_State *L, int size) {
-  int res;
+  int res = 1;
   lua_lock(L);
-  if ((L->top - L->base + size) > LUAI_MAXCSTACK)
+  if (size > LUAI_MAXCSTACK || (L->top - L->base + size) > LUAI_MAXCSTACK)
     res = 0;  /* stack overflow */
-  else {
+  else if (size > 0) {
     luaD_checkstack(L, size);
     if (L->ci->top < L->top + size)
       L->ci->top = L->top + size;
-    res = 1;
   }
   lua_unlock(L);
   return res;
