--- /usr/portage/net-fs/samba/samba-3.2.15-r1.ebuild	2010-08-17 18:12:11.000000000 +0200
+++ ./net-fs/samba/samba-3.2.15-r1.ebuild	2010-08-26 00:55:37.000000000 +0200
@@ -52,8 +52,7 @@
 	confutils_use_depend_all cifsupcall ads
 }
 
-src_unpack() {
-	unpack ${A}
+src_prepare() {
 	cd "${S}/source"
 
 	epatch \
@@ -75,7 +74,7 @@
 		"${S}/source/script/tests/selftest.sh" || die "sed failed"
 }
 
-src_compile() {
+src_configure() {
 	cd "${S}/source"
 
 	local myconf
@@ -137,10 +136,13 @@
 		$(use_with syslog) \
 		$(use_with winbind) \
 		${myconf} ${mylangs} ${mymod_shared} || die "econf failed"
+}
+
+src_compile() {
+	cd "${S}/source"
 
 	emake -j1 proto || die "emake proto failed"
 	emake -j1 everything || die "emake everything failed"
-
 }
 
 src_test() {
