--- /usr/portage/net-fs/samba/samba-3.0.37-r1.ebuild	2010-08-17 18:12:11.000000000 +0200
+++ ./net-fs/samba/samba-3.0.37-r1.ebuild	2010-08-26 00:55:29.000000000 +0200
@@ -47,8 +47,7 @@
 	confutils_use_depend_all ads ldap
 }
 
-src_unpack() {
-	unpack ${A}
+src_prepare() {
 	cd "${S}/source"
 
 	# lazyldflags.patch: adds "-Wl,-z,now" to smb{mnt,umount}
@@ -75,7 +74,7 @@
 
 }
 
-src_compile() {
+src_configure() {
 	cd "${S}/source"
 
 	local myconf
@@ -134,13 +133,6 @@
 		$(use_with winbind) \
 		${myconf} ${mylangs} ${mymod_shared}
 
-	emake -j1 proto || die "emake proto failed"
-	emake -j1 everything || die "emake everything failed"
-
-	if use python ; then
-		emake -j1 python_ext || die "emake python_ext failed"
-	fi
-
 	if use oav ; then
 		# maintainer-info:
 		# - there are no known releases of mks or kavdc,
@@ -155,6 +147,21 @@
 			--with-filetype \
 			--with-fileregexp \
 			$(use_enable debug)
+	fi
+}
+
+src_compile() {
+	cd "${S}/source"
+
+	emake -j1 proto || die "emake proto failed"
+	emake -j1 everything || die "emake everything failed"
+
+	if use python ; then
+		emake -j1 python_ext || die "emake python_ext failed"
+	fi
+
+	if use oav ; then
+		cd "${WORKDIR}/${VSCAN_P}"
 		emake -j1 || die "emake oav plugins failed"
 	fi
 }
