From 9f0e9e35ee33000c5f83a72d2c949c3d4f58885c Mon Sep 17 00:00:00 2001
From: Matthew Barnes <mbarnes@redhat.com>
Date: Mon, 21 Jun 2010 23:23:14 +0000
Subject: Bug 622329 - Help menu points to the wrong FAQ page

---
diff --git a/shell/e-shell-window-actions.c b/shell/e-shell-window-actions.c
index e6c0b50..c653560 100644
--- a/shell/e-shell-window-actions.c
+++ b/shell/e-shell-window-actions.c
@@ -34,7 +34,7 @@
 	"Copyright \xC2\xA9 1999 - 2008 Novell, Inc. and Others"
 
 #define EVOLUTION_FAQ \
-	"http://www.go-evolution.org/FAQ"
+	"http://live.gnome.org/Evolution/FAQ"
 
 #define EVOLUTION_WEBSITE \
 	"http://www.gnome.org/projects/evolution/"
--
cgit v0.8.3.1
