From 7f9a025cb6ca92ab74ea4493cdcdc232a5dfca76 Mon Sep 17 00:00:00 2001
From: Matthew Barnes <mbarnes@redhat.com>
Date: Mon, 21 Jun 2010 14:58:38 +0000
Subject: Bug 619347 - Contact List Editor calls wrong EDestination function

---
diff --git a/addressbook/gui/contact-list-editor/e-contact-list-editor.c b/addressbook/gui/contact-list-editor/e-contact-list-editor.c
index 23da590..5d96047 100644
--- a/addressbook/gui/contact-list-editor/e-contact-list-editor.c
+++ b/addressbook/gui/contact-list-editor/e-contact-list-editor.c
@@ -640,7 +640,7 @@ contact_list_editor_email_entry_updated_cb (GtkWidget *widget,
 	entry = E_NAME_SELECTOR_ENTRY (widget);
 	model = E_CONTACT_LIST_MODEL (editor->priv->model);
 
-	email = g_strdup (e_destination_get_address (destination));
+	email = g_strdup (e_destination_get_textrep (destination, TRUE));
 	store = e_name_selector_entry_peek_destination_store (entry);
 	e_destination_store_remove_destination (store, destination);
 	gtk_entry_set_text (GTK_ENTRY (WIDGET (EMAIL_ENTRY)), "");
--
cgit v0.8.3.1
