From ff2553440bd1f7d76667e71479d50efa1a537e96 Mon Sep 17 00:00:00 2001
From: Matthew Barnes <mbarnes@redhat.com>
Date: Mon, 29 Mar 2010 11:22:23 -0400
Subject: [PATCH 05/14] Bug 614049 - Attachment bar causes drawing issues in RTL locales

---
 widgets/misc/e-attachment-paned.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/widgets/misc/e-attachment-paned.c b/widgets/misc/e-attachment-paned.c
index 8e25098..aefd9fa 100644
--- a/widgets/misc/e-attachment-paned.c
+++ b/widgets/misc/e-attachment-paned.c
@@ -614,10 +614,15 @@ attachment_paned_init (EAttachmentPaned *paned)
 
 	/* Request the width to be as large as possible, and let the
 	 * GtkExpander allocate what space there is.  This effectively
-	 * packs the widget to expand. */
+	 * packs the widget to expand.
+	 *
+	 * XXX This hack causes nasty side-effects in RTL locales,
+	 *     so check the reading direction before applying it.
+	 *     See GNOME bug #614049 for details + screenshot. */
 	widget = gtk_hbox_new (FALSE, 6);
 	gtk_size_group_add_widget (size_group, widget);
-	gtk_widget_set_size_request (widget, G_MAXINT, -1);
+	if (gtk_widget_get_direction (widget) != GTK_TEXT_DIR_RTL)
+		gtk_widget_set_size_request (widget, G_MAXINT, -1);
 	gtk_expander_set_label_widget (GTK_EXPANDER (container), widget);
 	gtk_widget_show (widget);
 
-- 
1.7.1

