diff -ur popt-1.15.orig/testit.sh popt-1.15/testit.sh
--- popt-1.15.orig/testit.sh	2009-04-12 10:08:59.000000000 +0300
+++ popt-1.15/testit.sh	2009-09-24 14:10:40.000000000 +0300
@@ -111,7 +111,7 @@
 run test1 "test1 - 54" "arg1: 0 arg2: (none) aFlag: 0xface" --nobitclr
 
 run test1 "test1 - 55" "\
-Usage: lt-test1 [-I?] [-c|--cb2=STRING] [--arg1] [-2|--arg2=ARG]
+Usage: test1 [-I?] [-c|--cb2=STRING] [--arg1] [-2|--arg2=ARG]
         [-3|--arg3=ANARG] [-onedash] [--optional=STRING] [--val]
         [-i|--int=INT] [-l|--long=LONG] [-L|--longlong=LONGLONG]
         [-f|--float=FLOAT] [-d|--double=DOUBLE] [--randint=INT]
@@ -120,7 +120,7 @@
         [-c|--cb=STRING] [--longopt] [-?|--help] [--usage] [--simple=ARG]" --usage
 
 run test1 "test1 - 56" "\
-Usage: lt-test1 [OPTION...]
+Usage: test1 [OPTION...]
       --arg1                      First argument with a really long
                                   description. After all, we have to test
                                   argument help wrapping somehow, right?
