--- /usr/portage/dev-libs/nspr/nspr-4.8.ebuild	2009-11-12 16:35:35.000000000 +0100
+++ ./dev-libs/nspr/nspr-4.8.ebuild	2010-04-22 21:32:58.000000000 +0200
@@ -18,9 +18,7 @@
 DEPEND=">=dev-db/sqlite-3.5"
 RDEPEND="${DEPEND}"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	mkdir build inst
 	epatch "${FILESDIR}"/${PN}-4.6.1-config.patch
 	epatch "${FILESDIR}"/${PN}-4.6.1-config-1.patch
@@ -32,7 +30,7 @@
 		mozilla/nsprpub/config/rules.mk
 }
 
-src_compile() {
+src_configure() {
 	cd "${S}"/build
 
 	echo > "${T}"/test.c
@@ -50,6 +48,10 @@
 		$(use_enable ipv6) \
 		$(use_enable debug) \
 		${myconf} || die "econf failed"
+}
+
+src_compile() {
+	cd "${S}"/build
 	make CC="$(tc-getCC)" CXX="$(tc-getCXX)" || die
 }
 
@@ -66,7 +68,7 @@
 	done
 	# cope with libraries being in /usr/lib/nspr
 	dodir /etc/env.d
-	echo "LDPATH=/usr/$(get_libdir)/nspr" > "${D}/etc/env.d/08nspr"
+	echo "LDPATH=/usr/$(get_libdir)/nspr" > "${D}/etc/env.d/08nspr-${ABI}"
 
 	# install nspr-config
 	dobin "${S}"/build/config/nspr-config
