diff -ur ffmpeg.orig/libavcodec/sparc/simple_idct_vis.c ffmpeg/libavcodec/sparc/simple_idct_vis.c
--- ffmpeg.orig/libavcodec/sparc/simple_idct_vis.c	2008-05-09 14:56:36.000000000 +0300
+++ ffmpeg/libavcodec/sparc/simple_idct_vis.c	2008-12-13 11:52:04.000000000 +0200
@@ -384,7 +384,7 @@
         "st %%f14, [%12+" dest "] \n\t"\
 
 
-inline void ff_simple_idct_vis(DCTELEM *data) {
+void ff_simple_idct_vis(DCTELEM *data) {
     int out1, out2, out3, out4;
     DECLARE_ALIGNED_8(int16_t, temp[8*8]);
 
