--- /usr/portage/media-libs/imlib2/imlib2-1.4.4.ebuild	2010-07-24 02:06:06.000000000 +0200
+++ ./media-libs/imlib2/imlib2-1.4.4.ebuild	2010-07-26 18:34:17.000000000 +0200
@@ -23,7 +23,7 @@
 	X? ( x11-libs/libXext x11-proto/xextproto )
 	mp3? ( media-libs/libid3tag )"
 
-src_compile() {
+src_configure() {
 	# imlib2 has diff configure options for x86/amd64 mmx
 	local myconf=""
 	if [[ $(tc-arch) == "amd64" ]] ; then
@@ -45,5 +45,5 @@
 		$(use_with mp3 id3) \
 		${myconf} \
 	"
-	enlightenment_src_compile
+	enlightenment_src_configure
 }
