--- /usr/portage/media-libs/imlib2/imlib2-1.4.2-r1.ebuild	2010-03-08 16:06:07.000000000 +0100
+++ ./media-libs/imlib2/imlib2-1.4.2-r1.ebuild	2010-04-25 23:14:49.000000000 +0200
@@ -22,15 +22,13 @@
 	X? ( x11-libs/libXext x11-proto/xextproto )
 	mp3? ( media-libs/libid3tag )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-libpng14.patch
 	epatch "${FILESDIR}"/${P}-CVE-2008-5187.patch #248057
 	sed -i '/bumpmap_la_LIBADD/s:$: -lm:' src/modules/filters/Makefile.in #276285
 }
 
-src_compile() {
+src_configure() {
 	# imlib2 has diff configure options for x86/amd64 mmx
 	local myconf=""
 	if [[ $(tc-arch) == "amd64" ]] ; then
@@ -52,5 +50,5 @@
 		$(use_with mp3 id3) \
 		${myconf} \
 	"
-	enlightenment_src_compile
+	enlightenment_src_configure
 }
