--- jbigkit/Makefile
+++ jbigkit/Makefile
@@ -8,7 +8,7 @@
 CCFLAGS = -O2 -W
 #CCFLAGS = -O -g -W -Wall -ansi -pedantic #-DDEBUG  # developer only
 
-CFLAGS = $(CCFLAGS) -I../libjbig
+CFLAGS += -I../libjbig
 
 VERSION=1.6
 
--- jbigkit/pbmtools/Makefile
+++ jbigkit/pbmtools/Makefile
@@ -12,10 +12,10 @@ CFLAGS = -g -Wall -ansi -pedantic -I../l
 all: pbmtojbg jbgtopbm pbmtojbg.txt jbgtopbm.txt pbm.txt pgm.txt
 
 pbmtojbg: pbmtojbg.o ../libjbig/libjbig.a
-	$(CC) $(CFLAGS) -o pbmtojbg pbmtojbg.o -L../libjbig -ljbig
+	$(CC) $(CFLAGS) $(LDFLAGS) -o pbmtojbg pbmtojbg.o -L../libjbig -ljbig
 
 jbgtopbm: jbgtopbm.o ../libjbig/libjbig.a
-	$(CC) $(CFLAGS) -o jbgtopbm jbgtopbm.o -L../libjbig -ljbig
+	$(CC) $(CFLAGS) $(LDFLAGS) -o jbgtopbm jbgtopbm.o -L../libjbig -ljbig
 
 jbgtopbm.o: jbgtopbm.c ../libjbig/jbig.h
 pbmtojbg.o: pbmtojbg.c ../libjbig/jbig.h
