Index: jbigkit/Makefile
===================================================================
--- jbigkit.orig/Makefile
+++ jbigkit/Makefile
@@ -13,22 +13,22 @@ CFLAGS = $(CCFLAGS) -I../libjbig
 VERSION=1.6
 
 all: lib pbm
-	@echo "Enter 'make test' in order to start some automatic tests."
+	@echo "Enter '$(MAKE) test' in order to start some automatic tests."
 
 lib:
-	(cd libjbig;  make "CC=$(CC)" "CFLAGS=$(CFLAGS)")
+	(cd libjbig;  $(MAKE) "CC=$(CC)" "CFLAGS=$(CFLAGS)")
 
 pbm: lib
-	(cd pbmtools; make "CC=$(CC)" "CFLAGS=$(CFLAGS)")
+	(cd pbmtools; $(MAKE) "CC=$(CC)" "CFLAGS=$(CFLAGS)")
 
 test: lib pbm
-	(cd libjbig;  make "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
-	(cd pbmtools; make "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
+	(cd libjbig;  $(MAKE) "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
+	(cd pbmtools; $(MAKE) "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
 
 clean:
 	rm -f *~ core
-	(cd libjbig; make clean)
-	(cd pbmtools; make clean)
+	(cd libjbig; $(MAKE) clean)
+	(cd pbmtools; $(MAKE) clean)
 
 distribution: clean
 	rm -f libjbig/libjbig.a
Index: jbigkit/pbmtools/Makefile
===================================================================
--- jbigkit.orig/pbmtools/Makefile
+++ jbigkit/pbmtools/Makefile
@@ -21,25 +21,25 @@ jbgtopbm.o: jbgtopbm.c ../libjbig/jbig.h
 pbmtojbg.o: pbmtojbg.c ../libjbig/jbig.h
 
 ../libjbig/libjbig.a: ../libjbig/jbig.c ../libjbig/jbig.h
-	cd ../libjbig ; make libjbig.a
+	cd ../libjbig ; $(MAKE) libjbig.a
 
 test: pbmtojbg jbgtopbm
-	make IMG=ccitt1     OPTIONSJ=      OPTIONSP=       dotest1
-	make IMG=ccitt2     OPTIONSJ=      OPTIONSP=       dotest1
-	make IMG=ccitt3     OPTIONSJ=      OPTIONSP=       dotest1
-	make IMG=xvlogo     OPTIONSJ=     "OPTIONSP=-d 3"  dotest1
-	make IMG=sandra     OPTIONSP=      OPTIONSJ=       dotest2g
-	make IMG=sandra     OPTIONSP=-b    OPTIONSJ=-b     dotest2g
-	make IMG=sandra     OPTIONSP=-q    OPTIONSJ=       dotest2g
-	make IMG=sandra    "OPTIONSP=-o 0" OPTIONSJ=       dotest2g
-	make IMG=sandra    "OPTIONSP=-o 2" OPTIONSJ=       dotest2g
-	make IMG=multi      OPTIONSP=      OPTIONSJ=       dotest2g
-	make IMG=multi      OPTIONSP=-b    OPTIONSJ=-b     dotest2g
-	make IMG=mx        "OPTIONSP=-q -s 3 -m 128"       dotest1
-	make IMG=mx        "OPTIONSP=-q -s 3 -m 128"       dotest2b
-	make IMG=mx        "OPTIONSP=-q -s 3 -m 128 -p 92" dotest2b
-	make IMG=mx        "OPTIONSP=-q -Y -1"             dotest2b
-	make IMG=mx        "OPTIONSP=-Y -1"                dotest2b
+	$(MAKE) IMG=ccitt1     OPTIONSJ=      OPTIONSP=       dotest1
+	$(MAKE) IMG=ccitt2     OPTIONSJ=      OPTIONSP=       dotest1
+	$(MAKE) IMG=ccitt3     OPTIONSJ=      OPTIONSP=       dotest1
+	$(MAKE) IMG=xvlogo     OPTIONSJ=     "OPTIONSP=-d 3"  dotest1
+	$(MAKE) IMG=sandra     OPTIONSP=      OPTIONSJ=       dotest2g
+	$(MAKE) IMG=sandra     OPTIONSP=-b    OPTIONSJ=-b     dotest2g
+	$(MAKE) IMG=sandra     OPTIONSP=-q    OPTIONSJ=       dotest2g
+	$(MAKE) IMG=sandra    "OPTIONSP=-o 0" OPTIONSJ=       dotest2g
+	$(MAKE) IMG=sandra    "OPTIONSP=-o 2" OPTIONSJ=       dotest2g
+	$(MAKE) IMG=multi      OPTIONSP=      OPTIONSJ=       dotest2g
+	$(MAKE) IMG=multi      OPTIONSP=-b    OPTIONSJ=-b     dotest2g
+	$(MAKE) IMG=mx        "OPTIONSP=-q -s 3 -m 128"       dotest1
+	$(MAKE) IMG=mx        "OPTIONSP=-q -s 3 -m 128"       dotest2b
+	$(MAKE) IMG=mx        "OPTIONSP=-q -s 3 -m 128 -p 92" dotest2b
+	$(MAKE) IMG=mx        "OPTIONSP=-q -Y -1"             dotest2b
+	$(MAKE) IMG=mx        "OPTIONSP=-Y -1"                dotest2b
 	rm -f test-*.jbg test-*.pbm test-*.pgm
 	./jbgtopbm ../examples/ccitt1.jbg | ./pbmtojbg > test-ccitt1.jbg
 	cmp ../examples/ccitt1.jbg test-ccitt1.jbg
