--- configure.ac.old	2009-06-18 22:35:33.000000000 +0200
+++ configure.ac	2009-06-19 00:42:02.000000000 +0200
@@ -20,6 +20,7 @@
 AC_ARG_ENABLE(ansi,             [  --enable-ansi           enable -ansi -pedantic gcc flags],enable_ansi=$enableval,enable_ansi=no)
 AC_ARG_ENABLE(verbose-mode,     [  --enable-verbose-mode   support verbose debug mode],enable_verbose_mode=$enableval,enable_verbose_mode=$USE_MAINTAINER_MODE)
 AC_ARG_ENABLE(man-pages,        [  --enable-man-pages      build manual pages],enable_man_pages=$enableval,enable_man_pages=yes)
+AC_ARG_ENABLE(tests,            [  --enable-tests          build tests],enable_tests=$enableval,enable_tests=$USE_MAINTAINER_MODE)
 
 if test "${enable_man_page}" != no; then
 dnl
@@ -32,6 +33,8 @@
 fi
 AM_CONDITIONAL(MAN_PAGES_ENABLED, test x$enable_man_pages = xyes)
 
+AM_CONDITIONAL(BUILD_TESTS, test x$enable_tests = xyes)
+
 GTK_DOC_CHECK([1.3])
 
 #### gcc warning flags
--- src/Makefile.am.old	2009-06-19 00:31:14.000000000 +0200
+++ src/Makefile.am	2009-06-19 00:32:19.000000000 +0200
@@ -1,5 +1,11 @@
 
-SUBDIRS = eggdbus tests
+SUBDIRS = eggdbus
+
+if BUILD_TESTS
+
+SUBDIRS += tests
+
+endif # BUILD_TESTS
 
 clean-local :
 	rm -f *~
--- docs/Makefile.am.old	2009-06-19 00:52:29.000000000 +0200
+++ docs/Makefile.am	2009-06-19 00:53:05.000000000 +0200
@@ -1,5 +1,11 @@
 
-SUBDIRS = man tests eggdbus
+SUBDIRS = man eggdbus
+
+if BUILD_TESTS
+
+SUBDIRS += tests
+
+endif # BUILD_TESTS
 
 EXTRA_DIST = version.xml.in
 
