From 0eec8eca8693bf709105b17854b6daf1ac9d2203 Mon Sep 17 00:00:00 2001
From: David Zeuthen <davidz@redhat.com>
Date: Sat, 04 Jul 2009 23:13:55 +0000
Subject: Don't leak object path in EggDBusMessage

---
diff --git a/src/eggdbus/eggdbusmessage.c b/src/eggdbus/eggdbusmessage.c
index ed747a5..4899ddd 100644
--- a/src/eggdbus/eggdbusmessage.c
+++ b/src/eggdbus/eggdbusmessage.c
@@ -127,6 +127,7 @@ egg_dbus_message_finalize (GObject *object)
   g_free (priv->signal_name);
   g_free (priv->sender);
   g_free (priv->destination);
+  g_free (priv->object_path);
   g_free (priv->error_name);
   g_free (priv->error_message);
 
--
cgit v0.8.2
