diff -ruN libsigc++-2.0.5/sigc++/adaptors/lambda/macros/group.h.m4 libsigc++-2.0.5.new/sigc++/adaptors/lambda/macros/group.h.m4
--- libsigc++-2.0.5/sigc++/adaptors/lambda/macros/group.h.m4	2004-07-23 07:51:42.000000000 -0700
+++ libsigc++-2.0.5.new/sigc++/adaptors/lambda/macros/group.h.m4	2004-10-09 14:08:11.329353224 -0700
@@ -68,7 +68,7 @@
 
   template <LOOP(class T_arg%1=void,$2)>
   struct deduce_result_type
-    { typedef typename functor_type::deduce_result_type<LOOP([
+    { typedef typename functor_type::template deduce_result_type<LOOP([
           typename value%1_type::template deduce_result_type<LOOP([
             _P_(T_arg%1)],$2)>::type],$1)
         >::type type; };
