--- /usr/portage/sys-devel/flex/flex-2.5.35.ebuild	2009-03-26 05:38:11.000000000 +0100
+++ ./sys-devel/flex/flex-2.5.35.ebuild	2010-03-15 00:13:59.000000000 +0100
@@ -18,9 +18,7 @@
 DEPEND="nls? ( sys-devel/gettext )"
 RDEPEND=""
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	[[ -n ${DEB_VER} ]] && epatch "${WORKDIR}"/${PN}_${PV}-${DEB_VER}.diff
 	epatch "${FILESDIR}"/${PN}-2.5.34-isatty.patch #119598
 	epatch "${FILESDIR}"/${PN}-2.5.33-pic.patch
@@ -28,10 +26,9 @@
 	sed -i 's:^LDFLAGS:LOADLIBES:' tests/test-pthread/Makefile.in #262989
 }
 
-src_compile() {
+src_configure() {
 	use static && append-ldflags -static
 	econf $(use_enable nls) || die
-	emake || die
 }
 
 src_install() {
