--- /usr/portage/app-text/hunspell/hunspell-1.2.6.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./app-text/hunspell/hunspell-1.2.6.ebuild	2010-05-30 19:07:21.000000000 +0200
@@ -22,10 +22,7 @@
 
 S=${WORKDIR}/${MY_P}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Upstream package creates some executables which names are too generic
 	# to be placed in /usr/bin - this patch prefixes them with 'hunspell-'.
 	# It modifies a Makefile.am file, hence eautoreconf.
@@ -34,7 +31,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	# I wanted to put the include files in /usr/include/hunspell
 	# but this means the openoffice build won't find them.
 	econf \
@@ -42,8 +39,6 @@
 		$(use_with ncurses ui) \
 		$(use_with readline readline) \
 		|| die "econf failed"
-
-	emake || die "emake failed"
 }
 
 src_install() {
