--- /usr/portage/app-text/hunspell/hunspell-1.1.9.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./app-text/hunspell/hunspell-1.1.9.ebuild	2010-05-30 19:06:36.000000000 +0200
@@ -19,9 +19,7 @@
 	sys-devel/gettext"
 RDEPEND="${DEPEND}"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	sed -i -e 's:tail +:tail -n +:' "${S}"/tests/test.sh ||\
 		die "Failed to fix-up tail for POSIX compliance"
 	# Upstream package creates executables 'example', 'munch'
@@ -32,15 +30,13 @@
 	eautoreconf -f
 }
 
-src_compile() {
+src_configure() {
 	# I wanted to put the include files in /usr/include/hunspell
 	# but this means the openoffice build won't find them.
 	econf \
 		$(use_with readline readline) \
 		$(use_with ncurses ui) \
 		|| die "econf failed"
-
-	emake || die "emake failed"
 }
 
 src_install() {
