--- configure.in.orig	2008-11-10 06:05:46.799625029 +0000
+++ configure.in	2008-11-10 06:04:41.225032297 +0000
@@ -11,15 +11,18 @@
 
 # Increment if the interface has additions, changes, removals.
 LT_CURRENT=1
+AC_SUBST(LT_CURRENT)
 
 # Increment any time the source changes; set to
 # 0 if you increment CURRENT
 LT_REVISION=0
+AC_SUBST(LT_REVISION)
 
 # Increment if any interfaces have been added; set to 0
 # if any interfaces have been removed. removal has
 # precedence over adding, so set to 0 if both happened.
 LT_AGE=0
+AC_SUBST(LT_AGE)
 
 VERSION_INFO="$LT_CURRENT:$LT_REVISION:$LT_AGE"
 AC_SUBST(VERSION_INFO)
