--- /usr/portage/eclass/distutils.eclass	2010-05-25 17:35:50.000000000 +0200
+++ ./eclass/distutils.eclass	2010-06-03 14:10:23.000000000 +0200
@@ -23,7 +23,11 @@
 esac
 
 if [[ -z "$(declare -p PYTHON_DEPEND 2> /dev/null)" ]]; then
-	DEPEND="dev-lang/python[lib32?]"
+	if [[ $(number_abis) -gt 1 ]] ; then
+		DEPEND="dev-lang/python[lib32?]"
+	else
+		DEPEND="dev-lang/python"
+	fi
 	RDEPEND="${DEPEND}"
 fi
 
@@ -294,6 +298,24 @@
 		die "${FUNCNAME}() can be used only in src_install() phase"
 	fi
 
+	if is_final_abi || (! has_multilib_profile); then
+		if [ -n "${PYTHON_SLOT_VERSION}" ] ; then
+			python=python${PYTHON_SLOT_VERSION}
+		else
+			python=python
+		fi
+	else
+		[[ -z $(get_abi_var SETARCH_ARCH ${ABI}) ]] && die "SETARCH_ARCH_${ABI} is missing in your portage profile take a look at http://wiki.github.com/sjnewbury/multilib-overlay to get further information"
+		if [ -n "${PYTHON_SLOT_VERSION}" ] ; then
+			python="setarch $(get_abi_var SETARCH_ARCH ${ABI}) python${PYTHON_SLOT_VERSION}-${ABI}"
+		elif [[ -n "${PYTHON}" ]]; then
+			python="setarch $(get_abi_var SETARCH_ARCH ${ABI}) ${PYTHON}"
+		else
+			python="setarch $(get_abi_var SETARCH_ARCH ${ABI}) python"
+		fi	
+	fi
+	einfo Using ${python}
+
 	_python_initialize_prefix_variables
 	_python_set_color_variables
 
@@ -390,6 +412,27 @@
 		done
 	fi
 
+	if has "${EAPI:-0}" 0 1 2; then
+		if is_final_abi || (! has_multilib_profile); then
+			if [ -n "${PYTHON_SLOT_VERSION}" ] ; then
+				python=python${PYTHON_SLOT_VERSION}
+			else
+				python=python
+			fi
+		else
+			[[ -z $(get_abi_var SETARCH_ARCH ${ABI}) ]] && die "SETARCH_ARCH_${ABI} is missing in your portage profile take a look at http://wiki.github.com/sjnewbury/multilib-overlay to get further information"
+			if [ -n "${PYTHON_SLOT_VERSION}" ] ; then
+				python="setarch $(get_abi_var SETARCH_ARCH ${ABI}) python${PYTHON_SLOT_VERSION}-${ABI}"
+			elif [[ -n "${PYTHON}" ]]; then
+				python="setarch $(get_abi_var SETARCH_ARCH ${ABI}) ${PYTHON}"
+			else
+				python="setarch $(get_abi_var SETARCH_ARCH ${ABI}) python"
+			fi	
+		fi
+	else
+		python="die"
+	fi
+	einfo Using ${python}
 	if [[ -n "${PYTHON_MODNAME}" ]]; then
 		if ! has "${EAPI:-0}" 0 1 2 || [[ -n "${SUPPORT_PYTHON_ABIS}" ]]; then
 			python_mod_optimize ${PYTHON_MODNAME}
