diff --git b/eclass/gst-plugins-base.eclass a/eclass/gst-plugins-base.eclass
index f4099fc..a9d8f57 100644
--- b/eclass/gst-plugins-base.eclass
+++ a/eclass/gst-plugins-base.eclass
@@ -116,7 +116,9 @@ gst-plugins-base_src_unpack() {
 
 gst-plugins-base_src_compile() {
 
-	gst-plugins-base_src_configure ${@}
+	if [[ ${EAPI:-0} -lt 2 ]]; then
+		gst-plugins-base_src_configure ${@}
+	fi
 
 	gst-plugins10_find_plugin_dir
 	emake || die "compile failure"
