--- eclass/mozconfig-3.eclass.orig	2010-03-27 22:37:55.000000000 +0100
+++ ./eclass/mozconfig-3.eclass	2010-03-27 22:37:31.000000000 +0100
@@ -8,25 +8,25 @@
 
 IUSE="gnome dbus startup-notification"
 
-RDEPEND="x11-libs/libXrender
-	x11-libs/libXt
-	x11-libs/libXmu
-	>=media-libs/jpeg-7
-	dev-libs/expat
+RDEPEND="x11-libs/libXrender[lib32?]
+	x11-libs/libXt[lib32?]
+	x11-libs/libXmu[lib32?]
+	>=media-libs/jpeg-7[lib32?]
+	dev-libs/expat[lib32?]
 	app-arch/zip
 	app-arch/unzip
-	>=x11-libs/gtk+-2.8.6
-	>=dev-libs/glib-2.8.2
-	>=x11-libs/pango-1.10.1
-	>=dev-libs/libIDL-0.8.0
-	gnome? ( >=gnome-base/gnome-vfs-2.16.3
-		>=gnome-base/libgnomeui-2.16.1
-		>=gnome-base/gconf-2.16.0
-		>=gnome-base/libgnome-2.16.0 )
-	dbus? ( >=dev-libs/dbus-glib-0.72 )
-	startup-notification? ( >=x11-libs/startup-notification-0.8 )
+	>=x11-libs/gtk+-2.8.6[lib32?]
+	>=dev-libs/glib-2.8.2[lib32?]
+	>=x11-libs/pango-1.10.1[lib32?]
+	>=dev-libs/libIDL-0.8.0[lib32?]
+	gnome? ( >=gnome-base/gnome-vfs-2.16.3[lib32?]
+		>=gnome-base/libgnomeui-2.16.1[lib32?]
+		>=gnome-base/gconf-2.16.0[lib32?]
+		>=gnome-base/libgnome-2.16.0[lib32?] )
+	dbus? ( >=dev-libs/dbus-glib-0.72[lib32?] )
+	startup-notification? ( >=x11-libs/startup-notification-0.8[lib32?] )
 	!<x11-base/xorg-x11-6.7.0-r2
-	>=x11-libs/cairo-1.6.0"
+	>=x11-libs/cairo-1.6.0[lib32?]"
 	#According to bugs #18573, #204520, and couple of others in Mozilla's
 	#bugzilla. libmng and mng support has been removed in 2003.
 
