--- /usr/portage/eclass/python.eclass	2010-05-29 19:05:51.000000000 +0200
+++ ./eclass/python.eclass	2010-06-03 16:47:13.000000000 +0200
@@ -98,7 +98,7 @@
 					accepted_version="1"
 				fi
 				if [[ "${accepted_version}" == "1" ]]; then
-					_PYTHON_ATOMS+=("=dev-lang/python-${python_versions[${i}]}*")
+					_PYTHON_ATOMS+=("=dev-lang/python-${python_versions[${i}]}*[lib32?]")
 				fi
 				if [[ "${python_versions[${i}]}" == "${python_minimal_version}" ]]; then
 					accepted_version="0"
@@ -108,7 +108,7 @@
 
 		if [[ "${python_all}" == "1" ]]; then
 			if [[ -z "${python_minimal_version}" && -z "${python_maximal_version}" ]]; then
-				_PYTHON_ATOMS+=("dev-lang/python")
+				_PYTHON_ATOMS+=("dev-lang/python[lib32?]")
 			else
 				python_versions=("${_CPYTHON2_SUPPORTED_ABIS[@]}" "${_CPYTHON3_SUPPORTED_ABIS[@]}")
 				python_minimal_version="${python_minimal_version:-${python_versions[0]}}"
@@ -118,7 +118,7 @@
 		else
 			if [[ "${python3}" == "1" ]]; then
 				if [[ -z "${python3_minimal_version}" && -z "${python3_maximal_version}" ]]; then
-					_PYTHON_ATOMS+=("=dev-lang/python-3*")
+					_PYTHON_ATOMS+=("=dev-lang/python-3*[lib32?]")
 				else
 					python_versions=("${_CPYTHON3_SUPPORTED_ABIS[@]}")
 					python_minimal_version="${python3_minimal_version:-${python_versions[0]}}"
@@ -128,7 +128,7 @@
 			fi
 			if [[ "${python2}" == "1" ]]; then
 				if [[ -z "${python2_minimal_version}" && -z "${python2_maximal_version}" ]]; then
-					_PYTHON_ATOMS+=("=dev-lang/python-2*")
+					_PYTHON_ATOMS+=("=dev-lang/python-2*[lib32?]")
 				else
 					python_versions=("${_CPYTHON2_SUPPORTED_ABIS[@]}")
 					python_minimal_version="${python2_minimal_version:-${python_versions[0]}}"
@@ -164,11 +164,11 @@
 		eerror "Use PYTHON_DEPEND instead of NEED_PYTHON."
 		die "NEED_PYTHON cannot be used in this EAPI"
 	fi
-	_PYTHON_ATOMS=(">=dev-lang/python-${NEED_PYTHON}")
+	_PYTHON_ATOMS=(">=dev-lang/python-${NEED_PYTHON}[lib32?]")
 	DEPEND+="${DEPEND:+ }${_PYTHON_ATOMS[@]}"
 	RDEPEND+="${RDEPEND:+ }${_PYTHON_ATOMS[@]}"
 else
-	_PYTHON_ATOMS=("dev-lang/python")
+	_PYTHON_ATOMS=("dev-lang/python[lib32?]")
 fi
 
 # @ECLASS-VARIABLE: PYTHON_USE_WITH
