--- /usr/portage/eclass/python.eclass	2010-12-24 16:05:41.000000000 +0100
+++ ./eclass/python.eclass	2010-12-25 02:56:48.000000000 +0100
@@ -205,6 +205,11 @@
 # Set this to a name of a USE flag if you need to make either PYTHON_USE_WITH or
 # PYTHON_USE_WITH_OR atoms conditional under a USE flag.
 
+#add lib32? to PYTHON_USE_WITH to ensure that 32bit python is build if needed
+if [[ "${PN}" != "python" ]]; then
+	PYTHON_USE_WITH+="${PYTHON_USE_WITH:+ }lib32?"
+fi
+
 if ! has "${EAPI:-0}" 0 1 && [[ -n ${PYTHON_USE_WITH} || -n ${PYTHON_USE_WITH_OR} ]]; then
 	_PYTHON_USE_WITH_ATOMS_ARRAY=()
 	if [[ -n "${PYTHON_USE_WITH}" ]]; then
@@ -426,6 +431,8 @@
 
 if ! has "${EAPI:-0}" 0 1 2 3 || { has "${EAPI:-0}" 2 3 && [[ -n "${PYTHON_USE_WITH}" || -n "${PYTHON_USE_WITH_OR}" ]]; }; then
 	EXPORT_FUNCTIONS pkg_setup
+else
+	EMULTILIB_PYTHON_NOT_EXPORTED+=" pkg_setup"
 fi
 
 _PYTHON_SHEBANG_BASE_PART_REGEX='^#![[:space:]]*([^[:space:]]*/usr/bin/env[[:space:]]+)?([^[:space:]]*/)?(jython|python)'
@@ -617,6 +624,8 @@
 
 	if [[ -n "${PYTHON_EXPORT_PHASE_FUNCTIONS}" ]]; then
 		EXPORT_FUNCTIONS src_prepare src_configure src_compile src_test src_install
+	else
+		EMULTILIB_PYTHON_NOT_EXPORTED+=" src_prepare src_configure src_compile src_test src_install"
 	fi
 fi
 
