--- /usr/portage/eclass/python.eclass	2010-07-18 23:05:51.000000000 +0200
+++ ./eclass/python.eclass	2010-07-26 13:39:57.000000000 +0200
@@ -194,6 +194,11 @@
 # Set this to a name of a USE flag if you need to make either PYTHON_USE_WITH or
 # PYTHON_USE_WITH_OR atoms conditional under a USE flag.
 
+#add lib32? to PYTHON_USE_WITH to ensure that 32bit python is build if needed
+if [[ "${PN}" != "python" ]]; then
+	PYTHON_USE_WITH+="${PYTHON_USE_WITH:+ }lib32?"
+fi
+
 if ! has "${EAPI:-0}" 0 1 && [[ -n ${PYTHON_USE_WITH} || -n ${PYTHON_USE_WITH_OR} ]]; then
 	_PYTHON_USE_WITH_ATOMS_ARRAY=()
 	if [[ -n "${PYTHON_USE_WITH}" ]]; then
@@ -412,6 +412,8 @@ python_pkg_setup() {
 
 if ! has "${EAPI:-0}" 0 1 2 3 || has "${EAPI:-0}" 2 3 && [[ -n "${PYTHON_USE_WITH}" || -n "${PYTHON_USE_WITH_OR}" ]]; then
 	EXPORT_FUNCTIONS pkg_setup
+else
+	EMULTILIB_PYTHON_NOT_EXPORTED+=" pkg_setup"
 fi
 
 # @FUNCTION: python_convert_shebangs
@@ -600,6 +602,8 @@ if ! has "${EAPI:-0}" 0 1; then
 
 	if [[ -n "${PYTHON_EXPORT_PHASE_FUNCTIONS}" ]]; then
 		EXPORT_FUNCTIONS src_prepare src_configure src_compile src_test src_install
+	else
+		EMULTILIB_PYTHON_NOT_EXPORTED+=" src_prepare src_configure src_compile src_test src_install"
 	fi
 fi
 
