--- /usr/portage/eclass/qt4-build.eclass	2010-07-11 12:35:48.000000000 +0200
+++ ./eclass/qt4-build.eclass	2010-07-17 21:44:52.000000000 +0200
@@ -2,6 +2,12 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo-x86/eclass/qt4-build.eclass,v 1.82 2010/08/14 18:09:10 hwoarang Exp $
 
+export EMULTILIB_SAVE_VARS="QTBASEDIR QTPREFIXDIR QTBINDIR QTLIBDIR \
+		QMAKE_LIBDIR_QT QTPCDIR QTDATADIR QTDOCDIR QTHEADERDIR \
+		QTPLUGINDIR QTSYSCONFDIR QTTRANSDIR QTEXAMPLESDIR \
+		QTDEMOSDIR QT_INSTALL_PREFIX PLATFORM QMAKE_CFLAGS \
+		QMAKE_CXXFLAGS QMAKE_LDFLAGS QT4_EXTRACT_DIRECTORIES"
+
 # @ECLASS: qt4-build.eclass
 # @MAINTAINER:
 # Ben de Groot <yngwin@gentoo.org>,
@@ -397,6 +406,7 @@
 	fix_includes
 	# remove .la files since we are building only shared Qt libraries
 	find "${D}"${QTLIBDIR} -name "*.la" -print0 | xargs -0 rm 
+	prep_ml_includes
 }
 
 # @FUNCTION: setqtenv
