--- /usr/portage/eclass/toolchain.eclass	2010-11-21 22:35:45.000000000 +0100
+++ ./eclass/toolchain.eclass	2010-12-25 03:12:13.000000000 +0100
@@ -162,6 +162,7 @@
 			tc_version_is_at_least "4.0" && IUSE+=" objc-gc mudflap"
 			tc_version_is_at_least "4.1" && IUSE+=" objc++"
 			tc_version_is_at_least "4.2" && IUSE+=" openmp"
+			tc_version_is_at_least "4.2" && IUSE+=" nsplugin"
 			tc_version_is_at_least "4.3" && IUSE+=" fixed-point"
 			if tc_version_is_at_least "4.4" ; then
 				IUSE+=" graphite"
@@ -1116,7 +1117,7 @@
 			done
 	fi
 
-	disable_multilib_libjava || die "failed to disable multilib java"
+#	disable_multilib_libjava || die "failed to disable multilib java"
 }
 
 gcc-library-configure() {
@@ -1207,7 +1208,14 @@
 	if ! is_gcj ; then
 		confgcc="${confgcc} --disable-libgcj"
 	elif use gtk ; then
-		confgcc="${confgcc} --enable-java-awt=gtk"
+		confgcc="${confgcc} --enable-java-awt=gtk \
+		$(use_enable nsplugin plugin)"
+	elif use qt ; then
+		confgcc="${confgcc} --enable-java-awt=qt"
+	fi
+
+	if is_crosscompile && is_gcj ; then
+		confgcc="${confgcc} --disable-jni"
 	fi
 
 	case $(tc-arch) in
@@ -1956,10 +1964,24 @@
 		mv -f "${D}"${PREFIX}/lib/libgcj.spec "${D}"${LIBPATH}/libgcj.spec
 
 	# SLOT up libgcj.pc (and let gcc-config worry about links)
-	local libgcj=$(find "${D}"${PREFIX}/lib/pkgconfig/ -name 'libgcj*.pc')
-	if [[ -n ${libgcj} ]] ; then
-		sed -i "/^libdir=/s:=.*:=${LIBPATH}:" "${libgcj}"
-		mv "${libgcj}" "${D}"/usr/lib/pkgconfig/libgcj-${GCC_PV}.pc || die
+	local libgcj
+	for libgcj in "${D}${PREFIX}"/lib/pkgconfig/libgcj*.pc ; do
+		if [[ -f ${libgcj} ]] ; then
+			sed -i "/^libdir=/s:=.*:=${LIBPATH}:" "${libgcj}"
+			if ! is_crosscompile; then
+				mv "${libgcj}" "${D}"/${PREFIX}/lib/pkgconfig/libgcj-${GCC_PV}.pc || die
+			else
+				dodir /${PREFIX}/${CTARGET}/${PREFIX}/lib/pkgconfig
+				mv "${libgcj}" "${D}"/${PREFIX}/${CTARGET}/${PREFIX}/lib/pkgconfig/libgcj-${GCC_PV}.pc || die
+			fi
+		fi
+	done
+
+	# Move JNI native (CTARGET) java libraries  
+	if is_crosscompile; then
+		dodir /${PREFIX}/${CTARGET}/${PREFIX}/lib
+		find "${D}"/${PREFIX}/lib -name 'gcj-*' -type d -exec \
+			mv {} "${D}"/${PREFIX}/${CTARGET}/${PREFIX}/lib/ \;
 	fi
 
 	# Rename jar because it could clash with Kaffe's jar if this gcc is
