diff -up notification-daemon/src/themes/standard/theme.c.fix notification-daemon/src/themes/standard/theme.c
--- a/notification-daemon/src/themes/standard/theme.c	2008-03-21 17:17:37.000000000 +0100
+++ b/notification-daemon/src/themes/standard/theme.c	2008-03-21 17:18:11.000000000 +0100
@@ -812,7 +812,7 @@ set_notification_text(GtkWindow *nw, con
 	update_content_hbox_visibility(windata);
 
 	gtk_widget_set_size_request(
-		((body != NULL && *body == '\0')
+		((body != NULL && *body != '\0')
 		 ? windata->body_label : windata->summary_label),
 		WIDTH - (IMAGE_SIZE + IMAGE_PADDING) - 10,
 		-1);
