diff -Naur libraries/cmos/CmosRW.cpp libraries/cmos/CmosRW.cpp
--- libraries/cmos/CmosRW.cpp	2008-02-13 19:12:43.000000000 +0000
+++ libraries/cmos/CmosRW.cpp	2009-01-29 11:29:02.992136416 +0000
@@ -19,6 +19,7 @@
 // compat header should always be first header if including system headers
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
+#include <stdio.h>
 #include <errno.h>
 #include <string.h>
 #include "CmosRWImpl.h"
diff -Naur libraries/memory/Memory_Linux.cpp libraries/memory/Memory_Linux.cpp
--- libraries/memory/Memory_Linux.cpp	2007-10-19 22:15:12.000000000 +0000
+++ libraries/memory/Memory_Linux.cpp	2009-01-29 11:28:33.745839933 +0000
@@ -22,6 +22,7 @@
 
 #include <errno.h>
 #include <sys/mman.h>   /* mmap */
+#include <stdio.h>
 #include <unistd.h>     /* getpagesize */
 #include <string.h>     /* memcpy etc. */
 
diff -Naur libraries/smbios/SmbiosStrategy_Linux.cpp libraries/smbios/SmbiosStrategy_Linux.cpp
--- libraries/smbios/SmbiosStrategy_Linux.cpp	2007-10-19 22:15:12.000000000 +0000
+++ libraries/smbios/SmbiosStrategy_Linux.cpp	2009-01-29 11:28:04.639844346 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <cstdio>
 #include <sstream>
 #include <string.h>
 
diff -Naur libraries/smi/SmiImpl.h libraries/smi/SmiImpl.h
--- libraries/smi/SmiImpl.h	2007-11-27 17:50:54.000000000 +0000
+++ libraries/smi/SmiImpl.h	2009-01-29 11:44:52.811833602 +0000
@@ -22,6 +22,8 @@
 #include "SmiLowLevel.h"
 #include "ExceptionImpl.h"
 
+#include <cstdio>
+
 namespace smi
 {
     // define our exceptions
