--- ./Lib/distutils/command/install.py~	2009-04-24 22:52:33.000000000 +0100
+++ ./Lib/distutils/command/install.py	2009-04-24 22:56:25.227771188 +0100
@@ -45,8 +45,8 @@
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
-        'platlib': '$base/lib/python',
+        'purelib': '$base/@@GENTOO_LIBDIR@@/python',
+        'platlib': '$base/@@GENTOO_LIBDIR@@/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
--- ./Lib/distutils/sysconfig.py~	2009-04-24 23:12:06.000000000 +0100
+++ ./Lib/distutils/sysconfig.py	2009-04-24 23:26:42.073760340 +0100
@@ -99,6 +99,10 @@
         prefix = plat_specific and EXEC_PREFIX or PREFIX
 
     if os.name == "posix":
+        if plat_specific or standard_lib:
+            lib = '@@GENTOO_LIBDIR@@'
+        else:
+            lib = 'lib'
         libpython = os.path.join(prefix,
                                  "lib", 
                                  "python" + get_python_version())
