diff -Naur openexr-1.6.1-orig/exrenvmap/main.cpp openexr-1.6.1/exrenvmap/main.cpp
--- openexr-1.6.1-orig/exrenvmap/main.cpp	2007-04-24 19:07:51.000000000 -0600
+++ openexr-1.6.1/exrenvmap/main.cpp	2008-01-12 20:28:56.000000000 -0600
@@ -43,6 +43,7 @@
 #include <makeLatLongMap.h>
 #include <ImfEnvmap.h>
 #include <iostream>
+#include <cstring>
 #include <exception>
 #include <stdlib.h>
 
diff -Naur openexr-1.6.1-orig/exrmaketiled/main.cpp openexr-1.6.1/exrmaketiled/main.cpp
--- openexr-1.6.1-orig/exrmaketiled/main.cpp	2007-04-24 19:08:45.000000000 -0600
+++ openexr-1.6.1/exrmaketiled/main.cpp	2008-01-12 20:28:14.000000000 -0600
@@ -45,6 +45,7 @@
 #include <iostream>
 #include <exception>
 #include <string>
+#include <cstring>
 #include <stdlib.h>
 
 using namespace Imf;
