--- /usr/portage/gnome-base/librsvg/librsvg-2.26.0.ebuild	2010-01-16 18:35:44.000000000 +0100
+++ ./gnome-base/librsvg/librsvg-2.26.0.ebuild	2010-05-05 23:51:09.000000000 +0200
@@ -38,8 +38,8 @@
 		--enable-gtk-theme"
 }
 
-src_unpack() {
-	gnome2_src_unpack
+src_prepare() {
+	gnome2_src_prepare
 
 	# gcc-4.3.2-r3 related segfault with various apps like firefox -- bug 239992
 	epatch "${FILESDIR}/${PN}-2.22.3-fix-segfault-with-firefox.patch"
