--- /usr/portage/sys-libs/libselinux/libselinux-2.0.85.ebuild	2010-09-30 01:07:18.000000000 +0200
+++ ./sys-libs/libselinux/libselinux-2.0.85.ebuild	2010-12-25 02:13:49.000000000 +0100
@@ -25,10 +25,7 @@
 RDEPEND="=sys-libs/libsepol-${SEPOL_VER}*
 	ruby? ( dev-lang/ruby )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	[ ! -z "${BUGFIX_PATCH}" ] && epatch "${BUGFIX_PATCH}"
 	epatch "${FILESDIR}"/${P}-headers.patch #338302
 
