--- /usr/portage/sys-libs/libselinux/libselinux-2.0.85.ebuild	2010-02-14 20:06:25.000000000 +0100
+++ ./sys-libs/libselinux/libselinux-2.0.85.ebuild	2010-03-14 16:58:58.000000000 +0100
@@ -25,10 +25,7 @@
 RDEPEND="=sys-libs/libsepol-${SEPOL_VER}*
 	ruby? ( dev-lang/ruby )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	[ ! -z "${BUGFIX_PATCH}" ] && epatch "${BUGFIX_PATCH}"
 
 	# fix up paths for multilib
