--- /usr/portage/media-libs/freetype/freetype-1.4_pre20080316-r2.ebuild	2010-10-10 19:06:00.000000000 +0200
+++ ./media-libs/freetype/freetype-1.4_pre20080316-r2.ebuild	2010-12-25 01:11:43.000000000 +0100
@@ -21,10 +21,7 @@
 
 S="${WORKDIR}"/${PN}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# remove unneeded include for BSD (#104016)
 	epatch "${FILESDIR}"/freetype-1.4_pre-malloc.patch
 
@@ -72,23 +69,34 @@
 	done
 }
 
-src_compile() {
+src_configure() {
 	use kpathsea && kpathseaconf="--with-kpathsea-lib=/usr/$(get_libdir) --with-kpathsea-include=/usr/include"
 
 	# core
 	einfo "Building core library..."
 	econf $(use_enable nls) || die "econf failed"
-	emake || die "emake failed"
 
 	# contrib
 	cd "${S}"/freetype1-contrib/ttf2pk
 	einfo "Building ttf2pk..."
 	econf ${kpathseaconf} || die "econf ttf2pk failed"
-	emake || die "emake ttf2pk failed"
 	for x in ttf2bdf ttf2pfb ttfbanner; do
 		cd "${S}"/freetype1-contrib/${x}
 		einfo "Building ${x}..."
 		econf || die "econf ${x} failed"
+	done
+}
+
+src_compile() {
+	emake || die "emake failed"
+
+	# contrib
+	cd "${S}"/freetype1-contrib/ttf2pk
+	einfo "Building ttf2pk..."
+	emake || die "emake ttf2pk failed"
+	for x in ttf2bdf ttf2pfb ttfbanner; do
+		cd "${S}"/freetype1-contrib/${x}
+		einfo "Building ${x}..."
 		emake || die "emake ${x} failed"
 	done
 }
