--- /usr/portage/media-libs/freetype/freetype-1.4_pre20080316-r2.ebuild	2010-10-10 19:06:00.000000000 +0200
+++ ./media-libs/freetype/freetype-1.4_pre20080316-r2.ebuild	2011-01-23 17:50:33.000000000 +0100
@@ -21,10 +21,7 @@
 
 S="${WORKDIR}"/${PN}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# remove unneeded include for BSD (#104016)
 	epatch "${FILESDIR}"/freetype-1.4_pre-malloc.patch
 
@@ -72,6 +69,12 @@
 	done
 }
 
+src_configure() {
+	# to configure contrib we must have compiled the core already
+	# thus we do all in the compile step
+	:
+}
+
 src_compile() {
 	use kpathsea && kpathseaconf="--with-kpathsea-lib=/usr/$(get_libdir) --with-kpathsea-include=/usr/include"
 
