--- /usr/portage/media-libs/freetype/freetype-2.3.9-r1.ebuild	2009-05-06 18:08:01.000000000 +0200
+++ ./media-libs/freetype/freetype-2.3.9-r1.ebuild	2010-04-15 22:42:29.000000000 +0200
@@ -24,10 +24,7 @@
 RDEPEND="${DEPEND}
 		!<media-libs/fontconfig-2.3.2-r2"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	enable_option() {
 		sed -i -e "/#define $1/a #define $1" \
 			include/freetype/config/ftoption.h \
@@ -76,11 +73,14 @@
 	epunt_cxx
 }
 
-src_compile() {
+src_configure() {
 	append-flags -fno-strict-aliasing
 
 	type -P gmake &> /dev/null && export GNUMAKE=gmake
 	econf
+}
+
+src_compile() {
 	emake || die "emake failed"
 
 	if use utils; then
