--- /usr/portage/dev-libs/openssl/openssl-0.9.8m-r1.ebuild	2010-03-09 14:25:30.000000000 +0100
+++ ./dev-libs/openssl/openssl-0.9.8m-r1.ebuild	2010-04-02 18:52:10.000000000 +0200
@@ -22,10 +22,7 @@
 	test? ( sys-devel/bc )"
 PDEPEND="app-misc/ca-certificates"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.9.7e-gentoo.patch
 	epatch "${FILESDIR}"/${PN}-0.9.8b-doc-updates.patch
 	epatch "${FILESDIR}"/${PN}-0.9.8e-bsd-sparc64.patch
@@ -66,7 +63,7 @@
 	./config --test-sanity || die "I AM NOT SANE"
 }
 
-src_compile() {
+src_configure() {
 	unset APPS #197996
 
 	tc-export CC AR RANLIB
@@ -119,7 +116,9 @@
 		-e "/^CFLAG/s:=.*:=${CFLAG} ${CFLAGS}:" \
 		-e "/^SHARED_LDFLAGS=/s:$: ${LDFLAGS}:" \
 		Makefile || die
+}
 
+src_compile() {
 	# depend is needed to use $confopts
 	# rehash is needed to prep the certs/ dir
 	emake -j1 depend || die "depend failed"
