--- /usr/portage/dev-libs/openssl/openssl-0.9.8o-r2.ebuild	2010-08-14 07:53:33.000000000 +0200
+++ ./dev-libs/openssl/openssl-0.9.8o-r2.ebuild	2010-08-15 15:54:24.000000000 +0200
@@ -32,10 +32,7 @@
 		rm -f "${ROOT}"/usr/$(get_libdir)/libssl.so.0.9.8
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.9.7e-gentoo.patch
 	epatch "${FILESDIR}"/${PN}-0.9.8e-bsd-sparc64.patch
 	epatch "${FILESDIR}"/${PN}-0.9.8h-ldflags.patch #181438
@@ -69,7 +66,7 @@
 	./config --test-sanity || die "I AM NOT SANE"
 }
 
-src_compile() {
+src_configure() {
 	unset APPS #197996
 	unset SCRIPTS #312551
 
@@ -123,7 +120,9 @@
 		-e "/^CFLAG/s:=.*:=${CFLAG} ${CFLAGS}:" \
 		-e "/^SHARED_LDFLAGS=/s:$: ${LDFLAGS}:" \
 		Makefile || die
+}
 
+src_compile() {
 	# depend is needed to use $confopts
 	emake -j1 depend || die "depend failed"
 	emake -j1 build_libs || die "make build_libs failed"
