--- /usr/portage/dev-libs/openssl/openssl-1.0.0a.ebuild	2010-07-18 22:05:39.000000000 +0200
+++ ./dev-libs/openssl/openssl-1.0.0a.ebuild	2010-07-25 20:59:00.000000000 +0200
@@ -22,10 +22,7 @@
 	test? ( sys-devel/bc )"
 PDEPEND="app-misc/ca-certificates"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.9.7e-gentoo.patch
 	epatch "${FILESDIR}"/${PN}-0.9.8l-binutils.patch #289130
 
@@ -55,7 +52,7 @@
 	./config --test-sanity || die "I AM NOT SANE"
 }
 
-src_compile() {
+src_configure() {
 	unset APPS #197996
 	unset SCRIPTS #312551
 
@@ -110,7 +107,9 @@
 		-e "/^CFLAG/s:=.*:=${CFLAG} ${CFLAGS}:" \
 		-e "/^SHARED_LDFLAGS=/s:$: ${LDFLAGS}:" \
 		Makefile || die
+}
 
+src_compile() {
 	# depend is needed to use $confopts
 	# rehash is needed to prep the certs/ dir
 	emake -j1 depend || die "depend failed"
