From bf1997a08dee4f4aad17a00b06a4dbb52f17ffcd Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 28 Apr 2010 18:03:20 +0000
Subject: Bug #575038 - Cursor/caret mode breaks magic spacebar

---
diff --git a/gtkhtml/gtkhtml.c b/gtkhtml/gtkhtml.c
index 17a8692..c92de7e 100644
--- a/gtkhtml/gtkhtml.c
+++ b/gtkhtml/gtkhtml.c
@@ -1032,7 +1032,7 @@ update_mouse_cursor (GtkWidget *widget, guint state)
 	motion_notify_event (widget, &event);
 }
 
-static gint
+static gboolean
 key_press_event (GtkWidget *widget, GdkEventKey *event)
 {
 	GtkHTML *html = GTK_HTML (widget);
@@ -1116,7 +1116,7 @@ key_press_event (GtkWidget *widget, GdkEventKey *event)
 	return retval;
 }
 
-static gint
+static gboolean
 key_release_event (GtkWidget *widget, GdkEventKey *event)
 {
 	GtkHTML *html = GTK_HTML (widget);
@@ -4778,7 +4778,7 @@ scroll_command (GtkHTML *html,
 	gdouble page_size;
 
 	/* we dont want scroll in editable (move cursor instead) */
-	if (html_engine_get_editable (html->engine) || html->engine->caret_mode)
+	if (html_engine_get_editable (html->engine))
 		return FALSE;
 
 	adjustment = gtk_layout_get_vadjustment (GTK_LAYOUT (html));
--
cgit v0.8.3.1
