Index: libsndfile-1.0.17/examples/Makefile.am
===================================================================
--- libsndfile-1.0.17.orig/examples/Makefile.am
+++ libsndfile-1.0.17/examples/Makefile.am
@@ -2,7 +2,7 @@
 
 bin_PROGRAMS = sndfile-info sndfile-play sndfile-convert
 
-noinst_PROGRAMS = make_sine sfprocess list_formats cooledit-fixer generate
+EXTRA_PROGRAMS = make_sine sfprocess list_formats cooledit-fixer generate
 
 # This is the BeOS version of sndfile-play. It needs to be compiled with the C++
 # compiler.
Index: libsndfile-1.0.17/tests/Makefile.am
===================================================================
--- libsndfile-1.0.17.orig/tests/Makefile.am
+++ libsndfile-1.0.17/tests/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-noinst_PROGRAMS = sfversion floating_point_test write_read_test \
+check_PROGRAMS = sfversion floating_point_test write_read_test \
 	lossy_comp_test error_test ulaw_test alaw_test dwvw_test \
 	peak_chunk_test command_test stdin_test stdout_test stdio_test \
 	pcm_test headerless_test pipe_test benchmark header_test misc_test \
