Index: gnome-keyring-2.26.3/tests/test.make
===================================================================
--- gnome-keyring-2.26.3.orig/tests/test.make
+++ gnome-keyring-2.26.3/tests/test.make
@@ -22,9 +22,11 @@ noinst_PROGRAMS= \
 	run-auto-test \
 	run-prompt-test
 	
-run-auto-test.c: $(UNIT_AUTO) Makefile.am
+run-auto-test.h: $(UNIT_AUTO) Makefile.am
 	sh $(top_srcdir)/tests/prep-tests.sh -b run-auto-test $(UNIT_AUTO)
 
+run-auto-test.c: run-auto-test.h
+
 run_auto_test_SOURCES = \
 	run-auto-test.c \
 	run-auto-test.h \
@@ -37,9 +39,11 @@ run_auto_test_LDADD = \
 run_auto_test_CFLAGS = \
 	$(UNIT_FLAGS)
 
-run-prompt-test.c: $(UNIT_PROMPT) Makefile.am
+run-prompt-test.h: $(UNIT_PROMPT) Makefile.am
 	sh $(top_srcdir)/tests/prep-tests.sh -b run-prompt-test $(UNIT_PROMPT)
 
+run-prompt-test.c: run-prompt-test.h
+
 run_prompt_test_SOURCES = \
 	run-prompt-test.c \
 	run-prompt-test.h \
@@ -52,6 +56,12 @@ run_prompt_test_LDADD = \
 run_prompt_test_CFLAGS = \
 	$(UNIT_FLAGS)
 
+BUILT_SOURCES = \
+        run-auto-test.c \
+        run-auto-test.h \
+        run-prompt-test.c \
+        run-prompt-test.h
+
 # ------------------------------------------------------------------------------
 # Run the tests
 
