From 453c143338895bdd094c9398e8ef336d0f8cba8e Mon Sep 17 00:00:00 2001
From: Nshepperd <nshepperd@gmail.com>
Date: Wed, 05 May 2010 14:30:01 +0000
Subject: Fix checking of uninitialized value in prompting code.

Fixes problem repeatedly asking for password to unlock. See
bug #616071
---
diff --git a/daemon/prompt/gkd-prompt.c b/daemon/prompt/gkd-prompt.c
index a4ea7fc..645255a 100644
--- a/daemon/prompt/gkd-prompt.c
+++ b/daemon/prompt/gkd-prompt.c
@@ -796,7 +796,7 @@ gkd_prompt_get_transport_param (GkdPrompt *self, const gchar *name, gsize *n_val
 {
 	g_return_val_if_fail (GKD_IS_PROMPT (self), NULL);
 	g_return_val_if_fail (name, NULL);
-	g_return_val_if_fail (*n_value, NULL);
+	g_return_val_if_fail (n_value, NULL);
 
 	if (self->pv->failure)
 		return NULL;
--
cgit v0.8.3.1
