--- /usr/portage/net-nds/openldap/openldap-2.3.41.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./net-nds/openldap/openldap-2.3.41.ebuild	2010-04-03 21:50:03.000000000 +0200
@@ -173,9 +173,7 @@
 	enewuser ldap 439 -1 /usr/$(get_libdir)/openldap ldap
 }
 
-src_unpack() {
-	unpack ${A}
-
+src_prepare() {
 	# According to MDK, the link order needs to be changed so that
 	# on systems w/ MD5 passwords the system crypt library is used
 	# (the net result is that "passwd" can be used to change ldap passwords w/
@@ -220,7 +218,7 @@
 	epatch "${FILESDIR}/openldap-2.3.XY-gcc44.patch"
 }
 
-src_compile() {
+src_configure() {
 	local myconf
 
 	#Fix for glibc-2.8 and ucred. Bug 228457.
@@ -296,7 +294,9 @@
 		--enable-shared \
 		--libexecdir=/usr/$(get_libdir)/openldap \
 		${myconf} || die "configure failed"
+}
 
+src_compile() {
 	# Adding back -j1 as upstream didn't answer on parallel make issue yet
 	emake -j1 depend || die "make depend failed"
 	emake -j1 || die "make failed"
