--- lib/replace/libreplace_ld.m4.orig	2009-09-30 14:17:40.000000000 +0200
+++ lib/replace/libreplace_ld.m4	2009-10-07 02:07:26.000000000 +0200
@@ -249,6 +249,22 @@
 		esac
 ])
 
+AC_DEFUN([AC_LD_VERSIONSCRIPT],
+[
+       AC_SUBST(VERSIONSCRIPT)
+       VERSIONSCRIPT=""
+       case "$host_os" in 
+               *linux*|*gnu*)
+                       VERSIONSCRIPT="-Wl,--version-script"
+                       ;;
+               *solaris*)
+                       if test "${GCC}" = "yes"; then
+                               VERSIONSCRIPT="-Wl,--version-script"
+                       fi
+                       ;;
+               esac
+])
+
 AC_DEFUN([AC_LIBREPLACE_MDLD],
 [
 	AC_REQUIRE([AC_LIBREPLACE_LD_SHLIB_LINKER])
