=== capseo/tools/Makefile.am
==================================================================
--- capseo/tools/Makefile.am	(revision 5574)
+++ capseo/tools/Makefile.am	(local)
@@ -3,13 +3,13 @@
 ANSI_CXXFLAGS = -ansi -pedantic -Wall -Wno-long-long -Wno-unknown-pragmas
 AM_CXXFLAGS = $(ANSI_CXXFLAGS) $(CAPTURY_CFLAGS) -DVERSION="\"@CAPSEO_VERSION@\"" -DTHEORA=@THEORA@
 
-bin_PROGRAMS = cpsinfo cpsplay cpsrecode
+bin_PROGRAMS = cpsinfo cpsrecode
 
 cpsinfo_SOURCES = cpsinfo.cpp
 cpsinfo_LDFLAGS = $(top_builddir)/src/libcapseo.la
 
-cpsplay_SOURCES = cpsplay.cpp
-cpsplay_LDFLAGS = $(top_builddir)/src/libcapseo.la $(X11_LIBS) -lGL
+# cpsplay_SOURCES = cpsplay.cpp
+# cpsplay_LDFLAGS = $(top_builddir)/src/libcapseo.la $(X11_LIBS) -lGL
 
 cpsrecode_SOURCES = cpsrecode.cpp
 cpsrecode_LDFLAGS = $(top_builddir)/src/libcapseo.la
