diff -Naur tmp-old/kdc/Makefile.am tmp-new/kdc/Makefile.am
--- kdc/Makefile.am	2008-06-02 14:38:03.000000000 +0000
+++ kdc/Makefile.am	2008-06-03 04:06:19.000000000 +0000
@@ -84,12 +84,10 @@
 	$(LIB_roken) \
 	$(DBLIB) 
 
-if PKINIT
-LIB_pkinit = $(top_builddir)/lib/hx509/libhx509.la
-endif
+LIB_hx509 = $(top_builddir)/lib/hx509/libhx509.la
 
 libkdc_la_LIBADD = \
-	$(LIB_pkinit) \
+	$(LIB_hx509) \
 	$(top_builddir)/lib/hdb/libhdb.la \
 	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
