diff --git a/tests/can/Makefile.am b/tests/can/Makefile.am
index f881667..12cd2fe 100644
--- a/tests/can/Makefile.am
+++ b/tests/can/Makefile.am
@@ -14,7 +14,8 @@ port = 49188
 do_subst = sed -e 's,[@]srcdir[@],$(srcdir),g' \
 	-e 's,[@]port[@],$(port),g' \
 	-e 's,[@]objdir[@],$(top_builddir)/tests/can,g' \
-	-e 's,[@]EGREP[@],$(EGREP),g' 
+	-e 's,[@]EGREP[@],$(EGREP),g' \
+	-e 's,[@]env_setup[@],$(top_builddir)/tests/bin/setup-env,g'
 
 test_can: test_can.in Makefile
 	$(do_subst) < $(srcdir)/test_can.in > test_can.tmp
diff --git a/tests/can/check-can.in b/tests/can/check-can.in
index c687bce..70e5d38 100644
--- a/tests/can/check-can.in
+++ b/tests/can/check-can.in
@@ -36,12 +36,30 @@
 
 srcdir="@srcdir@"
 objdir="@objdir@"
+env_setup="@env_setup@"
+
+. ${env_setup}
 
 # If there is no useful db support compile in, disable test
 ../db/have-db || exit 77
 
 ./test_can TEST.H5L.SE heim-0.8 || exit 1
 ./test_can TEST.H5L.SE apple-10.4 || exit 1
-./test_can HEIMDAL.CITI.UMICH.EDU mit-pkinit-20070607 || exit 1
+
+rsa=yes
+pkinit=no
+if ${hxtool} info | grep 'rsa: hx509 null RSA' > /dev/null ; then
+    rsa=no
+fi
+if ${hxtool} info | grep 'rand: not available' > /dev/null ; then
+    rsa=no
+fi
+if ${kinit} --help 2>&1 | grep "CA certificates" > /dev/null; then
+    pkinit=yes
+fi
+
+if test "$pkinit" = yes -a "$rsa" = yes ; then
+	./test_can HEIMDAL.CITI.UMICH.EDU mit-pkinit-20070607 || exit 1
+fi
 
 exit 0
diff --git a/tests/ldap/init.ldif b/tests/ldap/init.ldif
index 9cf39b1..d087172 100644
--- a/tests/ldap/init.ldif
+++ b/tests/ldap/init.ldif
@@ -22,7 +22,7 @@ loginShell: /bin/bash
 gecos: Netbios root user
 structuralObjectClass: inetOrgPerson
 creatorsName: cn=root,dc=test,dc=h5l,dc=se
-userPassword:: AAAAAA
+userPassword:: AAAAAA==
 objectClass: krb5KDCEntry
 krb5KeyVersionNumber: 2
 krb5PrincipalName: suser@TEST.H5L.SE
diff --git a/tests/can/Makefile.in b/tests/can/Makefile.in
index e40eeb7..fb274b2 100644
--- a/tests/can/Makefile.in
+++ b/tests/can/Makefile.in
@@ -342,7 +342,8 @@ port = 49188
 do_subst = sed -e 's,[@]srcdir[@],$(srcdir),g' \
 	-e 's,[@]port[@],$(port),g' \
 	-e 's,[@]objdir[@],$(top_builddir)/tests/can,g' \
-	-e 's,[@]EGREP[@],$(EGREP),g' 
+	-e 's,[@]EGREP[@],$(EGREP),g' \
+	-e 's,[@]env_setup[@],$(top_builddir)/tests/bin/setup-env,g'
 
 CLEANFILES = $(TESTS) *.tmp *.cf \
 	current-db* \
