diff -Naur heimdal-0.7.2.orig/lib/editline/Makefile.am heimdal-0.7.2/lib/editline/Makefile.am
--- heimdal-0.7.2.orig/lib/editline/Makefile.am	2007-09-07 07:59:26.000000000 +0200
+++ heimdal-0.7.2/lib/editline/Makefile.am	2007-09-07 07:59:48.000000000 +0200
@@ -35,6 +35,8 @@
 	unix.h \
 	$(EXTRA_SOURCE)
 
+libeditline_la_LIBADD = @LIB_tgetent@
+
 EXTRA_SOURCE = $(ES) 
 
 libel_compat_la_SOURCES = edit_compat.c
diff -Naur heimdal-0.7.2.orig/lib/roken/Makefile.am heimdal-0.7.2/lib/roken/Makefile.am
--- heimdal-0.7.2.orig/lib/roken/Makefile.am	2007-09-07 07:59:26.000000000 +0200
+++ heimdal-0.7.2/lib/roken/Makefile.am	2007-09-07 08:00:17.000000000 +0200
@@ -104,7 +104,7 @@
 
 EXTRA_DIST = roken.awk roken.h.in
 
-libroken_la_LIBADD = @LTLIBOBJS@
+libroken_la_LIBADD = @LTLIBOBJS@ $(LIB_crypt)
 if have_cgetent
 else
 libroken_la_LIBADD += $(DBLIB)
