--- lib/krb5/principal.c.orig	2009-08-27 09:34:57.000000000 +0200
+++ lib/krb5/principal.c	2009-08-27 09:36:02.000000000 +0200
@@ -465,6 +465,12 @@
     return unparse_name(context, principal, name, 0);
 }
 
+void KRB5_LIB_FUNCTION
+krb5_free_unparsed_name(krb5_context context, char *str)
+{
+    krb5_xfree(str);
+}
+
 krb5_error_code KRB5_LIB_FUNCTION
 krb5_unparse_name_flags(krb5_context context,
 			krb5_const_principal principal,
--- lib/krb5/version-script.map.orig	2009-08-27 09:38:48.000000000 +0200
+++ lib/krb5/version-script.map	2009-08-27 09:39:22.000000000 +0200
@@ -280,6 +280,7 @@
 		krb5_free_principal;
 		krb5_free_salt;
 		krb5_free_ticket;
+		krb5_free_unparsed_name;
 		krb5_fwd_tgt_creds;
 		krb5_generate_random_block;
 		krb5_generate_random_keyblock;
