--- /usr/portage/dev-lang/tk/tk-8.5.7-r1.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./dev-lang/tk/tk-8.5.7-r1.ebuild	2010-06-06 01:14:48.000000000 +0200
@@ -39,9 +39,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-8.4.11-multilib.patch
 
 	# Bug 125971
@@ -51,7 +49,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	tc-export CC
 	cd "${S}"/unix
 
@@ -62,8 +60,6 @@
 		$(use_enable threads) \
 		$(use_enable truetype xft) \
 		$(use_enable debug symbols) || die
-
-	emake || die
 }
 
 src_install() {
