--- /usr/portage/dev-lang/tk/tk-8.5.7.ebuild	2010-06-16 02:35:39.000000000 +0200
+++ ./dev-lang/tk/tk-8.5.7.ebuild	2010-06-19 01:48:29.000000000 +0200
@@ -36,9 +36,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-8.4.11-multilib.patch
 
 	# Bug 125971
@@ -48,7 +46,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	tc-export CC
 	cd "${S}"/unix
 
@@ -59,8 +57,6 @@
 		$(use_enable threads) \
 		$(use_enable truetype xft) \
 		$(use_enable debug symbols)
-
-	emake || die
 }
 
 src_install() {
