--- /usr/portage/dev-lang/tk/tk-8.4.18.ebuild	2010-06-16 02:35:39.000000000 +0200
+++ ./dev-lang/tk/tk-8.4.18.ebuild	2010-06-19 01:47:17.000000000 +0200
@@ -34,9 +34,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/remove-control-v-8.4.9.diff
 	epatch "${FILESDIR}"/${PN}-8.4.9-man.patch
 	epatch "${FILESDIR}"/${PN}-8.4.11-multilib.patch
@@ -55,7 +53,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	tc-export CC
 	cd "${S}"/unix
 
@@ -65,8 +63,6 @@
 		--with-tcl=/usr/${mylibdir} \
 		$(use_enable threads) \
 		$(use_enable debug symbols)
-
-	emake || die
 }
 
 src_install() {
