--- /usr/portage/media-libs/capseo/capseo-0.3.0_pre200712251-r2.ebuild	2009-03-24 04:38:57.000000000 +0100
+++ ./media-libs/capseo/capseo-0.3.0_pre200712251-r2.ebuild	2010-06-01 00:00:43.000000000 +0200
@@ -21,12 +21,7 @@
 
 S="${WORKDIR}/captury-${PV}/${PN}"
 
-EMULTILIB_PKG="true"
-
-src_unpack() {
-	unpack ${A} || die
-
-	cd "${S}"
+src_prepare() {
 	einfo "pwd: $(pwd)"
 	epatch "${FILESDIR}/no-cpsplay.diff"
 
@@ -35,59 +30,23 @@
 	fi
 }
 
-src_compile() {
-	if [[ -z ${OABI} ]] && has_multilib_profile; then
-		use debug && append-flags -O0 -g3
-		use debug || append-flags -DNDEBUG=1
-
-		einfo "Building multilib ${PN} for ABIs: $(get_install_abis)"
-		OABI=${ABI}
-		for ABI in $(get_install_abis); do
-			export ABI=${ABI}
-			src_compile
-		done
-		ABI=${OABI}
-		return
-	fi
-
-	cd "${S}" || die
+src_configure() {
+	use debug && append-flags -O0 -g3
+	use debug || append-flags -DNDEBUG=1
 
-	# obviousely in src_install() it is set to "default" on non-multilib hosts,
-	# but isn't in src_compile()
-	ABI=${ABI:-default}
-
-	mkdir abi-${ABI}
-	cd abi-${ABI}
-
-	local myconf=
 	case ${ABI} in
 		amd64|x86)
 			myconf="${myconf} --with-accel=${ABI}"
 			;;
 	esac
 
-	if is_final_abi; then
-		myconf="${myconf} $(use_enable theora)"
-	else
-		# drop unnecessary theora dependency for secondary ABIs (as theora is
-		# only used for the cpsrecode tool anyway).
-		# see bug 200093.
-		myconf="${myconf} --disable-theora"
-	fi
-
-	../configure ${myconf} \
-		--prefix="/usr" \
-		--host="$(get_abi_CHOST ${ABI})" \
-		--libdir="/usr/$(get_libdir)" \
-		|| die "./configure for ABI ${ABI} failed"
+	myconf="${myconf} $(use_enable theora)"
 
-	emake || die "make for ABI ${ABI} failed"
+	econf ${myconf} || die "econf failed"
 }
 
 src_install() {
-	for ABI in $(get_install_abis); do
-		make -C abi-${ABI} install DESTDIR="${D}" || die "make install for ABI ${ABI} failed."
-	done
+	emake install DESTDIR="${D}" || die "emake install failed."
 
 	rm "${D}/usr/bin/cpsplay" # currently unsupported
 
