diff -Naur unixODBC-2.2.12.orig/configure.in unixODBC-2.2.12/configure.in
--- unixODBC-2.2.12.orig/configure.in	2008-05-29 15:54:09.000000000 +0200
+++ unixODBC-2.2.12/configure.in	2008-05-29 15:58:32.000000000 +0200
@@ -163,9 +163,6 @@
 dnl Configure libtool
 AM_PROG_LIBTOOL
 
-dnl Configure libltdl
-AC_CONFIG_SUBDIRS(libltdl)
-
 use_builtin_libtool="no"
 
 dnl override the libtool lines if we use the system libs
diff -Naur unixODBC-2.2.12.orig/exe/Makefile.am unixODBC-2.2.12/exe/Makefile.am
--- unixODBC-2.2.12.orig/exe/Makefile.am	2008-05-29 15:54:09.000000000 +0200
+++ unixODBC-2.2.12/exe/Makefile.am	2008-05-29 15:57:14.000000000 +0200
@@ -103,6 +103,6 @@
 
 if WITHLT
 dltest_DEPENDENCIES = $(LIBLTDL)
-dltest_LDADD = $(LIBLTDL) $(LIBADD_DL)
 else
 endif
+dltest_LDADD = $(LIBLTDL) $(LIBADD_DL)
diff -Naur unixODBC-2.2.12.orig/Makefile.am unixODBC-2.2.12/Makefile.am
--- unixODBC-2.2.12.orig/Makefile.am	2008-05-29 15:54:09.000000000 +0200
+++ unixODBC-2.2.12/Makefile.am	2008-05-29 15:57:56.000000000 +0200
@@ -3,7 +3,6 @@
     log \
 	lst \
 	ini \
-    	libltdl \
 	odbcinst \
 	DriverManager \
 	odbcinstQ \
