--- unixODBC-2.2.11/DataManager/Makefile.am	2004-09-13 16:46:11.000000000 +0200
+++ unixODBC.cvs/DataManager/Makefile.am	2005-03-09 17:39:17.000000000 +0100
@@ -8,6 +8,7 @@
 	@QT_LDFLAGS@ \
 	@USER_LDFLAGS@ \
 	@LIB_QT@ \
+	../ini/libinilc.la \
 	../odbcinst/libodbcinst.la \
 	../DriverManager/libodbc.la 
 
--- unixODBC-2.2.11/DataManagerII/Makefile.am	2004-11-20 13:57:38.000000000 +0100
+++ unixODBC.cvs/DataManagerII/Makefile.am	2005-03-09 17:39:18.000000000 +0100
@@ -8,6 +8,7 @@
 	@QT_LDFLAGS@ \
 	@USER_LDFLAGS@ \
 	@LIB_QT@ \
+	../ini/libinilc.la \
 	../odbcinst/libodbcinst.la \
 	../DriverManager/libodbc.la 
 
--- unixODBC-2.2.11/odbctest/Makefile.am	2004-09-13 16:47:34.000000000 +0200
+++ unixODBC.cvs/odbctest/Makefile.am	2005-03-09 17:39:27.000000000 +0100
@@ -8,6 +8,7 @@
 	@QT_LDFLAGS@ \
 	@USER_LDFLAGS@ \
 	@LIB_QT@ \
+	../ini/libinilc.la \
 	../odbcinst/libodbcinst.la \
 	../DriverManager/libodbc.la \
 	$(LIBLTDL) $(LIBADD_DL)
--- unixODBC-2.2.11/gODBCConfig/src/Makefile.am.orig	2005-07-21 10:03:12.051477479 +0200
+++ unixODBC-2.2.11/gODBCConfig/src/Makefile.am	2005-07-21 10:04:10.359597876 +0200
@@ -28,6 +28,7 @@
 		$(GNOMEUI_LIBS) \
 		$(INTLLIBS) \
 		-L$(ODBCLIBS) $(ODBCLIB) \
-		libgtkodbcconfig.la 
+		libgtkodbcconfig.la \
+		../../ini/.libs/libinilc.la
 		
 
