--- unixODBC-2.2.6/sqp/lex.l.orig	Wed Oct 17 18:40:32 2001
+++ unixODBC-2.2.6/sqp/lex.l	Thu Jul 10 11:23:35 2003
@@ -178,6 +178,7 @@
 
 '[^'\n]*$	{ 
 			yyerror( "Unterminated string" ); 
+			YY_FLUSH_BUFFER;
 		}
 
  /***************************************
@@ -236,8 +237,6 @@
 void yyerror( char *s )
 {
 	sprintf( g_szError, "%s processing '%s' on line %d", s, yytext, g_nLineNo );
-	
-	YY_FLUSH_BUFFER;
 }
 
 /**********************************************************************
