--- /usr/portage/media-libs/alsa-lib/alsa-lib-1.0.22.ebuild	2010-01-15 01:35:50.000000000 +0100
+++ ./media-libs/alsa-lib/alsa-lib-1.0.22.ebuild	2010-06-01 00:45:02.000000000 +0200
@@ -38,15 +38,12 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	elibtoolize
 	epunt_cxx
 }
 
-src_compile() {
+src_configure() {
 	local myconf
 	use elibc_uclibc && myconf="--without-versioned"
 
@@ -64,7 +61,9 @@
 		--disable-dependency-tracking \
 		${myconf} \
 		|| die "configure failed"
+}
 
+src_compile() {
 	emake || die "make failed"
 
 	if use doc; then
