--- /usr/portage/./media-libs/libcaca/libcaca-0.99_beta16.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ././media-libs/libcaca/libcaca-0.99_beta16.ebuild	2010-05-05 22:55:31.000000000 +0200
@@ -30,10 +30,7 @@
 
 S="${WORKDIR}/${MY_P}"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}/${PN}-0.99_beta14-deoptimise.patch"
 	epatch "${FILESDIR}/${P}-freeglut-2.6.patch"
 
@@ -41,10 +38,7 @@
 	elibtoolize
 }
 
-src_compile() {
-	# temp font fix #44128
-	export VARTEXFONTS="${T}/fonts"
-
+src_configure() {
 	econf \
 		$(use_enable doc) \
 		$(use_enable ncurses) \
@@ -56,8 +50,6 @@
 		$(use_enable mono csharp) \
 		$(use_enable ruby) \
 		|| die "econf failed"
-	emake || die "emake failed"
-	unset VARTEXFONTS
 }
 
 src_install() {
