--- /usr/portage/net-libs/openslp/openslp-1.2.1-r1.ebuild	2010-12-02 02:37:12.000000000 +0100
+++ ./net-libs/openslp/openslp-1.2.1-r1.ebuild	2010-12-25 01:30:34.000000000 +0100
@@ -17,16 +17,12 @@
 DEPEND="dev-libs/openssl"
 RDEPEND="${DEPEND}"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-fbsd.patch
 	eautoreconf
 }
 
 src_compile() {
-	econf || die
 	emake -j1 || die "make failed"
 }
 
