--- /usr/portage/net-libs/openslp/openslp-1.2.1-r1.ebuild	2010-01-22 10:08:02.000000000 +0100
+++ ./net-libs/openslp/openslp-1.2.1-r1.ebuild	2010-04-03 23:22:41.000000000 +0200
@@ -19,10 +19,7 @@
 DEPEND="dev-libs/openssl"
 RDEPEND="${DEPEND}"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-fbsd.patch
 	eautomake
 
@@ -30,7 +27,6 @@
 }
 
 src_compile() {
-	econf || die
 	emake -j1 || die "make failed"
 }
 
