--- /usr/portage/dev-libs/libxml2/libxml2-2.7.4-r1.ebuild	2010-03-08 23:36:12.000000000 +0100
+++ ./dev-libs/libxml2/libxml2-2.7.4-r1.ebuild	2010-03-28 00:52:55.000000000 +0100
@@ -44,13 +44,13 @@
 			"${S}"/xstc/ \
 			|| die "Failed to install test tarballs"
 	fi
+}
 
+src_prepare() {
 	# Fix inkscape extension loader problem, bug #285125,
 	# patch import from upstream bug #595128.
 	epatch "${FILESDIR}"/${P}-parser-grow.patch
-}
 
-src_prepare() {
 	epunt_cxx
 }
 
