--- /usr/portage/dev-libs/libxml2/libxml2-2.7.3-r2.ebuild	2010-03-08 23:36:12.000000000 +0100
+++ ./dev-libs/libxml2/libxml2-2.7.3-r2.ebuild	2010-03-28 00:47:46.000000000 +0100
@@ -42,7 +42,9 @@
 			"${S}"/xstc/ \
 			|| die "Failed to install test tarballs"
 	fi
+}
 
+src_prepare() {
 	# Fix macro conflict with wxGTK, bug #266653
 	epatch "${FILESDIR}/${P}-printf-rename.patch"
 
@@ -52,7 +54,7 @@
 	epunt_cxx
 }
 
-src_compile() {
+src_configure() {
 	# USE zlib support breaks gnome2
 	# (libgnomeprint for instance fails to compile with
 	# fresh install, and existing) - <azarah@gentoo.org> (22 Dec 2002).
@@ -89,8 +91,6 @@
 			-i ${x} \
 			|| die "sed failed"
 	done
-
-	emake || die "Compilation failed"
 }
 
 src_install() {
