--- /usr/portage/sys-fs/udev/udev-164-r1.ebuild	2010-12-12 23:12:43.000000000 +0100
+++ ./sys-fs/udev/udev-164-r1.ebuild	2010-12-25 02:05:02.000000000 +0100
@@ -145,9 +145,9 @@
 			mv "${WORKDIR}"/test/sys "${S}"/test/
 		fi
 	fi
+}
 
-	cd "${S}"
-
+src_prepare() {
 	# patches go here...
 
 	# backport some patches
@@ -192,7 +192,7 @@
 		rc/*/*
 }
 
-src_compile() {
+src_configure() {
 	filter-flags -fprefetch-loop-arrays
 
 	econf \
@@ -208,8 +208,6 @@
 		$(use_enable extras) \
 		--disable-introspection
 	# we don't have gobject-introspection in portage tree
-
-	emake || die "compiling udev failed"
 }
 
 src_install() {
