--- /usr/portage/sys-fs/udev/udev-154.ebuild	2010-08-02 23:15:45.000000000 +0200
+++ ./sys-fs/udev/udev-154.ebuild	2010-08-07 03:15:52.000000000 +0200
@@ -139,9 +139,9 @@
 			mv "${WORKDIR}"/test/sys "${S}"/test/
 		fi
 	fi
+}
 
-	cd "${S}"
-
+src_prepare() {
 	# patches go here...
 
 	# backport some patches
@@ -181,7 +181,7 @@
 	fi
 }
 
-src_compile() {
+src_configure() {
 	filter-flags -fprefetch-loop-arrays
 
 	econf \
@@ -197,8 +197,6 @@
 		$(use_enable extras) \
 		--disable-introspection
 	# we don't have gobject-introspection in portage tree
-
-	emake || die "compiling udev failed"
 }
 
 src_install() {
