--- pciutils-3.0.0/Makefile
+++ pciutils-3.0.0/Makefile
@@ -34,9 +34,11 @@
 lib/config.h lib/config.mk:
 	cd lib && ./configure
 
+pcimodules: pcimodules.o common.o lib/$(PCILIB)
 lspci: lspci.o common.o lib/$(PCILIB)
 setpci: setpci.o common.o lib/$(PCILIB)
 
+pcimodules.o: pcimodules.c pciutils.h $(PCIINC)
 lspci.o: lspci.c pciutils.h $(PCIINC)
 setpci.o: setpci.c pciutils.h $(PCIINC)
 common.o: common.c pciutils.h $(PCIINC)
