--- pciutils-3.1.0/Makefile
+++ pciutils-3.1.0/Makefile
@@ -62,6 +62,7 @@
 lib/config.h lib/config.mk:
 	cd lib && ./configure
 
+pcimodules: pcimodules.o common.o lib/$(PCILIB)
 lspci: lspci.o ls-vpd.o ls-caps.o ls-ecaps.o ls-kernel.o ls-tree.o ls-map.o common.o lib/$(PCILIB)
 setpci: setpci.o common.o lib/$(PCILIB)
 
@@ -74,6 +75,7 @@
 ls-tree.o: ls-tree.c $(LSPCIINC)
 ls-map.o: ls-map.c $(LSPCIINC)
 
+pcimodules.o: pcimodules.c pciutils.h $(PCIINC)
 setpci.o: setpci.c pciutils.h $(PCIINC)
 common.o: common.c pciutils.h $(PCIINC)
 
