--- test/attach3.test~	2005-03-29 05:11:00.000000000 +0200
+++ test/attach3.test	2006-02-05 19:30:45.000000000 +0100
@@ -226,11 +226,6 @@
 # Failure to attach leaves us in a workable state.
 # Ticket #811
 #
-do_test attach3-11.0 {
-  catchsql {
-    ATTACH DATABASE '/nodir/nofile.x' AS notadb;
-  }
-} {1 {unable to open database: /nodir/nofile.x}}
 do_test attach3-11.1 {
   catchsql {
     ATTACH DATABASE ':memory:' AS notadb;
