diff -urN sqlite-3.6.20-orig/Makefile.in sqlite-3.6.20/Makefile.in
--- sqlite-3.6.20-orig/Makefile.in	2009-11-13 20:10:50.617845091 -0600
+++ sqlite-3.6.20/Makefile.in	2009-11-13 20:10:54.994843274 -0600
@@ -150,6 +150,11 @@
 LIBTOOL = ./libtool
 ALLOWRELEASE = @ALLOWRELEASE@
 
+# FTS3
+#
+TCC += -DSQLITE_CORE
+TCC += -DSQLITE_ENABLE_FTS3=1
+
 # libtool compile/link/install
 LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(TCC) $(LTCOMPILE_EXTRAS)
 LTLINK = $(LIBTOOL) --mode=link $(TCC) $(LTCOMPILE_EXTRAS) @LDFLAGS@ $(LTLINK_EXTRAS)
@@ -187,7 +192,11 @@
 USE_AMALGAMATION = @USE_AMALGAMATION@
 LIBOBJ = $(OBJS$(USE_AMALGAMATION))
 
-
+# Object files for FTS
+#
+LIBOBJ += \
+	fts3.lo fts3_expr.lo fts3_hash.lo fts3_porter.lo fts3_tokenizer.lo fts3_tokenizer1.lo
+ 
 # All of the source code files.
 #
 SRC = \
@@ -832,3 +841,24 @@
 sqlite3.dll: $(REAL_LIBOBJ) sqlite3.def
 	$(TCC) -shared -o $@ sqlite3.def \
 		-Wl,"--strip-all" $(REAL_LIBOBJ)
+
+#
+# FTS3
+#
+fts3.lo: $(TOP)/ext/fts3/fts3.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3.c
+ 
+fts3_expr.lo: $(TOP)/ext/fts3/fts3_expr.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_expr.c
+ 
+fts3_hash.lo: $(TOP)/ext/fts3/fts3_hash.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_hash.c
+ 
+fts3_porter.lo: $(TOP)/ext/fts3/fts3_porter.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_porter.c
+ 
+fts3_tokenizer.lo: $(TOP)/ext/fts3/fts3_tokenizer.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_tokenizer.c
+ 
+fts3_tokenizer1.lo: $(TOP)/ext/fts3/fts3_tokenizer1.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_tokenizer1.c
diff -urN sqlite-3.6.20-orig/src/main.c sqlite-3.6.20/src/main.c
--- sqlite-3.6.20-orig/src/main.c	2009-11-13 20:10:50.674843345 -0600
+++ sqlite-3.6.20/src/main.c	2009-11-13 20:11:30.403822462 -0600
@@ -17,7 +17,7 @@
 #include "sqliteInt.h"
 
 #ifdef SQLITE_ENABLE_FTS3
-# include "fts3.h"
+# include "../ext/fts3/fts3.h"
 #endif
 #ifdef SQLITE_ENABLE_RTREE
 # include "rtree.h"
