--- ./Makefile.in.orig	2009-03-30 17:37:16.000000000 +0100
+++ ./Makefile.in	2009-04-09 11:49:05.151436460 +0100
@@ -776,7 +776,10 @@
 	$(INSTALL) -d $(DESTDIR)$(libdir)
 	$(LTINSTALL) libsqlite3.la $(DESTDIR)$(libdir)
 	
-install:	sqlite3$(BEXE) lib_install sqlite3.h sqlite3.pc ${HAVE_TCL:1=tcl_install}
+# libsqlite3.la must be installed before anything that depends on it,
+# including the tcl_install items.
+
+install:	sqlite3$(BEXE) libsqlite3.la sqlite3.h sqlite3.pc
 	$(INSTALL) -d $(DESTDIR)$(bindir)
 	$(LTINSTALL) sqlite3$(BEXE) $(DESTDIR)$(bindir)
 	$(INSTALL) -d $(DESTDIR)$(includedir)
@@ -784,6 +787,7 @@
 	$(INSTALL) -m 0644 $(TOP)/src/sqlite3ext.h $(DESTDIR)$(includedir)
 	$(INSTALL) -d $(DESTDIR)$(pkgconfigdir)
 	$(INSTALL) -m 0644 sqlite3.pc $(DESTDIR)$(pkgconfigdir)
+	if [ "$(HAVE_TCL)" = 1 ]; then $(MAKE) tcl_install; fi
 
 pkgIndex.tcl:
 	echo 'package ifneeded sqlite3 $(RELEASE) [list load $(TCLLIBDIR)/libtclsqlite3.so sqlite3]' > $@
