--- sqlite3-3.6.11.orig/Makefile.in	2009-03-24 10:52:35.000000000 -0700
+++ sqlite3-3.6.11/Makefile.in	2009-03-24 11:07:05.000000000 -0700
@@ -765,7 +765,10 @@
 		$(TEMP_STORE) -o $@ $(TESTFIXTURE_SRC) $(LIBTCL)
 
 
-install:	sqlite3$(BEXE) libsqlite3.la sqlite3.h sqlite3.pc ${HAVE_TCL:1=tcl_install}
+# libsqlite3.la must be installed before anything that depends on it,
+# including the tcl_install items.
+
+install:	sqlite3$(BEXE) libsqlite3.la sqlite3.h sqlite3.pc
 	$(INSTALL) -d $(DESTDIR)$(libdir)
 	$(LTINSTALL) libsqlite3.la $(DESTDIR)$(libdir)
 	$(INSTALL) -d $(DESTDIR)$(bindir)
@@ -775,6 +778,7 @@
 	$(INSTALL) -m 0644 $(TOP)/src/sqlite3ext.h $(DESTDIR)$(includedir)
 	$(INSTALL) -d $(DESTDIR)$(pkgconfigdir)
 	$(INSTALL) -m 0644 sqlite3.pc $(DESTDIR)$(pkgconfigdir)
+	if [ "$(HAVE_TCL)" = 1 ]; then $(MAKE) tcl_install; fi
 
 pkgIndex.tcl:
 	echo 'package ifneeded sqlite3 $(RELEASE) [list load $(TCLLIBDIR)/libtclsqlite3.so sqlite3]' > $@
