diff -up epiphany-2.20.3/embed/mozilla/EphyBrowser.h.gcc43~ epiphany-2.20.3/embed/mozilla/EphyBrowser.h
--- epiphany-2.20.3/embed/mozilla/EphyBrowser.h.gcc43~	2007-08-28 05:19:02.000000000 +0900
+++ epiphany-2.20.3/embed/mozilla/EphyBrowser.h	2008-02-17 14:42:42.000000000 +0900
@@ -187,7 +187,7 @@ public:
 
 	nsresult LoadURI(const char *aURI, 
 			 PRUint32 aLoadFlags = nsIWebNavigation::LOAD_FLAGS_NONE, 
-			 nsIURI *aURI = nsnull);
+			 nsIURI *aReferrer = nsnull);
 
         EphyEmbedDocumentType GetDocumentType ();
 
diff -up epiphany-2.20.3/embed/mozilla/mozilla-x509-cert.h.gcc43~ epiphany-2.20.3/embed/mozilla/mozilla-x509-cert.h
--- epiphany-2.20.3/embed/mozilla/mozilla-x509-cert.h.gcc43~	2007-08-28 05:19:02.000000000 +0900
+++ epiphany-2.20.3/embed/mozilla/mozilla-x509-cert.h	2008-02-17 14:45:10.000000000 +0900
@@ -56,7 +56,7 @@ GType	                   mozilla_x509_ce
 
 MozillaX509Cert           *mozilla_x509_cert_new              (nsIX509Cert *aMozCert);
 
-nsresult                   mozilla_x509_cert_get_mozilla_cert (MozillaX509Cert *cert, nsIX509Cert **cert);
+nsresult                   mozilla_x509_cert_get_mozilla_cert (MozillaX509Cert *cert, nsIX509Cert **aCert);
 
 G_END_DECLS
 
