--- /usr/portage/media-libs/speex/speex-1.2_beta3_p2.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./media-libs/speex/speex-1.2_beta3_p2.ebuild	2010-04-29 21:15:17.000000000 +0200
@@ -20,9 +20,7 @@
 
 S=${WORKDIR}/${MY_P}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.2_beta3-configure.patch
 
 	sed -i -e 's:noinst_PROGRAMS:check_PROGRAMS:' \
@@ -31,11 +29,10 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	append-flags -D_FILE_OFFSET_BITS=64
 
 	econf $(use_enable sse) $(use_enable ogg)
-	emake || die "emake failed."
 }
 
 src_install() {
