--- FTGL/include/FTTextureGlyph.h.old	2006-01-29 11:08:54.000000000 -0700
+++ FTGL/include/FTTextureGlyph.h	2006-01-29 11:09:06.000000000 -0700
@@ -52,7 +52,7 @@
          * Reset the currently active texture to zero to get into a known state before
          * drawing a string. This is to get round possible threading issues.
          */
-        static void FTTextureGlyph::ResetActiveTexture(){ activeTextureID = 0;}
+        static void ResetActiveTexture(){ activeTextureID = 0;}
         
     private:
         /**
