--- configure.old    2008-10-31 16:25:39.000000000 -0430
+++ configure        2008-10-31 16:24:39.000000000 -0430
@@ -6526,7 +6526,7 @@
   # Find out which ABI we are using (multilib Linux x86_64 hack).
   libsuff=
   case "$host_cpu" in
-  x86_64*)
+  x86_64* | i*86)
     echo '#line 6525 "configure"' > conftest.$ac_ext
     if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
@@ -6537,6 +6537,9 @@
       *64-bit*)
         libsuff=64
         ;;
+      *32-bit*)
+        libsuff=32
+        ;;
       esac
     fi
     rm -rf conftest* 
