--- pycairo-1.2.2.orig/configure.ac	2006-08-28 22:12:10.000000000 +0200
+++ pycairo-1.2.2/configure.ac	2006-08-28 22:13:38.000000000 +0200
@@ -32,6 +32,9 @@
             [AC_HELP_STRING([--with-pygtk], [support pygtk [default=yes]])],
 	    [],
 	    [with_pygtk="yes"])
+AC_ARG_WITH([numeric],
+            AS_HELP_STRING([--without-numeric],
+                           [Build without numeric support (default: test)]))
 
 
 # put the ACLOCAL flags in the makefile
@@ -79,14 +82,19 @@
 AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
 
 # Numeric Python
-save_CPPFLAGS="$CPPFLAGS"
-CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
-AC_CHECK_HEADER([Numeric/arrayobject.h],
-  [have_numpy=yes], [have_numpy=no],
-  [#include <Python.h>])
-CPPFLAGS="$save_CPPFLAGS"
-if test x$have_numpy = xyes; then
-  AC_DEFINE(HAVE_NUMPY, [1], [Define to 1 if Numeric python is found])
+if test "x$with_numeric" != "xno"; then
+  save_CPPFLAGS="$CPPFLAGS"
+  CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
+  AC_CHECK_HEADER([Numeric/arrayobject.h],
+    [have_numpy=yes], [have_numpy=no],
+    [#include <Python.h>])
+  CPPFLAGS="$save_CPPFLAGS"
+  if test x$have_numpy = xyes; then
+    AC_DEFINE(HAVE_NUMPY, [1], [Define to 1 if Numeric python is found])
+  fi
+else
+  # only needed for the summary printed at the end of configure
+  have_numpy=no
 fi
 
 # checks for types and compilers ---
