diff -ur xine-lib-1.1.16.3.orig/configure.ac xine-lib-1.1.16.3/configure.ac
--- xine-lib-1.1.16.3.orig/configure.ac	2009-04-02 21:44:45.000000000 +0300
+++ xine-lib-1.1.16.3/configure.ac	2009-07-26 15:34:10.000000000 +0300
@@ -1341,8 +1341,8 @@
 if test "x$enable_musepack" = "xno"; then
    AC_MSG_RESULT([musepack support disabled])
 elif test "x$with_external_libmpcdec" = "xyes"; then
-   AC_CHECK_LIB([mpcdec], [mpc_decoder_decode], [have_mpcdec=yes])
-   AC_CHECK_HEADERS([mpcdec/mpcdec.h], , [have_mpcdec=no])
+   AC_CHECK_LIB([mpcdecsv7], [mpc_decoder_decode], [have_mpcdec=yes])
+   AC_CHECK_HEADERS([mpcdecsv7/mpcdec.h], , [have_mpcdec=no])
    if test "x$have_mpcdec" != "xyes"; then
       AC_MSG_ERROR([Unable to find mpcdec])
    fi
diff -ur xine-lib-1.1.16.3.orig/src/libmusepack/Makefile.am xine-lib-1.1.16.3/src/libmusepack/Makefile.am
--- xine-lib-1.1.16.3.orig/src/libmusepack/Makefile.am	2009-02-10 19:32:23.000000000 +0200
+++ xine-lib-1.1.16.3/src/libmusepack/Makefile.am	2009-07-26 15:34:27.000000000 +0300
@@ -17,7 +17,7 @@
 endif
 
 if EXTERNAL_MPCDEC
-xineplug_decode_mpc_la_LIBADD = $(XINE_LIB) $(LTLIBINTL) -lmpcdec
+xineplug_decode_mpc_la_LIBADD = $(XINE_LIB) $(LTLIBINTL) -lmpcdecsv7
 else
 xineplug_decode_mpc_la_LIBADD = $(XINE_LIB) $(LTLIBINTL)
 endif
diff -ur xine-lib-1.1.16.3.orig/src/libmusepack/xine_musepack_decoder.c xine-lib-1.1.16.3/src/libmusepack/xine_musepack_decoder.c
--- xine-lib-1.1.16.3.orig/src/libmusepack/xine_musepack_decoder.c	2009-02-10 19:32:23.000000000 +0200
+++ xine-lib-1.1.16.3/src/libmusepack/xine_musepack_decoder.c	2009-07-26 15:34:49.000000000 +0300
@@ -46,7 +46,7 @@
 #include "xineutils.h"
 
 #ifdef HAVE_MPCDEC_MPCDEC_H
-# include <mpcdec/mpcdec.h>
+# include <mpcdecsv7/mpcdec.h>
 #else
 # include "musepack/musepack.h"
 #endif
