--- /usr/portage/app-text/djvu/djvu-3.5.21_p20090103.ebuild	2010-01-05 04:38:14.000000000 +0100
+++ ./app-text/djvu/djvu-3.5.21_p20090103.ebuild	2010-06-06 02:59:36.000000000 +0200
@@ -27,7 +27,7 @@
 	IUSE="${IUSE} linguas_${X}"
 done
 
-src_compile() {
+src_configure() {
 	local I18N
 	if use nls; then
 		for X in ${LANGS}; do
@@ -55,8 +55,6 @@
 		$(use_enable debug)
 
 	sed -e 's:nsdejavu::' -i "${S}"/gui/Makefile || die
-
-	emake || die "emake failed"
 }
 
 src_install() {
