--- /usr/portage/dev-dotnet/libgdiplus/libgdiplus-1.2.6.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./dev-dotnet/libgdiplus/libgdiplus-1.2.6.ebuild	2010-06-06 02:10:52.000000000 +0200
@@ -29,17 +29,14 @@
 
 RESTRICT="test"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	#epatch ${FILESDIR}/${PN}-1.1.13-libungif-configure-fix.diff
 	epatch "${FILESDIR}/${PN}-1.2.5-imglibs.patch"
 	epatch "${FILESDIR}/${PN}-1.2.6-autotool-variables.patch"
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	if [[ "$(gcc-major-version)" -gt "3" ]] || \
 	   ( [[ "$(gcc-major-version)" -eq "3" ]] && [[ "$(gcc-minor-version)" -gt "3" ]] )
 	then
@@ -57,8 +54,6 @@
 	if use ppc ; then
 		sed -i -e 's:-Werror::g' src/Makefile
 	fi
-
-	emake || die "compile failed"
 }
 
 src_install() {
