Index: drivers/lirc_dev/lirc_dev.c
===================================================================
RCS file: /cvsroot/lirc/lirc/drivers/lirc_dev/lirc_dev.c,v
retrieving revision 1.58
diff -u -p -u -p -r1.58 lirc_dev.c
--- drivers/lirc_dev/lirc_dev.c 14 May 2008 16:37:49 -0000 1.58
+++ drivers/lirc_dev/lirc_dev.c 4 Aug 2008 14:54:04 -0000
@@ -49,7 +49,11 @@
 #include <linux/uaccess.h>
 #include <linux/errno.h>
 #endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 27)
 #include <asm/semaphore.h>
+#else
+#include <linux/semaphore.h>
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
 #include <linux/wrapper.h>
 #endif
Index: drivers/lirc_i2c/lirc_i2c.c
===================================================================
RCS file: /cvsroot/lirc/lirc/drivers/lirc_i2c/lirc_i2c.c,v
retrieving revision 1.46
diff -u -p -u -p -r1.46 lirc_i2c.c
--- drivers/lirc_i2c/lirc_i2c.c 4 May 2008 13:49:53 -0000 1.46
+++ drivers/lirc_i2c/lirc_i2c.c 4 Aug 2008 14:54:04 -0000
@@ -54,7 +54,11 @@
 #include <linux/i2c.h>
 #include <linux/i2c-algo-bit.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 27)
 #include <asm/semaphore.h>
+#else
+#include <linux/semaphore.h>
+#endif
 
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h" 
