--- /usr/portage/media-video/vcdimager/vcdimager-0.7.23.ebuild	2009-04-05 10:08:24.000000000 +0200
+++ ./media-video/vcdimager/vcdimager-0.7.23.ebuild	2010-04-04 14:27:31.000000000 +0200
@@ -20,7 +20,7 @@
 DEPEND="${RDEPEND}
 	dev-util/pkgconfig"
 
-src_compile() {
+src_configure() {
 	local myconf
 
 	# We disable the xmltest because the configure script includes differently
@@ -32,8 +32,6 @@
 		$(use_with !minimal cli-frontends) \
 		${myconf} \
 		--disable-dependency-tracking || die "configure failed"
-
-	emake || die "make failed"
 }
 
 src_install() {
