Index: libtool-2.2.6/tests/localization.at
===================================================================
--- libtool-2.2.6.orig/tests/localization.at
+++ libtool-2.2.6/tests/localization.at
@@ -47,6 +47,7 @@ mv -f stdout expected-stdout
 mv -f stderr expected-stderr
 AT_CHECK([$LIBTOOL --mode=compile --tag=CC $CC $CPPFLAGS $CFLAGS -c a.c || exit 1],
 	 [1], [stdout], [stderr])
+AT_CHECK([grep LC_ALL stdout stderr && exit 77], [1])
 AT_CHECK([diff expected-stderr stderr])
 LTBASE=`$ECHO "$LIBTOOL" | sed 's,^.*/,,'`
 AT_CHECK([grep -v "^$LTBASE: compile" stdout | diff expected-stdout -])
