diff -Naur tk8.4.9.orig/library/text.tcl tk8.4.9/library/text.tcl
--- tk8.4.9.orig/library/text.tcl	2004-12-07 04:42:11.000000000 +0900
+++ tk8.4.9/library/text.tcl	2005-03-02 23:04:01.729780958 +0900
@@ -347,14 +347,6 @@
     catch { %W edit redo }
 }
 
-if {$tcl_platform(platform) ne "windows"} {
-bind Text <Control-v> {
-    if {!$tk_strictMotif} {
-	tk::TextScrollPages %W 1
-    }
-}
-}
-
 bind Text <Meta-b> {
     if {!$tk_strictMotif} {
 	tk::TextSetCursor %W [tk::TextPrevPos %W insert tcl_startOfPreviousWord]
