--- unix/Makefile.in
+++ unix/Makefile.in
@@ -280,7 +280,7 @@
 TK_RSRC_FILE		= @TK_RSRC_FILE@
 WISH_RSRC_FILE		= @WISH_RSRC_FILE@
 REZ			= @REZ@
-REZ_SWITCHES = @REZ_FLAGS@ -i $(GENERIC_DIR) -i $(TCL_GENERIC_DIR)
+REZ_SWITCHES = @REZ_FLAGS@ -i $(GENERIC_DIR) -i $(TCL_GENERIC_DIR) -i @GENTOO_PORTAGE_EPREFIX@/usr/include
 
 #----------------------------------------------------------------
 # The information below is modified by the configure script when
@@ -468,7 +468,7 @@
 
 AQUA_HDRS = $(MAC_OSX_DIR)/tkMacOSX.h $(GENERIC_DIR)/tkIntXlibDecls.h
 
-AQUA_XLIB_HDRS = $(XLIB_DIR)/X11/*.h $(XLIB_DIR)/xbytes.h
+AQUA_XLIB_HDRS = 
 
 AQUA_PRIVATE_HDRS = $(MAC_OSX_DIR)/tkMacOSXPort.h $(MAC_OSX_DIR)/tkMacOSXInt.h
 
