--- /usr/portage/dev-libs/DirectFB/DirectFB-1.4.3.ebuild	2010-10-25 21:35:41.000000000 +0200
+++ ./dev-libs/DirectFB/DirectFB-1.4.3.ebuild	2010-12-25 00:44:06.000000000 +0100
@@ -44,9 +44,7 @@
 DEPEND="${RDEPEND}
 	X? ( x11-proto/xextproto x11-proto/xproto )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.2.7-CFLAGS.patch
 	epatch "${FILESDIR}"/${PN}-1.2.0-headers.patch
 	epatch "${FILESDIR}"/${PN}-1.1.1-pkgconfig.patch
@@ -83,7 +81,7 @@
 	echo ${devs:-none}
 }
 
-src_compile() {
+src_configure() {
 	local sdlconf="--disable-sdl"
 	if use sdl ; then
 		# since SDL can link against DirectFB and trigger a
@@ -115,7 +113,6 @@
 		--with-inputdrivers="$(driver_list input_devices ${IUSE_INPUT_DEVICES})" \
 		--disable-vnc \
 		|| die
-	emake || die
 }
 
 src_install() {
