--- /usr/portage/dev-libs/DirectFB/DirectFB-1.4.2.ebuild	2010-09-13 21:06:28.000000000 +0200
+++ ./dev-libs/DirectFB/DirectFB-1.4.2.ebuild	2010-09-18 12:18:56.000000000 +0200
@@ -44,9 +44,7 @@
 DEPEND="${RDEPEND}
 	X? ( x11-proto/xextproto x11-proto/xproto )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.2.7-CFLAGS.patch
 	epatch "${FILESDIR}"/${PN}-1.2.0-headers.patch
 	epatch "${FILESDIR}"/${PN}-1.1.1-pkgconfig.patch
@@ -77,7 +75,7 @@
 	echo ${devs:-none}
 }
 
-src_compile() {
+src_configure() {
 	local sdlconf="--disable-sdl"
 	if use sdl ; then
 		# since SDL can link against DirectFB and trigger a
@@ -109,7 +107,6 @@
 		--with-inputdrivers="$(driver_list input_devices ${IUSE_INPUT_DEVICES})" \
 		--disable-vnc \
 		|| die
-	emake || die
 }
 
 src_install() {
