--- src/Makefile.am.orig	2009-02-02 18:46:49.000000000 +0000
+++ src/Makefile.am	2009-02-02 18:48:09.000000000 +0000
@@ -97,7 +97,8 @@
 $(BUILT_HEADER): $(BUILT_SOURCE)
 
 $(BUILT_SOURCE):
-	$(top_builddir)/util/makestrs -i $(top_srcdir) $(STRINGSABIOPTIONS) < $(STRING_LIST) > StringDefs.c
+	@# run the makestrs we just built and fall back to host makestrs for cross-compiles when native one fails
+	$(top_builddir)/util/makestrs -i $(top_srcdir) $(STRINGSABIOPTIONS) < $(STRING_LIST) > StringDefs.c || makestrs -i $(top_srcdir) $(STRINGSABIOPTIONS) < $(STRING_LIST) > StringDefs.c
 	mkdir -p $(top_builddir)/src
 	mkdir -p $(top_builddir)/include/X11
 	cp StringDefs.h $(top_builddir)/include/X11
