Index: libmodplug-0.7/src/Makefile.am
===================================================================
--- libmodplug-0.7.orig/src/Makefile.am
+++ libmodplug-0.7/src/Makefile.am
@@ -1,5 +1,6 @@
 INCLUDES = -Ilibmodplug
 lib_LTLIBRARIES = libmodplug.la
+libmodplug_la_LIBADD = -lm
 libmodplug_la_SOURCES = tables.cpp         \
                         sndmix.cpp         \
                         sndfile.cpp        \
