diff -uNr libmodplug-0.8.4.ORIG/src/libmodplug/stdafx.h libmodplug-0.8.4/src/libmodplug/stdafx.h
--- libmodplug-0.8.4.ORIG/src/libmodplug/stdafx.h	2006-12-23 00:40:42.000000000 +0000
+++ libmodplug-0.8.4/src/libmodplug/stdafx.h	2006-12-23 00:42:45.000000000 +0000
@@ -21,7 +21,7 @@
 
 inline void ProcessPlugins(int n) {}
 
-#elif defined(__x86_64__)
+#elif defined(__x86_64__) || defined(__powerpc64__)
 
 #include <stdlib.h>
 #include <stdio.h>
