https://bugs.icu-project.org/trac/ticket/7039

--- common/unicode/unistr.h
+++ common/unicode/unistr.h
@@ -4048,7 +4048,8 @@
 
 {
   // This dstSize value will be checked explicitly
-  return extract(start, _length, dst, dst!=0 ? 0xffffffff : 0, codepage);
+  // Ticket #7039: Clip length to the maximum valid length to the end of addressable memory given the starting address
+  return extract(start, _length, dst, dst!=0 ? ((dst >= (char*)((size_t)-1) - UINT32_MAX) ? (((char*)UINT32_MAX) - dst) : UINT32_MAX) : 0, codepage);
 }
 
 #endif
--- runConfigureICU
+++ runConfigureICU
@@ -318,7 +318,7 @@
     then
         case $CXX in
             g++|*/g++|*-g++-*|*/*-g++-*)
-                RELEASE_CXXFLAGS=-O
+                RELEASE_CXXFLAGS=-O3
                 ;;
         esac
     fi
