https://bugs.gentoo.org/show_bug.cgi?id=321343

diff -ur jack-audio-connection-kit-0.118.0.orig/libjack/client.c jack-audio-connection-kit-0.118.0/libjack/client.c
--- jack-audio-connection-kit-0.118.0.orig/libjack/client.c	2010-05-27 16:52:06.000000000 +0300
+++ jack-audio-connection-kit-0.118.0/libjack/client.c	2010-05-27 16:52:40.000000000 +0300
@@ -1469,7 +1469,7 @@
 {
 	jack_client_control_t *control = client->control;
 
-	DEBUG ("client polling on %s", client->pollmax == 2 ? x
+	DEBUG ("client polling on %s", client->pollmax == 2 ?
 	       "event_fd and graph_wait_fd..." :
 	       "event_fd only");
 	
