diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN esound-0.2.38.orig/Makefile.am esound-0.2.38/Makefile.am
--- esound-0.2.38.orig/Makefile.am	2007-04-19 10:43:59.000000000 -0400
+++ esound-0.2.38/Makefile.am	2007-05-30 11:03:00.000000000 -0400
@@ -61,7 +61,7 @@ AUDIO_BACKENDS = \
 	audio_win32.c
 
 libesddsp_la_LDFLAGS = -version-info $(ESD_VERSION_INFO)
-libesddsp_la_LIBADD = $(DL_LIB) -lm libesd.la
+libesddsp_la_LIBADD = libesd.la $(DL_LIB) -lm
 
 libesddsp_la_SOURCES = \
 	esddsp.c
