--- a/clients.c	2003-07-05 10:32:07.000000000 -0400
+++ b/clients.c	2004-02-03 23:34:29.000000000 -0500
@@ -155,7 +155,7 @@
       if ( esd_use_ipv6 ) {
 	char addrbuf[INET6_ADDRSTRLEN];
 
-	fd = accept( listen,(struct sockaddr *)&incoming6, &size_in6 );
+	fd = accept( listen,(struct sockaddr *)&incoming6, (socklen_t *) &size_in6 );
 	if ( fd < 0 )
                 goto again;
 	port = ntohs( incoming6.sin6_port );
@@ -168,7 +168,7 @@
       else
 #endif
       {
-	fd = accept( listen, (struct sockaddr*) &incoming, &size_in );
+	fd = accept( listen, (struct sockaddr*) &incoming, (socklen_t *) &size_in );
 	if ( fd < 0 )
 		goto again;
 	    port = ntohs( incoming.sin_port );
