--- /usr/portage/www-client/firefox/firefox-3.6.8.ebuild	2010-08-11 01:18:31.000000000 +0200
+++ ./www-client/firefox/firefox-3.6.8.ebuild	2010-08-11 22:33:10.000000000 +0200
@@ -120,11 +120,13 @@
 src_unpack() {
 	unpack firefox-${MY_PV}.source.tar.bz2 ${PATCH}.tar.bz2
 
-	linguas
-	for X in ${linguas}; do
-		# FIXME: Add support for unpacking xpis to portage
-		[[ ${X} != "en" ]] && xpi_unpack "mozilla-${P}-${X}.xpi"
-	done
+	if is_final_abi; then
+		linguas
+		for X in ${linguas}; do
+			# FIXME: Add support for unpacking xpis to portage
+			[[ ${X} != "en" ]] && xpi_unpack "mozilla-${P}-${X}.xpi"
+		done
+	fi
 }
 
 src_prepare() {
