--- /usr/portage/www-client/firefox/firefox-3.6.11.ebuild	2010-10-13 02:38:02.000000000 +0200
+++ ./www-client/firefox/firefox-3.6.11.ebuild	2010-12-25 02:18:41.000000000 +0100
@@ -120,11 +120,13 @@
 src_unpack() {
 	unpack firefox-${MY_PV}.source.tar.bz2 ${PATCH}.tar.bz2
 
-	linguas
-	for X in ${linguas}; do
-		# FIXME: Add support for unpacking xpis to portage
-		[[ ${X} != "en" ]] && xpi_unpack "${P}-${X}.xpi"
-	done
+	if is_final_abi; then
+		linguas
+		for X in ${linguas}; do
+			# FIXME: Add support for unpacking xpis to portage
+			[[ ${X} != "en" ]] && xpi_unpack "${P}-${X}.xpi"
+		done
+	fi
 }
 
 src_prepare() {
