 configure.ac |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index e804538..1e96988 100644
--- a/configure.ac
+++ b/configure.ac
@@ -68,8 +68,11 @@ for flag in $DESIRED_FLAGS ; do
 done
 
 #### libtool stuff ####
-LT_PREREQ(2.2)
-LT_INIT([dlopen win32-dll disable-static])
+AC_LIBTOOL_DLOPEN
+AC_LIBTOOL_WIN32_DLL
+AC_DISABLE_STATIC
+AM_DISABLE_STATIC
+AC_PROG_LIBTOOL
 
 dnl Unfortunately, even up to libtool 2.2.6a there is no way to know
 dnl exactly which version of libltdl is present in the system, so we
@@ -89,7 +92,7 @@ dnl can give the proper place to find libltdl through the standard
 dnl variables like LDFLAGS and CPPFLAGS.
 
 AC_CHECK_HEADER([ltdl.h],
-    [AC_CHECK_LIB([ltdl], [lt_dladvise_init], [LIBLTDL=-lltdl], [LIBLTDL=])],
+    [AC_CHECK_LIB([ltdl], [lt_dlopen], [LIBLTDL=-lltdl], [LIBLTDL=])],
     [LIBLTDL=])
 
 AS_IF([test "x$LIBLTDL" = "x"],
