--- libssh2-0.12/src/packet.c	2006-01-30 18:27:07.000000000 +0100
+++ libssh2-0.12-ced/src/packet.c	2006-01-30 17:56:18.000000000 +0100
@@ -515,6 +515,21 @@
 					LIBSSH2_FREE(session, data);
 					return 0;
 				}
+				{
+				  unsigned char packet[1];
+				  
+				  *packet = SSH_MSG_CHANNEL_FAILURE;
+
+				  if (libssh2_packet_write(session, packet, 1)) {
+				    libssh2_error(session, LIBSSH2_ERROR_SOCKET_SEND, "Unable to send channel failure", 0);
+				  }
+				 
+#ifdef LIBSSH2_DEBUG_CONNECTION
+				  _libssh2_debug(session, LIBSSH2_DBG_CONN, "Channel request failure");
+#endif
+				  LIBSSH2_FREE(session, data);
+				  return 0;
+				}
 			}
 			break;
 		case SSH_MSG_CHANNEL_CLOSE:
