--- ./glib.h.orig	2001-02-27 03:44:38.000000000 +0000
+++ glib.h	2009-04-07 11:43:43.004660013 +0100
@@ -204,7 +204,12 @@
 #ifndef G_INLINE_FUNC
 #  ifdef __GNUC__
 #    ifdef __OPTIMIZE__
-#      define G_INLINE_FUNC extern inline
+#  if __GNUC_PREREQ (4,2) && defined (__STDC_VERSION__) \
+   && __STDC_VERSION__ >= 199901L
+#    define G_INLINE_FUNC extern __inline __attribute__ ((__gnu_inline__))
+#  else
+#    define G_INLINE_FUNC extern __inline
+#  endif
 #    else
 #      undef G_CAN_INLINE
 #      define G_INLINE_FUNC extern
