--- /usr/portage/app-text/opensp/opensp-1.5.1.ebuild	2009-04-05 12:37:41.000000000 +0200
+++ ./app-text/opensp/opensp-1.5.1.ebuild	2010-06-06 02:44:58.000000000 +0200
@@ -17,15 +17,12 @@
 
 DEPEND="nls? ( sys-devel/gettext )"
 
-src_unpack() {
-	unpack "${A}"
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.5-gcc34.patch
 	epatch "${FILESDIR}"/opensp-1.5.1-gcc41.patch
 }
 
-src_compile() {
+src_configure() {
 	#
 	# The following filters are taken from openjade's ebuild. See bug #100828.
 	#
@@ -45,6 +42,9 @@
 	myconf="${myconf} --enable-default-search-path=/usr/share/sgml"
 	myconf="${myconf} --datadir=/usr/share/sgml/${P}"
 	econf ${myconf} $(use_enable nls) || die "econf failed"
+}
+
+src_compile() {
 	emake pkgdocdir=/usr/share/doc/${PF} || die "parallel make failed"
 }
 
