--- /usr/portage/app-text/opensp/opensp-1.5.2-r1.ebuild	2009-04-05 12:37:41.000000000 +0200
+++ ./app-text/opensp/opensp-1.5.2-r1.ebuild	2010-06-06 02:45:48.000000000 +0200
@@ -22,13 +22,11 @@
 	)"
 RDEPEND=""
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.5-gcc34.patch
 }
 
-src_compile() {
+src_configure() {
 	#
 	# The following filters are taken from openjade's ebuild. See bug #100828.
 	#
@@ -51,6 +49,9 @@
 		$(use_enable doc doc-build)"
 
 	econf ${myconf} || die "econf failed"
+}
+
+src_compile() {
 	emake pkgdocdir=/usr/share/doc/${PF} || die "Compilation failed"
 }
 
