--- /usr/portage/net-libs/xulrunner/xulrunner-1.9.2.13-r1.ebuild	2011-01-02 23:28:37.000000000 +0100
+++ ./net-libs/xulrunner/xulrunner-1.9.2.13-r1.ebuild	2011-01-06 17:22:20.000000000 +0100
@@ -214,7 +214,7 @@
 
 	# env.d file for ld search path
 	dodir /etc/env.d
-	echo "LDPATH=${EPREFIX}/${MOZLIBDIR}" > "${ED}"/etc/env.d/08xulrunner || die "env.d failed"
+	echo "LDPATH=${EPREFIX}/${MOZLIBDIR}" > "${ED}"/etc/env.d/08xulrunner-${ABI} || die "env.d failed"
 
 	# Add our defaults to xulrunner and out of firefox
 	cp "${FILESDIR}"/xulrunner-default-prefs.js \
@@ -229,6 +229,11 @@
 		java-pkg_regjar "${ED}/${SDKDIR}/lib/MozillaGlue.jar"
 		java-pkg_regjar "${ED}/${SDKDIR}/lib/MozillaInterfaces.jar"
 	fi
+
+	# each ABI should generate exactly one /etc/gre.d/*.system.conf file
+	for conf in "${D}"/etc/gre.d/*.system.conf ; do
+		mv "${conf}" "${conf%.conf}-${ABI}.conf"
+	done
 }
 
 pkg_postinst() {
