--- /usr/portage/net-libs/xulrunner/xulrunner-1.9.2.8.ebuild	2010-07-26 21:36:40.000000000 +0200
+++ ./net-libs/xulrunner/xulrunner-1.9.2.8.ebuild	2010-07-27 22:16:17.000000000 +0200
@@ -205,7 +205,7 @@
 
 	# env.d file for ld search path
 	dodir /etc/env.d
-	echo "LDPATH=${EPREFIX}/${MOZLIBDIR}" > "${ED}"/etc/env.d/08xulrunner || die "env.d failed"
+	echo "LDPATH=${EPREFIX}/${MOZLIBDIR}" > "${ED}"/etc/env.d/08xulrunner-${ABI} || die "env.d failed"
 
 	# Add our defaults to xulrunner and out of firefox
 	cp "${FILESDIR}"/xulrunner-default-prefs.js \
@@ -218,20 +218,27 @@
 		java-pkg_regjar "${ED}/${SDKDIR}/lib/MozillaGlue.jar"
 		java-pkg_regjar "${ED}/${SDKDIR}/lib/MozillaInterfaces.jar"
 	fi
+
+	# each ABI should generate exactly one /etc/gre.d/*.system.conf file
+	for conf in "${D}"/etc/gre.d/*.system.conf ; do
+		mv "${conf}" "${conf%.conf}-${ABI}.conf"
+	done
 }
 
 pkg_postinst() {
-	ewarn "If firefox fails to start with \"failed to load xpcom\", run revdep-rebuild"
-	ewarn "If that does not fix the problem, rebuild dev-libs/nss"
-	ewarn "Try dev-util/lafilefixer if you get build failures related to .la files"
-
-	einfo
-	einfo "All prefs can be overridden by the user. The preferences are to make"
-	einfo "use of xulrunner out of the box on an average system without the user"
-	einfo "having to go through and enable the basics."
-
-	einfo
-	ewarn "Any package that requires xulrunner:1.9 slot could and most likely will"
-	ewarn "have issues. These issues should be reported to maintainer, and mozilla herd"
-	ewarn "should be cc'd on the bug report. Thank you anarchy@gentoo.org ."
+	if is_final_abi ; then
+		ewarn "If firefox fails to start with \"failed to load xpcom\", run revdep-rebuild"
+		ewarn "If that does not fix the problem, rebuild dev-libs/nss"
+		ewarn "Try dev-util/lafilefixer if you get build failures related to .la files"
+
+		einfo
+		einfo "All prefs can be overridden by the user. The preferences are to make"
+		einfo "use of xulrunner out of the box on an average system without the user"
+		einfo "having to go through and enable the basics."
+
+		einfo
+		ewarn "Any package that requires xulrunner:1.9 slot could and most likely will"
+		ewarn "have issues. These issues should be reported to maintainer, and mozilla herd"
+		ewarn "should be cc'd on the bug report. Thank you anarchy@gentoo.org ."
+	fi
 }
