--- /usr/portage/sys-libs/gpm/gpm-1.20.1-r6.ebuild	2009-02-15 23:37:14.000000000 +0100
+++ ./sys-libs/gpm/gpm-1.20.1-r6.ebuild	2010-03-14 17:50:44.000000000 +0100
@@ -23,17 +23,18 @@
 #	emacs? ( virtual/emacs )"
 RDEPEND="selinux? ( sec-policy/selinux-gpm )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	EPATCH_SUFFIX="patch" epatch "${WORKDIR}"/patch
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		--libdir=/$(get_libdir) \
 		--sysconfdir=/etc/gpm \
 		|| die "econf failed"
+}
+
+src_compile() {
 	emake \
 		CC=$(tc-getCC) \
 		AR=$(tc-getAR) \
