--- /usr/portage/sys-libs/gpm/gpm-1.20.6.ebuild	2009-09-08 20:06:56.000000000 +0200
+++ ./sys-libs/gpm/gpm-1.20.6.ebuild	2010-03-14 17:53:48.000000000 +0100
@@ -19,19 +19,19 @@
 	|| ( app-arch/xz-utils app-arch/lzma-utils )"
 RDEPEND="selinux? ( sec-policy/selinux-gpm )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.20.5-abi.patch
 	epatch "${FILESDIR}"/0001-daemon-use-sys-ioctl.h-for-ioctl.patch #222099
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		--libdir=/$(get_libdir) \
 		--sysconfdir=/etc/gpm \
 		|| die "econf failed"
+}
 
+src_compile() {
 	# workaround broken release
 	find -name '*.o' | xargs rm
 	emake clean || die
