--- /usr/portage/sys-libs/gpm/gpm-1.20.6.ebuild	2010-09-12 09:07:06.000000000 +0200
+++ ./sys-libs/gpm/gpm-1.20.6.ebuild	2010-09-12 13:52:54.000000000 +0200
@@ -19,20 +19,20 @@
 	app-arch/xz-utils"
 RDEPEND="selinux? ( sec-policy/selinux-gpm )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-1.20.5-abi.patch
 	epatch "${FILESDIR}"/0001-daemon-use-sys-ioctl.h-for-ioctl.patch #222099
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		--libdir=/$(get_libdir) \
 		--sysconfdir=/etc/gpm \
 		emacs=/bin/false \
 		|| die "econf failed"
+}
 
+src_compile() {
 	# workaround broken release
 	find -name '*.o' | xargs rm
 	emake clean || die
