--- lib/Makefile	2005-04-05 01:08:04.000000000 +0200
+++ lib/Makefile	2005-10-08 13:15:52.000000000 +0200
@@ -22,6 +22,8 @@
     include ../Makefile.opts
 endif
 
+LIBDIR=$(prefix)/lib
+
 # Core
 CIN      := $(wildcard *.c)
 # NETLINK_ROUTE
@@ -67,12 +69,12 @@
 	find .deps -iname \*\.d -exec rm -f '{}' \;
 
 install:
-	mkdir -p $(DESTDIR)$(prefix)/lib/
-	install -o root -g root -m 0644 $(OUT_SLIB) $(DESTDIR)$(prefix)/lib
-	rm -f $(DESTDIR)$(prefix)/lib/$(LN1_SLIB) ; \
-	$(LN) -s $(DESTDIR)$(prefix)/lib/$(OUT_SLIB) $(DESTDIR)$(prefix)/lib/$(LN1_SLIB)
-	rm -f $(DESTDIR)$(prefix)/lib/$(LN_SLIB) ; \
-	$(LN) -s $(DESTDIR)$(prefix)/lib/$(LN1_SLIB) $(DESTDIR)$(prefix)/lib/$(LN_SLIB)
+	mkdir -p $(DESTDIR)$(LIBDIR)/
+	install -o root -g root -m 0644 $(OUT_SLIB) $(DESTDIR)$(LIBDIR)
+	rm -f $(DESTDIR)$(LIBDIR)/$(LN1_SLIB) ; \
+	$(LN) -s $(DESTDIR)$(LIBDIR)/$(OUT_SLIB) $(DESTDIR)$(LIBDIR)/$(LN1_SLIB)
+	rm -f $(DESTDIR)$(LIBDIR)/$(LN_SLIB) ; \
+	$(LN) -s $(DESTDIR)$(LIBDIR)/$(LN1_SLIB) $(DESTDIR)$(LIBDIR)/$(LN_SLIB)
 
 $(DEPS): ../Makefile.opts
 
