--- krb5-1.4/src/clients/ksu/Makefile.in.orig	2004-04-29 18:51:10.000000000 -0400
+++ krb5-1.4/src/clients/ksu/Makefile.in	2005-04-28 16:51:37.000000000 -0400
@@ -8,6 +8,7 @@
 PROG_RPATH=$(KRB5_LIBDIR)
 
 KSU_LIBS=@KSU_LIBS@
+LAZY_LDFLAGS=-Wl,-z,now
 
 SRCS = \
 	$(srcdir)/krb_auth_su.c \
@@ -28,7 +29,7 @@
 all:: ksu
 
 ksu: $(OBJS) $(KRB5_BASE_DEPLIBS)
-	$(CC_LINK) -o $@ $(OBJS) $(KRB5_BASE_LIBS) $(KSU_LIBS)
+	$(CC_LINK) -o $@ $(OBJS) $(KRB5_BASE_LIBS) $(KSU_LIBS) $(LAZY_LDFLAGS)
 
 clean::
 	$(RM) ksu
