Index: libarchive/test/test_write_compress_lzma.c
===================================================================
--- libarchive/test/test_write_compress_lzma.c	(revision 2811)
+++ libarchive/test/test_write_compress_lzma.c	(revision 2812)
@@ -185,10 +185,15 @@
 	archive_write_close(a);
 	assert(0 == archive_write_finish(a));
 
-	/* Level 0 really does result in larger data. */
+	/* It would be nice to assert that compression-level=0 produced
+	 * consistently larger/smaller results than the default compression,
+	 * but the results here vary a lot depending on the version of liblzma
+	 * being used. */
+	/* 
 	failure("Compression-level=0 wrote %d bytes; default wrote %d bytes",
 	    (int)used2, (int)used1);
 	assert(used2 > used1);
+	*/
 
 	assert((a = archive_read_new()) != NULL);
 	assertA(0 == archive_read_support_format_all(a));
Index: libarchive/test/test_write_compress_xz.c
===================================================================
--- libarchive/test/test_write_compress_xz.c	(revision 2811)
+++ libarchive/test/test_write_compress_xz.c	(revision 2812)
@@ -193,10 +193,14 @@
 	archive_write_close(a);
 	assert(0 == archive_write_finish(a));
 
-	/* Level 0 really does result in larger data. */
+	/* I would like to assert that compression-level=0 results in
+	 * larger data than the default compression, but that's not true
+	 * for all versions of liblzma. */
+	/*
 	failure("Compression-level=0 wrote %d bytes; default wrote %d bytes",
 	    (int)used2, (int)used1);
 	assert(used2 > used1);
+	*/
 
 	assert((a = archive_read_new()) != NULL);
 	assertA(0 == archive_read_support_format_all(a));

Egenskapsändringar för: .
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk:r2811

