Patch from bug #253284 thanks to Lifong Sun

diff -urN a/cpio/cpio.c b/cpio/cpio.c
--- a/cpio/cpio.c	2008-12-29 04:08:16.000000000 +0800
+++ b/cpio/cpio.c	2009-01-01 22:23:06.000000000 +0800
@@ -321,7 +321,15 @@
 	"Common Options:\n"
 	"  -v    Verbose\n"
 	"Create: %p -o [options]  < [list of files] > [archive]\n"
+#ifdef HAVE_BZLIB_H
+#ifdef HAVE_ZLIB_H
 	"  -z, -y  Compress archive with gzip/bzip2\n"
+#else
+	"  -y  Compress archive with bzip2\n"
+#endif
+#elif defined HAVE_ZLIB_H
+	"  -z  Compress archive with gzip\n"
+#endif
 	"  --format {odc|newc|ustar}  Select archive format\n"
 	"List: %p -it < [archive]\n"
 	"Extract: %p -i [options] < [archive]\n";
@@ -387,12 +395,16 @@
 	if (cpio->archive == NULL)
 		cpio_errc(1, 0, "Failed to allocate archive object");
 	switch (cpio->compress) {
+#ifdef HAVE_BZLIB_H
 	case 'j': case 'y':
 		archive_write_set_compression_bzip2(cpio->archive);
 		break;
+#endif
+#ifdef HAVE_ZLIB_H
 	case 'z':
 		archive_write_set_compression_gzip(cpio->archive);
 		break;
+#endif
 	case 'Z':
 		archive_write_set_compression_compress(cpio->archive);
 		break;
