diff -dpruN --exclude='config*' OpenSP-1.5.1.orig/include/InternalInputSource.h OpenSP-1.5.1/include/InternalInputSource.h
--- OpenSP-1.5.1.orig/include/InternalInputSource.h	2003-07-17 20:14:00.000000000 +0200
+++ OpenSP-1.5.1/include/InternalInputSource.h	2005-11-16 07:22:04.700397800 +0100
@@ -34,7 +34,7 @@ public:
   void pushCharRef(Char ch, const NamedCharRef &);
   Boolean rewind(Messenger &);
   const StringC *contents();
-  InternalInputSource *InternalInputSource::asInternalInputSource();
+  InternalInputSource *asInternalInputSource();
   ~InternalInputSource();
 private:
   InternalInputSource(const InternalInputSource &); // undefined
