--- /usr/portage/sys-libs/db/db-5.0.21.ebuild	2010-05-11 10:01:45.000000000 +0200
+++ ./sys-libs/db/db-5.0.21.ebuild	2010-05-13 15:13:28.000000000 +0200
@@ -38,9 +38,8 @@
 RDEPEND="tcl? ( dev-lang/tcl )
 	java? ( >=virtual/jre-1.5 )"
 
-src_unpack() {
-	unpack "${MY_P}".tar.gz
-	cd "${WORKDIR}"/"${MY_P}"
+src_prepare() {
+	cd "${S}"/..
 	for (( i=1 ; i<=${PATCHNO} ; i++ ))
 	do
 		epatch "${DISTDIR}"/patch."${MY_PV}"."${i}"
@@ -88,11 +87,11 @@
 	done
 }
 
-src_compile() {
+src_configure() {
 	local myconf=''
 
 	# compilation with -O0 fails on amd64, see bug #171231
-	if use amd64; then
+	if use amd64 && [ ${ABI} = "amd64" ]; then
 		replace-flags -O0 -O2
 		is-flagq -O[s123] || append-flags -O2
 	fi
@@ -140,8 +139,6 @@
 		${myconf} \
 		$(use_enable test) \
 		"$@"
-
-	emake || die "make failed"
 }
 
 src_install() {
