--- /usr/portage/sys-apps/file/file-5.05.ebuild	2011-01-18 23:07:28.000000000 +0100
+++ ./sys-apps/file/file-5.05.ebuild	2011-01-22 14:42:37.000000000 +0100
@@ -22,8 +22,9 @@
 
 src_unpack() {
 	unpack ${P}.tar.gz
-	cd "${S}"
+}	
 
+src_prepare() {
 	elibtoolize
 	epunt_cxx
 
@@ -31,11 +32,14 @@
 	mv python/README{,.python}
 }
 
-src_compile() {
+src_configure() {
 	# file uses things like strndup() and wcwidth()
 	append-flags -D_GNU_SOURCE
 
 	econf || die
+}
+
+src_compile() {
 	emake || die
 
 	use python && cd python && distutils_src_compile
