diff -u -r gimp-2.4.5/app/config/gimpguiconfig.c gimp-2.4.5.new/app/config/gimpguiconfig.c
--- gimp-2.4.5/app/config/gimpguiconfig.c	2007-11-20 10:31:08.000000000 +0100
+++ gimp-2.4.5.new/app/config/gimpguiconfig.c	2008-03-25 14:13:13.000000000 +0100
@@ -40,7 +40,7 @@
 #ifdef G_OS_WIN32
 #  define DEFAULT_WEB_BROWSER      "not used on Windows"
 #else
-#  define DEFAULT_WEB_BROWSER      "firefox %s"
+#  define DEFAULT_WEB_BROWSER      "xdg-open %s"
 #endif
 
 
Sólo en gimp-2.4.5.new/app/config: gimpguiconfig.c~
diff -u -r gimp-2.4.5/docs/gimprc.5.in gimp-2.4.5.new/docs/gimprc.5.in
--- gimp-2.4.5/docs/gimprc.5.in	2008-01-30 11:45:12.000000000 +0100
+++ gimp-2.4.5.new/docs/gimprc.5.in	2008-03-25 14:13:51.000000000 +0100
@@ -788,7 +788,7 @@
 web-browser.
 
 .TP
-(web-browser "firefox %s")
+(web-browser "xdg-open %s")
 
 Sets the external web browser to be used.  This can be an absolute path or the
 name of an executable to search for in the user's PATH. If the command
Sólo en gimp-2.4.5.new/docs: gimprc.5.in~
diff -u -r gimp-2.4.5/etc/gimprc gimp-2.4.5.new/etc/gimprc
--- gimp-2.4.5/etc/gimprc	2008-01-30 11:45:12.000000000 +0100
+++ gimp-2.4.5.new/etc/gimprc	2008-03-25 14:14:30.000000000 +0100
@@ -625,7 +625,7 @@
 # appended to the command with a space separating the two.  This is a single
 # filename.
 # 
-# (web-browser "firefox %s")
+# (web-browser "xdg-open %s")
 
 # The window type hint that is set on the toolbox. This may affect how your
 # window manager decorates and handles the toolbox window.  Possible values
Sólo en gimp-2.4.5.new/etc: gimprc~
