--- desktop/gimp.desktop.in.in~	2007-12-30 23:34:46.000000000 +0100
+++ desktop/gimp.desktop.in.in	2007-12-30 23:34:46.000000000 +0100
@@ -16,4 +15,4 @@
 X-GNOME-Bugzilla-Version=@GIMP_VERSION@
 X-GNOME-Bugzilla-OtherBinaries=gimp-@GIMP_APP_VERSION@
 StartupNotify=true
-MimeType=image/bmp;image/g3fax;image/gif;image/jpeg;image/jpg;image/pjpeg;image/png;image/tiff;image/x-bmp;image/x-compressed-xcf;image/x-fits;image/x-gray;image/x-pcx;image/x-png;image/x-portable-anymap;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-psd;image/x-sgi;image/x-sun-raster;image/x-tga;image/x-xbitmap;image/x-xcf;image/x-xpixmap;image/x-xwindowdump;
+MimeType=image/bmp;image/g3fax;image/gif;image/jpeg;image/jpg;image/pjpeg;image/png;image/tiff;image/x-bmp;image/x-compressed-xcf;image/x-fits;image/x-gray;image/x-pcx;image/x-png;image/x-portable-anymap;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-psd;image/x-sgi;image/x-sun-raster;image/x-tga;image/x-xbitmap;image/x-xcf;image/x-xpixmap;image/x-xwindowdump;image/svg+xml;
