--- mozilla/security/coreconf/Linux.mk.orig	2008-12-11 09:12:48.000000000 +0000
+++ mozilla/security/coreconf/Linux.mk	2008-12-11 09:13:51.000000000 +0000
@@ -119,6 +119,10 @@
 	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
 	CPU_ARCH        = mips
 else
+ifeq ($(OS_TEST),mips64)
+	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	CPU_ARCH        = mips64
+else
 ifeq (,$(filter-out i%86,$(OS_TEST)))
 	OS_REL_CFLAGS	= -DLINUX1_2 -Di386 -D_XOPEN_SOURCE
 	CPU_ARCH	= x86
@@ -140,6 +144,7 @@
 endif
 endif
 endif
+endif
 
 
 LIBC_TAG		= _glibc
