diff -ur mozilla.orig/security/nss/lib/ckfw/builtins/config.mk mozilla/security/nss/lib/ckfw/builtins/config.mk
--- mozilla.orig/security/nss/lib/ckfw/builtins/config.mk	2009-04-15 15:29:14.000000000 +0000
+++ mozilla/security/nss/lib/ckfw/builtins/config.mk	2009-04-15 15:31:02.000000000 +0000
@@ -79,3 +79,10 @@
 endif
 endif
 endif
+
+ifeq ($(OS_TARGET),Linux)
+# The -R '$ORIGIN' linker option instructs this library to search for its
+# dependencies in the same directory where it resides.
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
+
Only in mozilla/security/nss/lib/ckfw/builtins: config.mk.orig
Only in mozilla/security/nss/lib/ckfw/builtins: config.mk.rej
diff -ur mozilla.orig/security/nss/lib/ckfw/capi/config.mk mozilla/security/nss/lib/ckfw/capi/config.mk
--- mozilla.orig/security/nss/lib/ckfw/capi/config.mk	2009-04-15 15:29:14.000000000 +0000
+++ mozilla/security/nss/lib/ckfw/capi/config.mk	2009-04-15 15:29:42.000000000 +0000
@@ -71,3 +71,9 @@
 MKSHLIB += -R '$$ORIGIN'
 endif
 
+ifeq ($(OS_TARGET),Linux)
+# The -R '$ORIGIN' linker option instructs this library to search for its
+# dependencies in the same directory where it resides.
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
+
Only in mozilla/security/nss/lib/ckfw/capi: config.mk.orig
diff -ur mozilla.orig/security/nss/lib/freebl/Makefile mozilla/security/nss/lib/freebl/Makefile
--- mozilla.orig/security/nss/lib/freebl/Makefile	2009-04-15 15:29:14.000000000 +0000
+++ mozilla/security/nss/lib/freebl/Makefile	2009-04-15 15:29:42.000000000 +0000
@@ -143,6 +143,7 @@
 endif
 
 ifeq ($(OS_TARGET),Linux)
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
 ifeq ($(CPU_ARCH),x86_64)
     ASFILES  = arcfour-amd64-gas.s mpi_amd64_gas.s
     ASFLAGS += -march=opteron -m64 -fPIC
Only in mozilla/security/nss/lib/freebl: Makefile.orig
diff -ur mozilla.orig/security/nss/lib/nss/config.mk mozilla/security/nss/lib/nss/config.mk
--- mozilla.orig/security/nss/lib/nss/config.mk	2009-04-15 15:29:14.000000000 +0000
+++ mozilla/security/nss/lib/nss/config.mk	2009-04-15 15:29:42.000000000 +0000
@@ -83,6 +83,11 @@
 
 endif
 
+ifeq ($(OS_TARGET),Linux)
+# The -R '$ORIGIN' linker option instructs this library to search for its
+# dependencies in the same directory where it resides.
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
 
 # $(PROGRAM) has explicit dependencies on $(EXTRA_LIBS)
 SHARED_LIBRARY_LIBS = \
Only in mozilla/security/nss/lib/nss: config.mk.orig
diff -ur mozilla.orig/security/nss/lib/smime/config.mk mozilla/security/nss/lib/smime/config.mk
--- mozilla.orig/security/nss/lib/smime/config.mk	2009-04-15 15:29:14.000000000 +0000
+++ mozilla/security/nss/lib/smime/config.mk	2009-04-15 15:31:30.000000000 +0000
@@ -106,3 +106,8 @@
 endif
 endif
 
+ifeq ($(OS_TARGET),Linux)
+# The -R '$ORIGIN' linker option instructs this library to search for its
+# dependencies in the same directory where it resides.
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
Only in mozilla/security/nss/lib/smime: config.mk.orig
Only in mozilla/security/nss/lib/smime: config.mk.rej
diff -ur mozilla.orig/security/nss/lib/softoken/config.mk mozilla/security/nss/lib/softoken/config.mk
--- mozilla.orig/security/nss/lib/softoken/config.mk	2009-04-15 15:29:14.000000000 +0000
+++ mozilla/security/nss/lib/softoken/config.mk	2009-04-15 15:29:42.000000000 +0000
@@ -114,6 +114,12 @@
 endif
 endif
 
+ifeq ($(OS_TARGET),Linux)
+# The -R '$ORIGIN' linker option instructs this library to search for its
+# dependencies in the same directory where it resides.
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
+
 ifeq ($(OS_TARGET),WINCE)
 DEFINES += -DDBM_USING_NSPR
 endif
Only in mozilla/security/nss/lib/softoken: config.mk.orig
diff -ur mozilla.orig/security/nss/lib/ssl/config.mk mozilla/security/nss/lib/ssl/config.mk
--- mozilla.orig/security/nss/lib/ssl/config.mk	2009-04-15 15:29:14.000000000 +0000
+++ mozilla/security/nss/lib/ssl/config.mk	2009-04-15 15:29:42.000000000 +0000
@@ -130,4 +130,10 @@
 endif
 endif
 
+ifeq ($(OS_TARGET),Linux)
+# The -R '$ORIGIN' linker option instructs this library to search for its
+# dependencies in the same directory where it resides.
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
+
 endif
Only in mozilla/security/nss/lib/ssl: config.mk.orig
