--- mozilla/security/nss/lib/freebl/intel-aes.s-old	2009-06-21 14:20:10.491641710 -0500
+++ mozilla/security/nss/lib/freebl/intel-aes.s	2009-06-21 14:21:33.846655888 -0500
@@ -1633,4 +1633,10 @@
 
 	xor	%eax, %eax
 	ret
-	.size intel_aes_decrypt_cbc_256, .-intel_aes_decrypt_cbc_256
+	.size intel_aes_decrypt_cbc_256, .-intel_aes_decrypt_cbc_256
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
+
+

