Index: libcaca-0.99.beta14/configure.ac
===================================================================
--- libcaca-0.99.beta14.orig/configure.ac
+++ libcaca-0.99.beta14/configure.ac
@@ -290,7 +290,7 @@ fi
 if test "${enable_vga}" = "yes"; then
   ac_cv_my_have_vga="yes"
   CPPFLAGS="${CPPFLAGS} -I. -D__KERNEL__ -nostdinc -include kernel/kernel.h -fno-stack-protector"
-  CFLAGS="${CFLAGS} -fno-builtin -O2 -Wall -fno-stack-protector"
+  CFLAGS="${CFLAGS} -Wall -fno-stack-protector"
   CCASFLAGS="${CCASFLAGS} -I. -fno-stack-protector"
   LDFLAGS="${LDFLAGS} -nostdlib -Wl,-N -Wl,-Ttext -Wl,100000 -fno-stack-protector"
   AC_DEFINE(USE_VGA, 1, Define to 1 to activate the VGA backend driver)
@@ -347,7 +347,7 @@ fi
 AC_SUBST(CUCUL_TYPES)
 
 # Optimizations
-CFLAGS="${CFLAGS} -g -O2 -fno-strength-reduce -fomit-frame-pointer"
+CFLAGS="${CFLAGS}"
 # Code qui fait des warnings == code de porc == deux baffes dans ta gueule
 CFLAGS="${CFLAGS} -Wall -Wpointer-arith -Wcast-align -Wcast-qual -Wstrict-prototypes -Wshadow -Waggregate-return -Wmissing-prototypes -Wnested-externs -Wsign-compare"
 
