--- /usr/portage/gnome-base/gnome-keyring/gnome-keyring-2.22.3-r2.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./gnome-base/gnome-keyring/gnome-keyring-2.22.3-r2.ebuild	2010-05-06 00:10:15.000000000 +0200
@@ -39,8 +39,8 @@
 		--with-root-certs=/usr/share/ca-certificates/"
 }
 
-src_unpack() {
-	gnome2_src_unpack
+src_prepare() {
+	gnome2_src_prepare
 
 	# Backport from trunk for fixing upstream bug #511285, bug #238098
 	epatch "${FILESDIR}/${P}-warnings.patch"
