--- /usr/portage/net-misc/openssh/openssh-5.3_p1-r1.ebuild	2010-03-12 18:38:59.000000000 +0100
+++ ./net-misc/openssh/openssh-5.3_p1-r1.ebuild	2010-03-14 01:44:07.000000000 +0100
@@ -72,14 +72,18 @@
 	unpack ${PARCH}.tar.gz
 	cd "${S}"
 
+	if use pkcs11 ; then
+		cd "${WORKDIR}"
+		unpack "${PKCS11_PATCH}"
+	fi
+}
+
+src_prepare() {
 	sed -i \
 		-e '/_PATH_XAUTH/s:/usr/X11R6/bin/xauth:/usr/bin/xauth:' \
 		pathnames.h || die
 
 	if use pkcs11 ; then
-		cd "${WORKDIR}"
-		unpack "${PKCS11_PATCH}"
-		cd "${S}"
 		# This patch is included with X509, so exclude it if X509 is going to be
 		# applied.
 		use X509 && mv -f "${WORKDIR}"/*pkcs11*/1000_all_log.patch "${WORKDIR}"
@@ -124,6 +128,9 @@
 	# Disable PATH reset, trust what portage gives us. bug 254615
 	sed -i -e 's:^PATH=/:#PATH=/:' configure || die
 
+	# Use CC not LD otherwise invalid LDFLAGS get passed to ld
+	sed -i -e 's:$(LD):$(CC):' Makefile.in || die
+
 	eautoreconf
 }
 
@@ -141,7 +148,7 @@
 	use_with "$@"
 }
 
-src_compile() {
+src_configure() {
 	addwrite /dev/ptmx
 	addpredict /etc/skey/skeykeys #skey configure code triggers this
 
@@ -169,7 +176,6 @@
 		$(use_with tcpd tcp-wrappers) \
 		${myconf} \
 		|| die "bad configure"
-	emake || die "compile problem"
 }
 
 src_install() {
