--- /usr/portage/net-misc/openssh/openssh-5.4_p1-r1.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./net-misc/openssh/openssh-5.4_p1-r1.ebuild	2010-04-02 18:05:05.000000000 +0200
@@ -63,10 +63,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${PARCH}.tar.gz
-	cd "${S}"
-
+src_prepare() {
 	sed -i \
 		-e '/_PATH_XAUTH/s:/usr/X11R6/bin/xauth:/usr/bin/xauth:' \
 		pathnames.h || die
@@ -98,6 +95,9 @@
 	# Disable PATH reset, trust what portage gives us. bug 254615
 	sed -i -e 's:^PATH=/:#PATH=/:' configure || die
 
+	# Use CC not LD otherwise invalid LDFLAGS get passed to ld
+	sed -i -e 's:$(LD):$(CC):' Makefile.in || die
+
 	eautoreconf
 }
 
@@ -115,7 +115,7 @@
 	use_with "$@"
 }
 
-src_compile() {
+src_configure() {
 	addwrite /dev/ptmx
 	addpredict /etc/skey/skeykeys #skey configure code triggers this
 
@@ -139,7 +139,6 @@
 		$(use_with skey) \
 		$(use_with tcpd tcp-wrappers) \
 		|| die
-	emake || die
 }
 
 src_install() {
