--- /usr/portage/dev-libs/fribidi/fribidi-0.19.1.ebuild	2010-01-14 22:35:29.000000000 +0100
+++ ./dev-libs/fribidi/fribidi-0.19.1.ebuild	2010-06-05 23:09:19.000000000 +0200
@@ -11,6 +11,9 @@
 KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sh ~sparc ~x86 ~x86-fbsd ~amd64-linux ~x86-linux ~ppc-macos ~x64-solaris ~x86-solaris"
 IUSE=""
 
+DEPEND="dev-util/pkgconfig"
+RDEPEND=""
+
 src_install() {
 	emake DESTDIR="${D}" install || die
 	dodoc AUTHORS NEWS README ChangeLog THANKS TODO
