diff -Naur gegl-0.0.18/operations/external/ff-load.c gegl-0.0.18-1/operations/external/ff-load.c
--- gegl-0.0.18/operations/external/ff-load.c	2008-06-14 18:40:04.000000000 +0200
+++ gegl-0.0.18-1/operations/external/ff-load.c	2008-06-26 14:45:14.000000000 +0200
@@ -27,7 +27,7 @@
 
 #include "gegl-chant.h"
 #include <errno.h>
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 
 typedef struct
 {
diff -Naur gegl-0.0.18/operations/workshop/external/ff-save.c gegl-0.0.18-1/operations/workshop/external/ff-save.c
--- gegl-0.0.18/operations/workshop/external/ff-save.c	2008-06-14 18:40:04.000000000 +0200
+++ gegl-0.0.18-1/operations/workshop/external/ff-save.c	2008-06-26 14:45:03.000000000 +0200
@@ -27,7 +27,7 @@
 #define GEGL_CHANT_C_FILE       "ff-save.c"
 
 #include "gegl-chant.h"
-#include "ffmpeg/avformat.h"
+#include "libavformat/avformat.h"
 
 typedef struct
 {
