From a1e3ab9e55047c08a4006ec389c1a99b72bc672c Mon Sep 17 00:00:00 2001
From: Ben Skeggs <bskeggs@redhat.com>
Date: Fri, 17 Apr 2009 09:29:19 +1000
Subject: [PATCH 2/4] nouveau: write posting got lost somewhere, bring it back

---
 libdrm/nouveau/nouveau_dma.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libdrm/nouveau/nouveau_dma.c b/libdrm/nouveau/nouveau_dma.c
index 23da64b..b084f70 100644
--- a/libdrm/nouveau/nouveau_dma.c
+++ b/libdrm/nouveau/nouveau_dma.c
@@ -41,6 +41,7 @@ WRITE_PUT(struct nouveau_channel_priv *nvchan, uint32_t val)
 	volatile int dum;
 
 	NOUVEAU_DMA_BARRIER;
+	dum = nvchan->pushbuf[0];
 	dum = READ_GET(nvchan);
 
 	*nvchan->put = put;
-- 
1.6.2.3

