From 78e753d03314e3aa92af8ec601e9dbb869a938d5 Mon Sep 17 00:00:00 2001
From: Ben Skeggs <bskeggs@redhat.com>
Date: Wed, 15 Apr 2009 12:44:21 +1000
Subject: [PATCH 1/4] nouveau: store bo handle in public struct in bo_ref_handle

---
 libdrm/nouveau/nouveau_bo.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/libdrm/nouveau/nouveau_bo.c b/libdrm/nouveau/nouveau_bo.c
index 023c6be..66466e3 100644
--- a/libdrm/nouveau/nouveau_bo.c
+++ b/libdrm/nouveau/nouveau_bo.c
@@ -404,7 +404,8 @@ nouveau_bo_handle_ref(struct nouveau_device *dev, uint32_t handle,
 		nvbo->size = req.size;
 		nvbo->handle = req.handle;
 	}
- 
+
+	nvbo->base.handle = nvbo->handle;
 	return 0;
 } 
 
-- 
1.6.2.3

