From 5e9cdc3321f6f273ae1844e7bee2e6ba6072243f Mon Sep 17 00:00:00 2001
From: Jamey Sharp <jamey@minilop.net>
Date: Thu, 18 Sep 2008 16:47:04 -0700
Subject: [PATCH] Receive selected replies in event queue.

Commit by Josh Triplett and Jamey Sharp.
---
 src/xcb_in.c |    5 +++--
 src/xcbext.h |    3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/xcb_in.c b/src/xcb_in.c
index f613772..2e32719 100644
--- a/src/xcb_in.c
+++ b/src/xcb_in.c
@@ -183,8 +183,9 @@ static int read_packet(xcb_connection_t *c)
         ((xcb_generic_event_t *) buf)->full_sequence = c->in.request_read;
 
     /* reply, or checked error */
-    if( genrep.response_type == XCB_REPLY ||
-       (genrep.response_type == XCB_ERROR && pend && (pend->flags & XCB_REQUEST_CHECKED)))
+    if(!(pend && (pend->flags & XCB_REQUEST_QUEUE_RESPONSE)) &&
+       ((pend && (pend->flags & XCB_REQUEST_CHECKED) && genrep.response_type == XCB_ERROR) ||
+        genrep.response_type == XCB_REPLY))
     {
         reader_list *reader;
         struct reply_list *cur = malloc(sizeof(struct reply_list));
diff --git a/src/xcbext.h b/src/xcbext.h
index 01dd590..856ad3d 100644
--- a/src/xcbext.h
+++ b/src/xcbext.h
@@ -54,7 +54,8 @@ typedef struct {
 enum xcb_send_request_flags_t {
     XCB_REQUEST_CHECKED = 1 << 0,
     XCB_REQUEST_RAW = 1 << 1,
-    XCB_REQUEST_DISCARD_REPLY = 1 << 2
+    XCB_REQUEST_DISCARD_REPLY = 1 << 2,
+    XCB_REQUEST_QUEUE_RESPONSE = 1 << 3 /* Put the response in the event queue */
 };
 
 unsigned int xcb_send_request(xcb_connection_t *c, int flags, struct iovec *vector, const xcb_protocol_request_t *request);
-- 
1.6.0.2.307.gc427.dirty

