--- /usr/portage/media-libs/freeglut/freeglut-2.4.0-r1.ebuild	2009-12-14 13:05:50.000000000 +0100
+++ ./media-libs/freeglut/freeglut-2.4.0-r1.ebuild	2010-05-30 17:39:19.000000000 +0200
@@ -29,10 +29,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# fixes bug #97390
 	epatch "${FILESDIR}"/${P}-macos.patch
 
@@ -52,10 +49,9 @@
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	# (#191589) Don't let -Werror get tagged on
 	econf --disable-warnings || die "econf failed"
-	emake || die "emake failed"
 }
 
 src_install() {
