--- /usr/portage/gnome-base/gnome-desktop/gnome-desktop-2.24.3-r1.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./gnome-base/gnome-desktop/gnome-desktop-2.24.3-r1.ebuild	2010-06-05 21:35:43.000000000 +0200
@@ -44,8 +44,8 @@
 	G2CONF="${G2CONF} --with-gnome-distributor=Gentoo --disable-scrollkeeper"
 }
 
-src_unpack() {
-	gnome2_src_unpack
+src_prepare() {
+	gnome2_src_prepare
 
 	# Do not load background if not needed, bug #251350
 	epatch "${FILESDIR}/${PN}-2.24.2-background.patch"
