From Gentoo bug #202618
diff -ur graphviz-2.16.1.orig/tclpkg/gv/Makefile.am graphviz-2.16.1/tclpkg/gv/Makefile.am
--- graphviz-2.16.1.orig/tclpkg/gv/Makefile.am	2007-12-17 22:10:02.000000000 +0100
+++ graphviz-2.16.1/tclpkg/gv/Makefile.am	2007-12-17 22:12:40.000000000 +0100
@@ -124,7 +124,8 @@
 libgv_python_la_SOURCES = $(BASESOURCES) gv_dummy_init.cpp
 libgv_python_la_LIBADD = $(BASELIBS) @PYTHON_LIBS@
 libgv_python_la_LDFLAGS = -module -avoid-version
-gv_python.cpp $(PYTHON_data): gv.i
+$(PYTHON_data): gv_python.cpp
+gv_python.cpp: gv.i
 	$(SWIG) -c++ -python -o gv_python.cpp $(srcdir)/gv.i
 
 pkgpython23dir = $(pkglibdir)/python23
