don't instal dead symlinks, reported in bug #212872

diff -ru graphviz-2.16.1.orig/tclpkg/gv/Makefile.am graphviz-2.16.1/tclpkg/gv/Makefile.am
--- graphviz-2.16.1.orig/tclpkg/gv/Makefile.am	2008-03-15 14:51:01.000000000 +0100
+++ graphviz-2.16.1/tclpkg/gv/Makefile.am	2008-03-15 14:57:18.000000000 +0100
@@ -298,8 +298,14 @@
 endif
 if WITH_PYTHON
 	(cd $(DESTDIR)$(pkgpythondir); rm -f _gv.so; $(LN_S) libgv_python.so _gv.so;)
+endif
+if WITH_PYTHON23
 	(cd $(DESTDIR)$(pkgpython23dir); rm -f _gv.so; $(LN_S) libgv_python23.so _gv.so;)
+endif
+if WITH_PYTHON24
 	(cd $(DESTDIR)$(pkgpython24dir); rm -f _gv.so; $(LN_S) libgv_python24.so _gv.so;)
+endif
+if WITH_PYTHON25
 	(cd $(DESTDIR)$(pkgpython25dir); rm -f _gv.so; $(LN_S) libgv_python25.so _gv.so;)
 endif
 if WITH_RUBY
