fix tcl/tc building for 8.5 - bug 181526

diff -ur graphviz-2.16.1.orig/tclpkg/tkspline/Makefile.am graphviz-2.16.1/tclpkg/tkspline/Makefile.am
--- graphviz-2.16.1.orig/tclpkg/tkspline/Makefile.am	2007-12-29 01:13:37.000000000 +0100
+++ graphviz-2.16.1/tclpkg/tkspline/Makefile.am	2007-12-29 01:13:58.000000000 +0100
@@ -11,7 +11,7 @@
 
 # If all else fails there is a local copy of tkInt.h in tkstubs/
 AM_CPPFLAGS = \
-	-I@TK_SRC_DIR@/generic \
+	-I@TK_SRC_DIR@/generic -I@TK_SRC_DIR@/unix \
 	-I@TCL_SRC_DIR@/generic \
 	@TCL_INCLUDES@ @TCLINT_INCLUDES@ \
 	@TK_INCLUDES@ @TKINT_INCLUDES@ -I$(top_srcdir)/tclpkg/tkstubs
