--- /usr/portage/sys-apps/pciutils/pciutils-3.0.2.ebuild	2009-02-07 16:06:39.000000000 +0100
+++ ./sys-apps/pciutils/pciutils-3.0.2.ebuild	2010-04-02 19:14:08.000000000 +0200
@@ -15,9 +15,7 @@
 
 DEPEND="zlib? ( sys-libs/zlib )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-3.0.0-build.patch #233314
 	epatch "${FILESDIR}"/pcimodules-${PN}-3.0.0.patch
 	epatch "${FILESDIR}"/${PN}-2.2.7-update-pciids-both-forms.patch
