--- /usr/portage/sys-apps/pciutils/pciutils-3.0.0.ebuild	2009-01-26 04:39:24.000000000 +0100
+++ ./sys-apps/pciutils/pciutils-3.0.0.ebuild	2010-04-02 19:12:23.000000000 +0200
@@ -15,9 +15,7 @@
 
 DEPEND="zlib? ( sys-libs/zlib )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-3.0.0-build.patch #233314
 	epatch "${FILESDIR}"/${PN}-3.0.0-resolv.patch #218555
 	epatch "${FILESDIR}"/pcimodules-${PN}-3.0.0.patch
