--- /usr/portage/sys-apps/pciutils/pciutils-3.1.2.ebuild	2009-02-03 06:35:57.000000000 +0100
+++ ./sys-apps/pciutils/pciutils-3.1.2.ebuild	2010-04-02 19:15:15.000000000 +0200
@@ -15,9 +15,7 @@
 
 DEPEND="zlib? ( sys-libs/zlib )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/pcimodules-${PN}-3.1.0.patch
 	epatch "${FILESDIR}"/${PN}-2.2.7-update-pciids-both-forms.patch
 	cp "${FILESDIR}"/pcimodules.c . || die
