--- /usr/portage/media-gfx/sane-backends/sane-backends-1.0.21.ebuild	2010-05-11 22:36:06.000000000 +0200
+++ ./media-gfx/sane-backends/sane-backends-1.0.21.ebuild	2010-06-03 18:31:11.000000000 +0200
@@ -151,10 +151,7 @@
 	done
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	cat >> backend/dll.conf.in <<-EOF
 	# Add support for the HP-specific backend.  Needs net-print/hplip installed.
 	hpaio
@@ -162,7 +159,7 @@
 	epatch "${DISTDIR}/${P}-i18n.patch"
 }
 
-src_compile() {
+src_configure() {
 	append-flags -fno-strict-aliasing
 
 	myconf=$(use_enable usb libusb)
@@ -181,7 +178,9 @@
 		$(use_enable ipv6) \
 		$(use_enable avahi) \
 		${myconf} || die "econf failed"
+}
 
+src_compile() {
 	emake VARTEXFONTS="${T}/fonts" || die
 
 	if use usb; then
