diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN lcms-1.17.orig/configure lcms-1.17/configure
--- lcms-1.17.orig/configure	2007-07-27 06:10:39.000000000 -0400
+++ lcms-1.17/configure	2007-08-14 14:43:28.000000000 -0400
@@ -5436,7 +5436,7 @@ echo "${ECHO_T}$am_cv_python_pyexecdir" 
   if test -n "$PYTHON"
   then
 	LCMS_PYTHON_PREFIX=`$PYTHON -c "import sys; print sys.prefix"`
-	LCMS_PYEXECDIR=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib()"`
+	LCMS_PYEXECDIR=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(True)"`
     LCMS_PYINCLUDE=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_inc()"`
 	LCMS_PYLIB=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(0,1)"`
 	as_ac_File=`echo "ac_cv_file_$LCMS_PYINCLUDE/Python.h" | $as_tr_sh`
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN lcms-1.17.orig/configure.ac lcms-1.17/configure.ac
--- lcms-1.17.orig/configure.ac	2007-01-31 06:35:24.000000000 -0500
+++ lcms-1.17/configure.ac	2007-08-14 14:42:19.000000000 -0400
@@ -94,7 +94,7 @@ then
   if test -n "$PYTHON"
   then
 	LCMS_PYTHON_PREFIX=`$PYTHON -c "import sys; print sys.prefix"`
-	LCMS_PYEXECDIR=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib()"`
+	LCMS_PYEXECDIR=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(True)"`
     LCMS_PYINCLUDE=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_inc()"`	
 	LCMS_PYLIB=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(0,1)"`
 	AC_CHECK_FILE([$LCMS_PYINCLUDE/Python.h],PYTHON_OK='yes',PYTHON_OK='no')
