--- /usr/portage/media-libs/libcanberra/libcanberra-0.11.ebuild	2009-04-28 14:08:52.000000000 +0200
+++ ./media-libs/libcanberra/libcanberra-0.11.ebuild	2010-05-02 20:58:18.000000000 +0200
@@ -26,7 +26,7 @@
 	>=dev-util/pkgconfig-0.17
 	doc? ( >=dev-util/gtk-doc-1.9 )"
 
-src_compile() {
+src_configure() {
 	econf --disable-static \
 		$(use_enable alsa) \
 		$(use_enable gstreamer) \
@@ -37,8 +37,6 @@
 		--disable-tdb \
 		--disable-lynx
 	# tdb support would need a split-out from samba before we can use it
-
-	emake || die "emake failed."
 }
 
 src_install() {
