--- /usr/portage/media-libs/libcanberra/libcanberra-0.22.ebuild	2010-04-11 16:35:58.000000000 +0200
+++ ./media-libs/libcanberra/libcanberra-0.22.ebuild	2010-05-02 21:00:08.000000000 +0200
@@ -28,10 +28,7 @@
 	dev-util/gtk-doc-am
 	doc? ( >=dev-util/gtk-doc-1.9 )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Fix bug 277739, replace LT_PREREQ and LT_INIT by AC_LIBTOOL*
 	# macros (equivalent for earlier version), preserve backward
 	# compatibility with libtool-1
@@ -49,7 +46,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	econf --disable-static \
 		--docdir=/usr/share/doc/${PF} \
 		$(use_enable alsa) \
@@ -62,8 +59,6 @@
 		--disable-lynx \
 		--with-html-dir=/usr/share/doc/${PF}/html
 	# tdb support would need a split-out from samba before we can use it
-
-	emake || die "emake failed."
 }
 
 src_install() {
