commit 05d39e4085d70d0693af1afb24ad08dee14b3d99
Author: Matthias Clasen <mclasen@redhat.com>
Date:   Wed Sep 2 22:14:47 2009 -0400

    Add a missing include
    
    This was reported in bug 593606.

diff --git a/gtk/gtkcellrendereraccel.c b/gtk/gtkcellrendereraccel.c
index 4b25289..dedbfda 100644
--- a/gtk/gtkcellrendereraccel.c
+++ b/gtk/gtkcellrendereraccel.c
@@ -24,6 +24,7 @@
 #include "gtkcellrendereraccel.h"
 #include "gtklabel.h"
 #include "gtkeventbox.h"
+#include "gtkmain.h"
 #include "gtkprivate.h"
 #include "gdk/gdkkeysyms.h"
 #include "gtkalias.h"
