--- /usr/portage/media-libs/libmodplug/libmodplug-0.8.7.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./media-libs/libmodplug/libmodplug-0.8.7.ebuild	2010-05-02 18:15:29.000000000 +0200
@@ -17,9 +17,7 @@
 RDEPEND=""
 DEPEND="dev-util/pkgconfig"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}/${PN}-0.8.4-timidity-patches.patch"
 	epatch "${FILESDIR}/${PN}-0.8.4-endian.patch"
 
