Index: lame-398/frontend/mp3rtp.c
===================================================================
--- lame-398.orig/frontend/mp3rtp.c
+++ lame-398/frontend/mp3rtp.c
@@ -12,8 +12,6 @@
 
 #ifdef HAVE_CONFIG_H
 # include <config.h>
-/* std int types already defined in config.h */
-#define _STDINT_H
 #endif
 
 #ifdef STDC_HEADERS
Index: lame-398/frontend/rtp.c
===================================================================
--- lame-398.orig/frontend/rtp.c
+++ lame-398/frontend/rtp.c
@@ -2,8 +2,6 @@
 
 #ifdef HAVE_CONFIG_H
 # include <config.h>
-/* std int types already defined in config.h */
-# define _STDINT_H
 #endif
 
 #ifndef __GNUC__
