--- /usr/portage/gnome-base/gnome-menus/gnome-menus-2.26.2.ebuild	2010-02-14 20:06:25.000000000 +0100
+++ ./gnome-base/gnome-menus/gnome-menus-2.26.2.ebuild	2010-06-05 21:34:00.000000000 +0200
@@ -33,8 +33,8 @@
 	G2CONF="${G2CONF} $(use_enable python) --disable-static"
 }
 
-src_unpack() {
-	gnome2_src_unpack
+src_prepare() {
+	gnome2_src_prepare
 
 	# Don't show KDE standalone settings desktop files in GNOME others menu
 	epatch "${FILESDIR}/${PN}-2.18.3-ignore_kde_standalone.patch"
