--- ./media-libs/sdl-image/sdl-image-1.2.10.ebuild.orig	2010-02-27 00:17:06.000000000 +0100
+++ ./media-libs/sdl-image/sdl-image-1.2.10.ebuild	2010-02-27 00:17:23.000000000 +0100
@@ -23,7 +23,7 @@
 
 S=${WORKDIR}/${MY_P}
 
-src_compile() {
+src_configure() {
 	econf \
 		$(use_enable gif) \
 		$(use_enable jpeg jpg) \
@@ -38,7 +38,6 @@
 		--enable-xpm \
 		--enable-xv \
 		|| die
-	emake || die "emake failed"
 }
 
 src_install() {
