--- /usr/portage/dev-db/unixODBC/unixODBC-2.2.12.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./dev-db/unixODBC/unixODBC-2.2.12.ebuild	2010-06-06 02:16:31.000000000 +0200
@@ -24,10 +24,7 @@
 DEPEND="${RDEPEND}
 	gnome? ( dev-util/cvs )" # see Bug 173256
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${WORKDIR}"/${PATCH_P}/*
 	epatch \
 		"${FILESDIR}/350-${P}-gODBCConfig-as-needed.patch" \
@@ -46,7 +43,7 @@
 	fi
 }
 
-src_compile() {
+src_configure() {
 	local myconf="--enable-gui=no"
 
 	econf --host=${CHOST} \
@@ -57,7 +54,6 @@
 		--enable-fdb \
 		--enable-ltdllib \
 		${myconf} || die "econf failed"
-	emake -j1 || die "emake failed"
 
 	if use gnome; then
 		# Symlink for configure
@@ -73,7 +69,15 @@
 			--sysconfdir=/etc/${PN} || die "econf gODBCConfig failed"
 		ln -s ../depcomp .
 		ln -s ../libtool .
+		cd ..
+	fi
+}
 
+src_compile() {
+	emake -j1 || die "emake failed"
+
+	if use gnome; then
+		cd gODBCConfig
 		emake || die "emake gODBCConfig failed"
 		cd ..
 	fi
