--- /usr/portage/gnome-base/gnome-vfs/gnome-vfs-2.24.3.ebuild	2010-04-06 12:51:34.000000000 +0200
+++ ./gnome-base/gnome-vfs/gnome-vfs-2.24.3.ebuild	2010-04-13 22:32:16.000000000 +0200
@@ -75,9 +75,7 @@
 	use gnutls && use ssl && G2CONF="${G2CONF} --disable-openssl"
 }
 
-src_unpack() {
-	gnome2_src_unpack
-
+src_prepare() {
 	# Allow the Trash on afs filesystems (#106118)
 	epatch "${FILESDIR}"/${PN}-2.12.0-afs.patch
 
