From 5a5eb6dd706179da9294e939ede807170b99c1a5 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <benjamin@sipsolutions.net>
Date: Tue, 22 Sep 2009 09:43:28 +0000
Subject: Correct the SYSTEM_LUA conditional check. (patch from bug #594014)

---
diff --git a/configure.ac b/configure.ac
index 06f388c..9061044 100644
--- a/configure.ac
+++ b/configure.ac
@@ -47,7 +47,7 @@ AC_ARG_ENABLE(deprecated,	[  --disable-deprecated    disable deprecated function
 AC_ARG_ENABLE(schemas,		[  --disable-schema        disable engine schemas],	[schemas=$enableval],	[schemas="yes"])
 AC_ARG_WITH(system-lua,	[  --with-system-lua        link with system Lua library], 		[system_lua=$withval], 	[system_lua="no"])
 
-AM_CONDITIONAL([SYSTEM_LUA], [test x$system_lua = xtrue])
+AM_CONDITIONAL([SYSTEM_LUA], [test x$system_lua = xyes])
 
 BUILD_ENGINES=""
 BUILD_THEMES=""
--
cgit v0.8.2
