From ee5d780dc2e424bce8a75b21eb6f1b74ea177247 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <benjamin@sipsolutions.net>
Date: Thu, 29 Apr 2010 15:07:31 +0000
Subject: Remove Xfce workaround. (bug #616124)

---
diff --git a/engines/clearlooks/src/clearlooks_style.c b/engines/clearlooks/src/clearlooks_style.c
index 97ff857..647e447 100644
--- a/engines/clearlooks/src/clearlooks_style.c
+++ b/engines/clearlooks/src/clearlooks_style.c
@@ -250,9 +250,8 @@ clearlooks_style_draw_shadow (DRAW_ARGS)
 		clearlooks_set_widget_parameters (widget, style, state_type, &params);
 		params.corners = CR_CORNER_NONE;
 
-		if (widget && !g_str_equal ("XfcePanelWindow", gtk_widget_get_name (gtk_widget_get_toplevel (widget))))
-			STYLE_FUNCTION(draw_frame) (cr, colors, &params, &frame,
-			                            x, y, width, height);
+		STYLE_FUNCTION(draw_frame) (cr, colors, &params, &frame,
+		                            x, y, width, height);
 	}
 	else if (DETAIL ("scrolled_window") || DETAIL ("viewport") || detail == NULL)
 	{
--
cgit v0.8.3.1
