diff -Naur tmp-old/configure tmp-new/configure
--- configure	2009-12-24 08:50:48.000000000 +0000
+++ configure	2009-12-24 09:12:00.000000000 +0000
@@ -10258,10 +10258,10 @@
 
 fi
 
-# Some versions of libedit contain strlcpy(); so disregard that library while
+# Some versions of libedit and libroken contain strlcpy(); so disregard that library while
 # checking for these standard libc functions.
 pgac_save_LIBS="$LIBS"
-LIBS=`echo "$LIBS" | sed -e 's/-ledit//g' -e 's/-lreadline//g'`
+LIBS=`echo "$LIBS" | sed -e 's/-ledit//g' -e 's/-lreadline//g' -e 's/-lgssapi//g' -e 's/-lkrb5//g'`
 
 for ac_func in crypt getopt getrusage inet_aton random rint srandom strdup strerror strlcat strlcpy strtol strtoul
 do :
diff -Naur tmp-old/configure.in tmp-new/configure.in
--- configure.in	2009-12-24 09:12:39.000000000 +0000
+++ configure.in	2009-12-24 09:21:38.000000000 +0000
@@ -1240,10 +1240,10 @@
   AC_CHECK_FUNCS([fpclass fp_class fp_class_d class], [break])
 fi
 
-# Some versions of libedit contain strlcpy(); so disregard that library while
+# Some versions of libedit and libroken contain strlcpy(); so disregard that library while
 # checking for these standard libc functions.
 pgac_save_LIBS="$LIBS"
-LIBS=`echo "$LIBS" | sed -e 's/-ledit//g' -e 's/-lreadline//g'`
+LIBS=`echo "$LIBS" | sed -e 's/-ledit//g' -e 's/-lreadline//g' -e 's/-lgssapi//g' -e 's/-lkrb5//g'`
 
 AC_REPLACE_FUNCS([crypt getopt getrusage inet_aton random rint srandom strdup strerror strlcat strlcpy strtol strtoul])
 
