diff -Naur postgresql-9.0beta3.orig/configure.in postgresql-9.0beta3/configure.in
--- postgresql-9.0beta3.orig/configure.in	2010-07-08 22:43:12.000000000 -0400
+++ postgresql-9.0beta3/configure.in	2010-07-12 00:13:37.585527031 -0400
@@ -19,10 +19,6 @@
 
 AC_INIT([PostgreSQL], [9.0beta3], [pgsql-bugs@postgresql.org])
 
-m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.63], [], [m4_fatal([Autoconf version 2.63 is required.
-Untested combinations of 'autoconf' and PostgreSQL versions are not
-recommended.  You can remove the check from 'configure.in' but it is then
-your responsibility whether the result works or not.])])
 AC_COPYRIGHT([Copyright (c) 1996-2010, PostgreSQL Global Development Group])
 AC_CONFIG_SRCDIR([src/backend/access/common/heaptuple.c])
 AC_CONFIG_AUX_DIR(config)
