=== contrib/Makefile
==================================================================
--- contrib/Makefile	(revision 10)
+++ contrib/Makefile	(local)
@@ -5,41 +5,8 @@
 include $(top_builddir)/src/Makefile.global
 
 WANTED_DIRS = \
-		btree_gist	\
-		chkpass		\
-		cube		\
-		dbase		\
-		dblink		\
-		dbmirror	\
-		dbsize		\
-		earthdistance	\
-		findoidjoins	\
-		fulltextindex	\
-		fuzzystrmatch	\
-		intagg		\
-		intarray	\
-		isbn_issn	\
-		lo		\
-		ltree		\
-		miscutil	\
-		mysql		\
-		noupdate	\
 		oid2name	\
-		pg_autovacuum	\
-		pg_dumplo	\
-		pg_trgm		\
 		pgbench		\
-		pgcrypto	\
-		pgstattuple	\
-		rtree_gist	\
-		seg		\
-		spi		\
-		string		\
-		tablefunc	\
-		tips		\
-		tsearch		\
-		tsearch2	\
-		userlock	\
 		vacuumlo
 
 # Missing:
=== src/Makefile
==================================================================
--- src/Makefile	(revision 10)
+++ src/Makefile	(local)
@@ -15,15 +15,11 @@
 
 all install installdirs uninstall dep depend distprep:
 	$(MAKE) -C port $@
-	$(MAKE) -C timezone $@
 	$(MAKE) -C backend $@
-	$(MAKE) -C backend/utils/mb/conversion_procs $@
 	$(MAKE) -C include $@
 	$(MAKE) -C interfaces $@
 	$(MAKE) -C bin $@
-	$(MAKE) -C pl $@
 	$(MAKE) -C makefiles $@
-	$(MAKE) -C utils $@
 
 install: install-local
 
=== src/backend/Makefile
==================================================================
--- src/backend/Makefile	(revision 10)
+++ src/backend/Makefile	(local)
@@ -34,7 +34,7 @@
 
 ##########################################################################
 
-all: submake-libpgport postgres $(POSTGRES_IMP)
+all: someheaders
 
 ifneq ($(PORTNAME), cygwin)
 ifneq ($(PORTNAME), win32)
@@ -104,8 +104,7 @@
 
 .PHONY: $(DIRS:%=%-recursive)
 # Update the commonly used headers before building the subdirectories
-$(DIRS:%=%-recursive): $(top_builddir)/src/include/parser/parse.h $(top_builddir)/src/include/utils/fmgroids.h
-	$(MAKE) -C $(subst -recursive,,$@) all
+someheaders $(DIRS:%=%-recursive): $(top_builddir)/src/include/parser/parse.h $(top_builddir)/src/include/utils/fmgroids.h
 
 
 $(DLLINIT): $(DLLINIT:%.o=%.c)
@@ -151,22 +150,7 @@
 
 ##########################################################################
 
-install: all installdirs install-bin
-ifeq ($(PORTNAME), cygwin)
-ifeq ($(MAKE_DLL), true)
-	$(INSTALL_DATA) libpostgres.a $(DESTDIR)$(libdir)/libpostgres.a
-endif
-endif
-ifeq ($(PORTNAME), win32)
-ifeq ($(MAKE_DLL), true)
-	$(INSTALL_DATA) libpostgres.a $(DESTDIR)$(libdir)/libpostgres.a
-endif
-endif
-	$(MAKE) -C catalog install-data
-	$(INSTALL_DATA) $(srcdir)/libpq/pg_hba.conf.sample $(DESTDIR)$(datadir)/pg_hba.conf.sample
-	$(INSTALL_DATA) $(srcdir)/libpq/pg_ident.conf.sample $(DESTDIR)$(datadir)/pg_ident.conf.sample
-	$(INSTALL_DATA) $(srcdir)/utils/misc/postgresql.conf.sample $(DESTDIR)$(datadir)/postgresql.conf.sample
-	$(INSTALL_DATA) $(srcdir)/access/transam/recovery.conf.sample $(DESTDIR)$(datadir)/recovery.conf.sample
+install:
 
 install-bin: postgres $(POSTGRES_IMP) installdirs
 	$(INSTALL_PROGRAM) postgres$(X) $(DESTDIR)$(bindir)/postgres$(X)
=== src/bin/Makefile
==================================================================
--- src/bin/Makefile	(revision 10)
+++ src/bin/Makefile	(local)
@@ -13,8 +13,8 @@
 top_builddir = ../..
 include $(top_builddir)/src/Makefile.global
 
-DIRS := initdb ipcclean pg_ctl pg_dump \
-	psql scripts pg_config pg_controldata pg_resetxlog
+DIRS := pg_dump \
+	psql scripts pg_config
 ifeq ($(PORTNAME), win32)
 DIRS+=pgevent
 endif
=== src/include/pg_config_manual.h
==================================================================
--- src/include/pg_config_manual.h	(revision 10)
+++ src/include/pg_config_manual.h	(local)
@@ -160,7 +160,7 @@
  * here's where to twiddle it.  You can also override this at runtime
  * with the postmaster's -k switch.
  */
-#define DEFAULT_PGSOCKET_DIR  "/tmp"
+#define DEFAULT_PGSOCKET_DIR  "/var/run/postgresql"
 
 /*
  * The random() function is expected to yield values between 0 and
