diff -ruN hal-0.5.7-old/fdi/policy/10osvendor/20-storage-methods.fdi hal-0.5.7/fdi/policy/10osvendor/20-storage-methods.fdi
--- hal-0.5.7-old/fdi/policy/10osvendor/20-storage-methods.fdi	2006-02-22 15:12:23.000000000 +0100
+++ hal-0.5.7/fdi/policy/10osvendor/20-storage-methods.fdi	2006-03-15 22:48:07.000000000 +0100
@@ -71,6 +71,29 @@
       <merge key="volume.ignore" type="bool">false</merge>
       <match key="@block.storage_device:storage.hotpluggable" bool="false">
         <match key="@block.storage_device:storage.removable" bool="false">
+          <merge key="volume.ignore" type="bool">true</merge>
+
+	  <match key="volume.is_mounted" bool="true">
+	    <!-- Show windows drives -->
+            <match key="volume.fstype" string="vfat">
+	      <merge key="volume.ignore" type="bool">false</merge>
+	    </match>
+            <match key="volume.fstype" string="ntfs">
+	      <merge key="volume.ignore" type="bool">false</merge>
+	    </match>
+
+	    <!-- Show /media/ and /mnt drives -->
+            <match key="volume.mount_point" compare_gt="/media">
+              <match key="volume.mount_point" compare_lt="/media0">
+	        <merge key="volume.ignore" type="bool">false</merge>
+	      </match>
+	    </match>
+            <match key="volume.mount_point" compare_gt="/mnt">
+              <match key="volume.mount_point" compare_lt="/mnt0">
+	        <merge key="volume.ignore" type="bool">false</merge>
+	      </match>
+	    </match>
+          </match>
 
           <!-- Should always ignore Apple Bootstrap partitions (it would be -->
 	  <!-- a security hole to mount it) - TODO: should use the bootable -->
