--- hal-0.5.7.1/fdi/policy/10osvendor/20-storage-methods.fdi	2006-02-22 15:15:33.000000000 
0100
+++ hal-0.5.7.1/fdi/policy/10osvendor/20-storage-methods.fdi	2006-10-09 10:46:36.000000000 
0200
@@ -63,21 +63,54 @@
     </match>
 
 
+    <match key="volume.fsusage" string="crypto">
+
+      <!-- Here follow volumes we specifically want to ignore - it is the -->
+      <!-- responsibility of software higher in the stack (e.g. gnome-vfs) -->
+      <!-- amd mount programs (e.g. Mount() on HAL) to respect volume.ignore -->
+      <merge key="volume.ignore" type="bool">false</merge>
+      <match key="@block.storage_device:storage.hotpluggable" bool="false">
+        <match key="@block.storage_device:storage.removable" bool="false">
+          <merge key="volume.ignore" type="bool">true</merge>
+
+	  <match key="volume.is_mounted" bool="true">
+	    <!-- Show /media/ drives -->
+            <match key="volume.mount_point" compare_gt="/media">
+              <match key="volume.mount_point" compare_lt="/media0">
+	        <merge key="volume.ignore" type="bool">false</merge>
+	      </match>
+	    </match>
+          </match>
+	</match>
+      </match>
+    </match>
+
     <match key="volume.fsusage" string="filesystem">
 
       <!-- Here follow volumes we specifically want to ignore - it is the -->
       <!-- responsibility of software higher in the stack (e.g. gnome-vfs) -->
       <!-- amd mount programs (e.g. Mount() on HAL) to respect volume.ignore -->
       <merge key="volume.ignore" type="bool">false</merge>
+
+      <!-- Should always ignore Apple Bootstrap partitions (it would be -->
+      <!-- a security hole to mount it) - TODO: should use the bootable -->
+      <!-- flag from the Mac partition table instead -->
+      <match key="volume.fstype" string="hfs">
+	<match key="volume.label" string="bootstrap">
+	  <merge key="volume.ignore" type="bool">true</merge>
+	</match>
+      </match>
+
       <match key="@block.storage_device:storage.hotpluggable" bool="false">
         <match key="@block.storage_device:storage.removable" bool="false">
+          <merge key="volume.ignore" type="bool">true</merge>
 
-          <!-- Should always ignore Apple Bootstrap partitions (it would be -->
-	  <!-- a security hole to mount it) - TODO: should use the bootable -->
-  	  <!-- flag from the Mac partition table instead -->
-          <match key="volume.fstype" string="hfs">
-	    <match key="volume.label" string="bootstrap">
-    	      <merge key="volume.ignore" type="bool">true</merge>
+	  <match key="volume.is_mounted" bool="true">
+	    <!-- Show /media/ drives -->
+            <match key="volume.mount_point" compare_gt="/media">
+              <match key="volume.mount_point" compare_lt="/media0">
+	        <merge key="volume.ignore" type="bool">false</merge>
+	      </match>
 	    </match>
           </match>
 
