--- hal-0.5.7.1/tools/hal-device.c.fixdbus	2006-11-21 15:30:47.000000000 +0100
+++ hal-0.5.7.1/tools/hal-device.c	2006-11-21 15:30:48.000000000 +0100
@@ -155,7 +155,6 @@
 
 	libhal_ctx_shutdown(hal_ctx, &error);
 	libhal_ctx_free(hal_ctx);
-	dbus_connection_close(conn);
 	dbus_connection_unref(conn);
 	dbus_error_free(&error);
 
--- hal-0.5.7.1/tools/lshal.c.fixdbus	2006-11-21 15:30:47.000000000 +0100
+++ hal-0.5.7.1/tools/lshal.c	2006-11-21 15:30:48.000000000 +0100
@@ -701,7 +701,6 @@
 	libhal_ctx_shutdown (hal_ctx, &error);
 	libhal_ctx_free (hal_ctx);
 
-	dbus_connection_close (conn);
 	dbus_connection_unref (conn);
 
 	if (show_device)
--- hal-0.5.7.1/libhal/libhal.c.fixdbus	2006-02-12 22:47:28.000000000 +0100
+++ hal-0.5.7.1/libhal/libhal.c	2006-11-21 15:33:31.000000000 +0100
@@ -1525,7 +1525,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -1710,7 +1710,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -1762,7 +1762,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -1814,7 +1814,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -1865,7 +1865,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -1926,7 +1926,7 @@
 							   message, -1,
 							   error);
 
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		if (strcmp (error->name,
 			    "org.freedesktop.Hal.DeviceAlreadyLocked") == 0) {
 			if (reason_why_locked != NULL) {
@@ -1985,7 +1985,7 @@
 							   message, -1,
 							   error);
 
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -2040,7 +2040,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return NULL;
 	}
@@ -2125,7 +2125,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -2179,7 +2179,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -2365,7 +2365,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -2652,7 +2652,7 @@
 	reply = dbus_connection_send_with_reply_and_block (ctx->connection,
 							   message, -1,
 							   error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -2788,7 +2788,7 @@
 			    "type='signal',"
 			    "interface='org.freedesktop.Hal.Device',"
 			    "sender='org.freedesktop.Hal'", error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		return FALSE;
 	}
 	return TRUE;
@@ -2820,7 +2820,7 @@
 		  "sender='org.freedesktop.Hal'," "path=%s", udi);
 
 	dbus_bus_add_match (ctx->connection, buf, error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		return FALSE;
 	}
 	return TRUE;
@@ -2848,7 +2848,7 @@
 		  "sender='org.freedesktop.Hal'," "path=%s", udi);
 
 	dbus_bus_remove_match (ctx->connection, buf, error);
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		return FALSE;
 	}
 	return TRUE;
@@ -3223,7 +3223,7 @@
 							   message, -1,
 							   error);
 
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -3273,7 +3273,7 @@
 							   message, -1,
 							   error);
 
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
@@ -3342,7 +3342,7 @@
 							   message, -1,
 							   error);
 
-	if (dbus_error_is_set (error)) {
+	if (error != NULL && dbus_error_is_set (error)) {
 		dbus_message_unref (message);
 		return FALSE;
 	}
