--- tools/hal-system-power-hibernate	2006-01-16 07:47:40.000000000 -0500
+++ tools/hal-system-power-hibernate-new	2006-11-19 18:50:43.000000000 -0500
@@ -36,6 +36,9 @@
 	elif [ -x "/usr/sbin/pmi" ] ; then
 		/usr/sbin/pmi action hibernate force
 		RET=$?
+	elif [ -x "/usr/sbin/hibernate-ram" ] ; then
+		/usr/sbin/hibernate-ram
+		RET=$?
 	elif [ -x "/usr/sbin/pm-hibernate" ] ; then
 		/usr/sbin/pm-hibernate
 		RET=$?
