diff -Naur hal-0.5.7.1.orig/hald/linux2/hotplug.c hal-0.5.7.1/hald/linux2/hotplug.c
--- hal-0.5.7.1.orig/hald/linux2/hotplug.c	2007-01-22 15:31:45.322111812 +0100
+++ hal-0.5.7.1/hald/linux2/hotplug.c	2007-01-22 15:31:15.594276651 +0100
@@ -283,6 +283,7 @@
 
 		len = strlen(hotplug_event->sysfs.sysfs_path);
 		is_partition = (isdigit(hotplug_event->sysfs.sysfs_path[len - 1])
+				&& (! strstr(hotplug_event->sysfs.sysfs_path, "/block/fd"))
 				&& ! strstr(hotplug_event->sysfs.sysfs_path, "/block/sr")) ||
 			       strstr (hotplug_event->sysfs.sysfs_path, "/fakevolume") ;
 
