diff --git a/hald-runner/main.c b/hald-runner/main.c
index 3b170b3..3842cb8 100644
--- a/hald-runner/main.c
+++ b/hald-runner/main.c
@@ -128,9 +128,10 @@ handle_start(DBusConnection *con, DBusMessage *msg)
 		goto malformed;
 
 	if (run_request_run(r, con, NULL, &pid)) {
+		gint64 ppid = pid;
 		reply = dbus_message_new_method_return(msg);
 		dbus_message_append_args (reply, 
-					  DBUS_TYPE_INT64, &pid,
+					  DBUS_TYPE_INT64, &ppid,
 					  DBUS_TYPE_INVALID);
 					  
 	} else {
diff --git a/hald-runner/runner.c b/hald-runner/runner.c
index 57a2a80..8cbc777 100644
--- a/hald-runner/runner.c
+++ b/hald-runner/runner.c
@@ -175,11 +175,12 @@ out:
 	/* emit a signal that this PID exited */
 	if(rd->con != NULL && rd->emit_pid_exited) {
 		DBusMessage *signal;
+		gint64 ppid = rd->pid;
 		signal = dbus_message_new_signal ("/org/freedesktop/HalRunner",
 						  "org.freedesktop.HalRunner",
 						  "StartedProcessExited");
 		dbus_message_append_args (signal, 
-					  DBUS_TYPE_INT64, &(rd->pid),
+					  DBUS_TYPE_INT64, &(ppid),
 					  DBUS_TYPE_INVALID);
 		dbus_connection_send(rd->con, signal, NULL);
 	}
