--- /usr/portage/sys-libs/ncurses/ncurses-5.6-r2.ebuild	2009-10-11 08:08:27.000000000 +0200
+++ ./sys-libs/ncurses/ncurses-5.6-r2.ebuild	2010-03-14 15:50:42.000000000 +0100
@@ -21,18 +21,23 @@
 
 S=${WORKDIR}/${MY_P}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	[[ -n ${PV_SNAP} ]] && epatch "${WORKDIR}"/${MY_P}-${PV_SNAP}-patch.sh
 	epatch "${WORKDIR}"/${P}-coverity.patch
 	epatch "${FILESDIR}"/${PN}-5.6-gfbsd.patch
 	epatch "${FILESDIR}"/${PN}-5.6-build.patch #184700
+
+	# Becaus of adding -L/usr/$(get_lib_dir) to LDFLAGS we see a bug when
+	# upgrading this lib. This is because the buildsystem try to link against the old
+	# version installed in the system. This patch should fix that
+	epatch "${FILESDIR}"/${PN}-5.6-ldflags-multilib-overlay.patch 
 }
 
-src_compile() {
+src_configure() {
 	export ac_cv_prog_AWK=gawk #259510
-	tc-export BUILD_CC
+	# The ebuild keeps failing if this variable is set when a
+	# crossdev compiler is installed so is better to remove it
+	#tc-export BUILD_CC
 
 	# Protect the user from themselves #115036
 	unset TERMINFO
@@ -41,18 +46,20 @@
 	use nocxx && myconf="${myconf} --without-cxx --without-cxx-binding"
 
 	# First we build the regular ncurses ...
-	mkdir "${WORKDIR}"/narrowc
-	cd "${WORKDIR}"/narrowc
-	do_compile ${myconf}
+	einfo "Configuring regular ncurses in ${WORKDIR}/narrowc.${ABI} ..."
+	mkdir "${WORKDIR}"/narrowc.${ABI}
+	cd "${WORKDIR}"/narrowc.${ABI}
+	do_configure ${myconf}
 
 	# Then we build the UTF-8 version
 	if use unicode ; then
-		mkdir "${WORKDIR}"/widec
-		cd "${WORKDIR}"/widec
-		do_compile ${myconf} --enable-widec --includedir=/usr/include/ncursesw
+		einfo "Configuring unicode ncurses in ${WORKDIR}/widec.${ABI} .."
+		mkdir "${WORKDIR}"/widec.${ABI}
+		cd "${WORKDIR}"/widec.${ABI}
+		do_configure ${myconf} --enable-widec
 	fi
 }
-do_compile() {
+do_configure() {
 	ECONF_SOURCE=${S}
 
 	# We need the basic terminfo files in /etc, bug #37026.  We will
@@ -90,23 +97,35 @@
 		${conf_abi} \
 		"$@" \
 		|| die "configure failed"
+}
 
+src_compile() {
 	# A little hack to fix parallel builds ... they break when
 	# generating sources so if we generate the sources first (in
 	# non-parallel), we can then build the rest of the package
 	# in parallel.  This is not really a perf hit since the source
 	# generation is quite small.  -vapier
+
+	cd "${WORKDIR}"/narrowc.${ABI}
+	einfo "Compiling regular ncurses in ${WORKDIR}/narrowc.${ABI} ..."
 	emake -j1 sources || die "make sources failed"
 	emake || die "make failed"
+
+	if use unicode ; then
+		cd "${WORKDIR}"/widec.${ABI}
+		einfo "Compiling unicode ncurses in ${WORKDIR}/widec.${ABI} .."
+		emake -j1 sources || die "make sources failed"
+		emake || die "make failed"
+	fi
 }
 
 src_install() {
 	# install unicode version second so that the binaries in /usr/bin
 	# support both wide and narrow
-	cd "${WORKDIR}"/narrowc
+	cd "${WORKDIR}"/narrowc.${ABI}
 	emake DESTDIR="${D}" install || die "make narrowc install failed"
 	if use unicode ; then
-		cd "${WORKDIR}"/widec
+		cd "${WORKDIR}"/widec.${ABI}
 		emake DESTDIR="${D}" install || die "make widec install failed"
 	fi
 
