--- /usr/portage/media-libs/libdc1394/libdc1394-2.0.2.ebuild	2009-04-05 10:08:24.000000000 +0200
+++ ./media-libs/libdc1394/libdc1394-2.0.2.ebuild	2010-05-02 20:25:55.000000000 +0200
@@ -17,7 +17,7 @@
 		juju? ( >=sys-kernel/linux-headers-2.6.23-r3 )
 		X? ( x11-libs/libSM x11-libs/libXv )"
 
-src_compile() {
+src_configure() {
 	local myconf=""
 	if use juju; then
 		myconf="--with-juju-dir"
@@ -28,7 +28,6 @@
 		$(use_with X x) \
 		${myconf} \
 		|| die "econf failed"
-	emake || die "emake failed"
 }
 
 src_install() {
