--- /usr/portage/media-libs/libdc1394/libdc1394-1.2.1.ebuild	2009-12-04 15:07:38.000000000 +0100
+++ ./media-libs/libdc1394/libdc1394-1.2.1.ebuild	2010-05-02 20:23:45.000000000 +0200
@@ -19,21 +19,18 @@
 	!=sys-libs/libdc1394-1.0.0
 	sys-devel/libtool"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	if ! use X; then
 		epatch "${FILESDIR}"/${P}-nox11.patch
 	fi
 }
 
-src_compile() {
+src_configure() {
 	if has_version '>=sys-libs/glibc-2.4' ; then
 		append-flags "-DCLK_TCK=CLOCKS_PER_SEC"
 	fi
 
 	econf
-	emake || die
 }
 
 src_install() {
