--- /usr/portage/dev-python/pygobject/pygobject-2.18.0.ebuild	2010-04-02 23:37:18.000000000 +0200
+++ ./dev-python/pygobject/pygobject-2.18.0.ebuild	2010-04-24 11:51:20.000000000 +0200
@@ -30,9 +30,7 @@
 		$(use_with libffi ffi)"
 }
 
-src_unpack() {
-	gnome2_src_unpack
-
+src_prepare() {
 	# Fix FHS compliance, see upstream bug #535524
 	epatch "${FILESDIR}/${PN}-2.15.4-fix-codegen-location.patch"
 
