--- /usr/portage/sys-apps/dbus/dbus-1.3.0.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./sys-apps/dbus/dbus-1.3.0.ebuild	2010-04-03 17:51:39.000000000 +0200
@@ -23,10 +23,7 @@
 	doc? (	app-doc/doxygen
 		app-text/xmlto )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Tests were restricted because of this
 	sed -e 's/.*bus_dispatch_test.*/printf ("Disabled due to excess noise\\n");/' \
 		-e '/"dispatch"/d' -i "${S}/bus/test-main.c"
@@ -37,7 +34,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	# so we can get backtraces from apps
 	append-flags -rdynamic
 
@@ -66,8 +63,6 @@
 	# after the compile, it uses a selinuxfs interface to
 	# check if the SELinux policy has the right support
 	use selinux && addwrite /selinux/access
-
-	emake || die "make failed"
 }
 
 src_test() {
