--- /usr/portage/sys-apps/dbus/dbus-1.2.3-r1.ebuild	2009-02-05 20:06:58.000000000 +0100
+++ ./sys-apps/dbus/dbus-1.2.3-r1.ebuild	2010-04-03 17:48:49.000000000 +0200
@@ -23,9 +23,7 @@
 	doc? (	app-doc/doxygen
 		app-text/xmlto )"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	# Fix potential DoS issue. fdo bug #17803. Gentoo bug #240308
 	epatch "${FILESDIR}"/${PN}-1.2.3-panic-from-dbus_signature_validate.patch
 	# Fix runtime error on FreeBSD. Gentoo bug #236779, fdo bug #17061
@@ -33,7 +31,7 @@
 	epatch "${FILESDIR}"/${P}-bsd.patch
 }
 
-src_compile() {
+src_configure() {
 	# so we can get backtraces from apps
 	append-flags -rdynamic
 
@@ -64,8 +62,6 @@
 	# after the compile, it uses a selinuxfs interface to
 	# check if the SELinux policy has the right support
 	use selinux && addwrite /selinux/access
-
-	emake || die "make failed"
 }
 
 src_test() {
