--- ./config.orig/acinclude.m4	2009-05-18 00:02:14.000000000 +0100
+++ ./config/acinclude.m4	2009-11-05 14:05:37.874808447 +0000
@@ -705,6 +705,25 @@
 ])
 
 dnl ------------------------------------------------------------------
+dnl @synopsis AC_PROG_PKG_CONFIG([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
+dnl Sets output variables PKG_CONFIG
+dnl ------------------------------------------------------------------
+
+
+AC_DEFUN([AC_PROG_PKG_CONFIG], 
+[
+  AC_ARG_VAR(PKG_CONFIG,[Location of the pkg-config program.])
+  AC_ARG_VAR(PKG_CONFIG_PATH, [Path for pkg-config descriptors.])
+  AC_PATH_PROG(PKG_CONFIG, pkg-config)
+  if test -z "$PKG_CONFIG" ; then
+      ifelse([$2],,:,[$2])
+  else
+      ifelse([$1],,:,[$1])
+  fi
+])
+
+
+dnl ------------------------------------------------------------------
 dnl @synopsis AC_PATH_QT([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
 dnl Process option --with-qt=DIR
 dnl Define HAVE_QT and set it to QT_VERSION
@@ -715,6 +734,7 @@
 [
   AC_REQUIRE([AC_PATH_X])
   AC_REQUIRE([AC_PATH_XTRA])
+  AC_REQUIRE([AC_PROG_PKG_CONFIG])        
   if test ${no_x-no} != yes ; then
     X_LIBS="$X_LIBS $X_PRE_LIBS -lXext -lX11 $X_EXTRA_LIBS"
   fi
@@ -745,72 +765,79 @@
        ac_userdef_qt=yes
        ac_has_qt="user defined QT_CFLAGS and QT_LIBS" # no questions asked
     else
-       ac_qt_dirs="/usr/local /usr/X11R6 /usr"
-       for lib in $libs ; do 
-         for n in /usr/$lib/qt* ; do
-           test -d $n && ac_qt_dirs="$n $ac_qt_dirs" ; 
+       if test "$PKG_CONFIG" != "no" && `$PKG_CONFIG --exists qt-mt` ; then
+         QT_CFLAGS=`$PKG_CONFIG --cflags qt-mt`
+         QT_LIBS=`$PKG_CONFIG --libs qt-mt`
+         ac_has_qt="pkg-config"
+         ac_userdef_qt=yes
+       else
+         ac_qt_dirs="/usr/local /usr/X11R6 /usr"
+         for lib in $libs ; do 
+           for n in /usr/$lib/qt* ; do
+             test -d $n && ac_qt_dirs="$n $ac_qt_dirs" ; 
+           done
          done
-       done
-       if test -d "$QTDIR" ; then 
-         ac_qt_dirs="$QTDIR $ac_qt_dirs"
-       fi
-       for dir in $ac_qt_dirs ; do
-          if test -r $dir/include/qwidget.h ; then
-            ac_has_qt=$dir
-            QTDIR=$dir
-            break
-          fi
-       done
-    fi
-    # Unusual install
-    if test "x$ac_has_qt" = xno ; then
-      ac_qt_names="qt3 qt2 qt"
-      ac_qt_dirs="/usr /usr/X11R6 /usr/local"
-      case "$host" in
-      *-darwin* | *-macos10*)
-        if test -d /opt/local ; then
-          ac_qt_dirs="/opt/local $ac_qt_dirs"
-        elif test -d /sw ; then
-          ac_qt_dirs="/sw $ac_qt_dirs"
-	fi
-      ;;
-      esac
-      ac_qt_dirs="$QTDIR $prefix $ac_qt_dirs"
-      for d in $ac_qt_dirs ; do
-        for n in $ac_qt_names ; do
-          if test -r $d/include/$n/qwidget.h ; then
-            for lib in $libs ; do
-              for l in lib$n.so lib$n-mt.so lib$n-mt.dylib lib$n.a lib$n-mt.a ; do
-                  if test -r $d/$lib/$l ; then
-                      QT_CFLAGS="-I$d/include/$n"
-                      QT_LIBS="-L$d/$lib -l$n"
-                      QTDIR=$d
-                      ac_has_qt="bsd-style Qt install"
-                      break 3
-                  fi
-              done
-              for l in libqt.dll.a libqt-mt.dll.a libqt-mt.la libqt.a libqt-mt.a; do
-                  if test -r $d/$lib/$n/$lib/$l ; then
-                      QT_CFLAGS="-I$d/include/$n"
-                      QT_LIBS="-L$d/$lib/$n/$lib -lqt"
-                      QTDIR=$d/$lib/$n
-                      ac_has_qt="cygwin-style Qt install"
-                      break 3
-                  fi
-              done
-              for l in libqt.so libqt-mt.so libqt-mt.dylib libqt.a libqt-mt.a; do
-                  if test -r $d/$lib/$l ; then
-                      QT_CFLAGS="-I$d/include/$n"
-                      QT_LIBS="-L$d/$lib -lqt"
-                      QTDIR=$d
-                      ac_has_qt="debian-style Qt install"
-                      break 3
-                  fi
+         if test -d "$QTDIR" ; then 
+           ac_qt_dirs="$QTDIR $ac_qt_dirs"
+         fi
+         for dir in $ac_qt_dirs ; do
+            if test -r $dir/include/qwidget.h ; then
+              ac_has_qt=$dir
+              QTDIR=$dir
+              break
+            fi
+         done
+      fi
+      # Unusual install
+      if test "x$ac_has_qt" = xno ; then
+        ac_qt_names="qt3 qt2 qt"
+        ac_qt_dirs="/usr /usr/X11R6 /usr/local"
+        case "$host" in
+        *-darwin* | *-macos10*)
+          if test -d /opt/local ; then
+            ac_qt_dirs="/opt/local $ac_qt_dirs"
+          elif test -d /sw ; then
+            ac_qt_dirs="/sw $ac_qt_dirs"
+  	fi
+        ;;
+        esac
+        ac_qt_dirs="$QTDIR $prefix $ac_qt_dirs"
+        for d in $ac_qt_dirs ; do
+          for n in $ac_qt_names ; do
+            if test -r $d/include/$n/qwidget.h ; then
+              for lib in $libs ; do
+                for l in lib$n.so lib$n-mt.so lib$n-mt.dylib lib$n.a lib$n-mt.a ; do
+                    if test -r $d/$lib/$l ; then
+                        QT_CFLAGS="-I$d/include/$n"
+                        QT_LIBS="-L$d/$lib -l$n"
+                        QTDIR=$d
+                        ac_has_qt="bsd-style Qt install"
+                        break 3
+                    fi
+                done
+                for l in libqt.dll.a libqt-mt.dll.a libqt-mt.la libqt.a libqt-mt.a; do
+                    if test -r $d/$lib/$n/$lib/$l ; then
+                        QT_CFLAGS="-I$d/include/$n"
+                        QT_LIBS="-L$d/$lib/$n/$lib -lqt"
+                        QTDIR=$d/$lib/$n
+                        ac_has_qt="cygwin-style Qt install"
+                        break 3
+                    fi
+                done
+                for l in libqt.so libqt-mt.so libqt-mt.dylib libqt.a libqt-mt.a; do
+                    if test -r $d/$lib/$l ; then
+                        QT_CFLAGS="-I$d/include/$n"
+                        QT_LIBS="-L$d/$lib -lqt"
+                        QTDIR=$d
+                        ac_has_qt="debian-style Qt install"
+                        break 3
+                    fi
+                done
               done
-            done
-          fi
+            fi
+          done
         done
-      done
+      fi
     fi
     # Print result
     AC_MSG_RESULT($ac_has_qt)
--- ./configure.ac.orig	2009-11-05 15:50:33.880583471 +0000
+++ configure.ac	2009-11-05 15:49:49.879564385 +0000
@@ -28,6 +28,7 @@
 AC_INIT(djvulibre, 3.5.22)
 AC_REVISION($Id: configure.ac,v 1.99 2009/05/21 15:15:50 leonb Exp $)
 AC_CONFIG_AUX_DIR(config)
+AC_CONFIG_MACRO_DIR(config)
 AC_CONFIG_SRCDIR(INSTALL)
 AC_CONFIG_HEADER(config.h:config/config.h.in)
 AC_PREFIX_DEFAULT(/usr/local)
