--- /usr/portage/app-text/ghostscript-gpl/ghostscript-gpl-8.70-r1.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./app-text/ghostscript-gpl/ghostscript-gpl-8.70-r1.ebuild	2010-06-06 02:55:31.000000000 +0200
@@ -50,10 +50,7 @@
 	IUSE="${IUSE} linguas_${X}"
 done
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# remove internal copies of expat, jasper, jpeg, libpng and zlib
 	rm -rf "${S}/expat"
 	rm -rf "${S}/jasper"
@@ -118,7 +115,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		$(use_enable cairo) \
 		$(use_enable cups) \
@@ -137,10 +134,14 @@
 		sed -i -e 's!$(DD)bbox.dev!& $(DD)djvumask.dev $(DD)djvusep.dev!g' Makefile
 	fi
 
+	cd "${S}/ijs"
+	econf || die "ijs econf failed"
+}
+
+src_compile() {
 	emake -j1 so all || die "emake failed"
 
 	cd "${S}/ijs"
-	econf || die "ijs econf failed"
 	emake || die "ijs emake failed"
 }
 
