--- /usr/portage/app-text/ghostscript-gpl/ghostscript-gpl-8.64-r3.ebuild	2010-03-31 04:36:03.000000000 +0200
+++ ./app-text/ghostscript-gpl/ghostscript-gpl-8.64-r3.ebuild	2010-06-06 02:53:43.000000000 +0200
@@ -67,7 +67,9 @@
 	rm -rf "${S}/zlib"
 	# remove internal urw-fonts
 	rm -rf "${S}/Resource/Font"
+}
 
+src_prepare() {
 	# Fedora patches
 	# http://cvs.fedora.redhat.com/viewcvs/devel/ghostscript/
 	epatch "${WORKDIR}/patches/${PN}-8.64-fPIC.patch"
@@ -128,7 +130,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		$(use_enable cairo) \
 		$(use_enable cups) \
@@ -147,10 +149,14 @@
 		sed -i -e 's!$(DD)bbox.dev!& $(DD)djvumask.dev $(DD)djvusep.dev!g' Makefile
 	fi
 
+	cd "${S}/ijs"
+	econf || die "ijs econf failed"
+}
+
+src_compile() {
 	emake -j1 so all || die "emake failed"
 
 	cd "${S}/ijs"
-	econf || die "ijs econf failed"
 	emake || die "ijs emake failed"
 }
 
