--- /usr/portage/app-text/poppler/poppler-0.12.3-r3.ebuild	2010-03-11 23:07:38.000000000 +0100
+++ ./app-text/poppler/poppler-0.12.3-r3.ebuild	2010-06-06 02:42:02.000000000 +0200
@@ -46,6 +46,7 @@
 	epatch "${FILESDIR}"/${P}-gdk.patch
 	epatch "${FILESDIR}"/${P}-darwin-gtk-link.patch
 	epatch "${FILESDIR}"/${P}-config.patch  #304407
+	epatch "${FILESDIR}"/${P}-cmake-cflags.patch
 }
 
 src_configure() {
