Index: Linux-PAM-1.1.1/configure.in
===================================================================
--- Linux-PAM-1.1.1.orig/configure.in
+++ Linux-PAM-1.1.1/configure.in
@@ -360,6 +360,10 @@ AC_SUBST(LIBAUDIT)
 AC_CHECK_HEADERS(xcrypt.h crypt.h)
+AS_IF([test "x$ac_cv_header_xcrypt_h" = "xyes"],
+  [crypt_libs="xcrypt crypt"],
+  [crypt_libs="crypt"])
+
 BACKUP_LIBS=$LIBS
-AC_SEARCH_LIBS([crypt],[xcrypt crypt], LIBCRYPT="-l$ac_lib", LIBCRYPT="")
+AC_SEARCH_LIBS([crypt],[$crypt_libs], LIBCRYPT="-l$ac_lib", LIBCRYPT="")
 AC_CHECK_FUNCS(crypt_r crypt_gensalt_rn)
 LIBS=$BACKUP_LIBS
 AC_SUBST(LIBCRYPT)
