Index: Linux-PAM-0.99.7.0/configure.in
===================================================================
--- Linux-PAM-0.99.7.0.orig/configure.in
+++ Linux-PAM-0.99.7.0/configure.in
@@ -420,10 +420,12 @@ AC_CHECK_FUNCS(inet_ntop inet_pton ruser
 AC_CHECK_FUNCS(unshare, [UNSHARE=yes], [UNSHARE=no])
 AM_CONDITIONAL([HAVE_UNSHARE], [test "$UNSHARE" = yes])
 
+AC_ARG_ENABLE([regenerate-man],
+  AC_HELP_STRING([--disable-regenerate-man], [Don't re-build manpages from XML souces]),
+  [enable_man=$enableval], [enable_man=yes])
 dnl
 dnl Check for xsltproc
 dnl
-enable_man=yes
 AC_PATH_PROG([XSLTPROC], [xsltproc])
 if test -z "$XSLTPROC"; then
      enable_man=no
