Index: mozilla/nsprpub/pr/include/prtypes.h
diff -u mozilla/nsprpub/pr/include/prtypes.h:3.30 mozilla/nsprpub/pr/include/prtypes.h:3.30.2.1
--- mozilla/nsprpub/pr/include/prtypes.h:3.30   Fri Apr 29 21:02:54 2005
+++ mozilla/nsprpub/pr/include/prtypes.h        Tue Jan  3 17:43:35 2006
@@ -178,7 +178,9 @@

 #else /* Unix */

-#ifdef HAVE_VISIBILITY_PRAGMA
+/* GCC 3.3 and later support the visibility attribute. */
+#if (__GNUC__ >= 4) || \
+    (__GNUC__ == 3 && __GNUC_MINOR__ >= 3)
 #define PR_VISIBILITY_DEFAULT __attribute__((visibility("default")))
 #else
 #define PR_VISIBILITY_DEFAULT
