--- WebKit/gtk/WebCoreSupport/EditorClientGtk.cpp~	2009-02-07 14:07:51.000000000 +0000
+++ WebKit/gtk/WebCoreSupport/EditorClientGtk.cpp	2009-02-07 14:16:04.297682251 +0000
@@ -169,7 +169,7 @@
 {
     WebKitWebView* webView = reinterpret_cast<WebKitWebView*>(data);
     Frame* frame = core(webView)->focusController()->focusedOrMainFrame();
-    PassRefPtr<Range> selectedRange = frame->selection()->toRange();
+    PassRefPtr<Range> selectedRange = frame->selection()->toNormalizedRange();
 
     if (static_cast<gint>(info) == WEBKIT_WEB_VIEW_TARGET_INFO_HTML) {
         String markup = createMarkup(selectedRange.get(), 0, AnnotateForInterchange);
