--- /usr/portage/sys-libs/pam/pam-1.1.1-r1.ebuild	2010-03-10 17:04:07.000000000 +0100
+++ ./sys-libs/pam/pam-1.1.1-r1.ebuild	2010-03-14 15:29:32.000000000 +0100
@@ -78,10 +78,7 @@
 	check_old_modules
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Avoid building xtests during "make all"; note that for what
 	# we're concerned xtests are not even executed, so we should
 	# probably use EXTRA_PROGRAMS.
@@ -103,7 +100,7 @@
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	local myconf
 
 	if use hppa || use elibc_FreeBSD; then
@@ -126,6 +123,9 @@
 		--disable-dependency-tracking \
 		--disable-prelude \
 		${myconf} || die "econf failed"
+}
+
+src_compile() {
 	emake sepermitlockdir="/var/run/sepermit" || die "emake failed"
 }
 
