--- /usr/portage/sys-libs/pam/pam-1.0.4.ebuild	2010-01-06 17:06:49.000000000 +0100
+++ ./sys-libs/pam/pam-1.0.4.ebuild	2010-03-14 15:25:22.000000000 +0100
@@ -91,10 +91,7 @@
 	check_old_modules
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	mkdir -p doc/txts
 	for readme in modules/pam_*/README; do
 		cp -f "${readme}" doc/txts/README.$(dirname "${readme}" | \
@@ -118,7 +115,7 @@
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	local myconf
 
 	if use hppa || use elibc_FreeBSD; then
@@ -148,6 +145,9 @@
 		--disable-prelude \
 		--disable-regenerate-man \
 		${myconf} || die "econf failed"
+}
+
+src_compile() {
 	emake sepermitlockdir="/var/run/sepermit" || die "emake failed"
 }
 
