--- /usr/portage/sys-libs/pam/pam-1.1.0.ebuild	2010-04-26 14:06:59.000000000 +0200
+++ ./sys-libs/pam/pam-1.1.0.ebuild	2010-04-28 22:53:19.000000000 +0200
@@ -75,10 +75,7 @@
 	check_old_modules
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Avoid building xtests during "make all"; note that for what
 	# we're concerned xtests are not even executed, so we should
 	# probably use EXTRA_PROGRAMS.
@@ -105,7 +102,7 @@
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	local myconf
 
 	if use hppa || use elibc_FreeBSD; then
@@ -132,6 +129,9 @@
 		--disable-dependency-tracking \
 		--disable-prelude \
 		${myconf} || die "econf failed"
+}
+
+src_compile() {
 	emake sepermitlockdir="/var/run/sepermit" || die "emake failed"
 }
 
