diff -ur xvidcore.orig/src/bitstream/x86_asm/cbp_mmx.asm xvidcore/src/bitstream/x86_asm/cbp_mmx.asm
--- xvidcore.orig/src/bitstream/x86_asm/cbp_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/bitstream/x86_asm/cbp_mmx.asm	2009-05-29 20:08:30.000000000 +0300
@@ -128,6 +128,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/bitstream/x86_asm/cbp_sse2.asm xvidcore/src/bitstream/x86_asm/cbp_sse2.asm
--- xvidcore.orig/src/bitstream/x86_asm/cbp_sse2.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/bitstream/x86_asm/cbp_sse2.asm	2009-05-29 20:08:46.000000000 +0300
@@ -118,6 +118,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/dct/x86_asm/fdct_mmx_ffmpeg.asm xvidcore/src/dct/x86_asm/fdct_mmx_ffmpeg.asm
--- xvidcore.orig/src/dct/x86_asm/fdct_mmx_ffmpeg.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/dct/x86_asm/fdct_mmx_ffmpeg.asm	2009-05-29 20:00:19.000000000 +0300
@@ -402,6 +402,6 @@
 MAKE_FDCT_FUNC fdct_xmm_ffmpeg, FDCT_ROW_XMM
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/dct/x86_asm/fdct_mmx_skal.asm xvidcore/src/dct/x86_asm/fdct_mmx_skal.asm
--- xvidcore.orig/src/dct/x86_asm/fdct_mmx_skal.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/dct/x86_asm/fdct_mmx_skal.asm	2009-05-29 19:59:46.000000000 +0300
@@ -501,6 +501,6 @@
 MAKE_FDCT_FUNC fdct_xmm_skal, fMTX_MULT_XMM
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/dct/x86_asm/fdct_sse2_skal.asm xvidcore/src/dct/x86_asm/fdct_sse2_skal.asm
--- xvidcore.orig/src/dct/x86_asm/fdct_sse2_skal.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/dct/x86_asm/fdct_sse2_skal.asm	2009-05-29 20:01:18.000000000 +0300
@@ -609,6 +609,6 @@
   times 8 dw 0 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/dct/x86_asm/idct_3dne.asm xvidcore/src/dct/x86_asm/idct_3dne.asm
--- xvidcore.orig/src/dct/x86_asm/idct_3dne.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/dct/x86_asm/idct_3dne.asm	2009-05-29 20:00:38.000000000 +0300
@@ -700,6 +700,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/dct/x86_asm/idct_mmx.asm xvidcore/src/dct/x86_asm/idct_mmx.asm
--- xvidcore.orig/src/dct/x86_asm/idct_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/dct/x86_asm/idct_mmx.asm	2009-05-29 20:00:00.000000000 +0300
@@ -610,6 +610,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/dct/x86_asm/idct_sse2_dmitry.asm xvidcore/src/dct/x86_asm/idct_sse2_dmitry.asm
--- xvidcore.orig/src/dct/x86_asm/idct_sse2_dmitry.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/dct/x86_asm/idct_sse2_dmitry.asm	2009-05-29 20:01:37.000000000 +0300
@@ -327,6 +327,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/colorspace_rgb_mmx.asm xvidcore/src/image/x86_asm/colorspace_rgb_mmx.asm
--- xvidcore.orig/src/image/x86_asm/colorspace_rgb_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/colorspace_rgb_mmx.asm	2009-05-29 20:04:41.000000000 +0300
@@ -531,6 +531,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/colorspace_yuv_mmx.asm xvidcore/src/image/x86_asm/colorspace_yuv_mmx.asm
--- xvidcore.orig/src/image/x86_asm/colorspace_yuv_mmx.asm	2009-05-28 18:04:35.000000000 +0300
+++ xvidcore/src/image/x86_asm/colorspace_yuv_mmx.asm	2009-05-29 20:02:30.000000000 +0300
@@ -475,6 +475,6 @@
 MAKE_YV12_TO_YV12	yv12_to_yv12_xmm, 1
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/colorspace_yuyv_mmx.asm xvidcore/src/image/x86_asm/colorspace_yuyv_mmx.asm
--- xvidcore.orig/src/image/x86_asm/colorspace_yuyv_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/colorspace_yuyv_mmx.asm	2009-05-29 20:05:16.000000000 +0300
@@ -342,6 +342,6 @@
 MAKE_COLORSPACE  yv12_to_uyvyi_mmx,0,   2,8,4,  YV12_TO_YUYVI, 1, -1
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/deintl_sse.asm xvidcore/src/image/x86_asm/deintl_sse.asm
--- xvidcore.orig/src/image/x86_asm/deintl_sse.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/deintl_sse.asm	2009-05-29 20:05:52.000000000 +0300
@@ -133,5 +133,5 @@
 ;//////////////////////////////////////////////////////////////////////
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
diff -ur xvidcore.orig/src/image/x86_asm/gmc_mmx.asm xvidcore/src/image/x86_asm/gmc_mmx.asm
--- xvidcore.orig/src/image/x86_asm/gmc_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/gmc_mmx.asm	2009-05-29 20:07:06.000000000 +0300
@@ -222,5 +222,5 @@
 ;//////////////////////////////////////////////////////////////////////
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
diff -ur xvidcore.orig/src/image/x86_asm/interpolate8x8_3dn.asm xvidcore/src/image/x86_asm/interpolate8x8_3dn.asm
--- xvidcore.orig/src/image/x86_asm/interpolate8x8_3dn.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/interpolate8x8_3dn.asm	2009-05-29 20:03:34.000000000 +0300
@@ -465,6 +465,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/interpolate8x8_3dne.asm xvidcore/src/image/x86_asm/interpolate8x8_3dne.asm
--- xvidcore.orig/src/image/x86_asm/interpolate8x8_3dne.asm	2009-05-28 18:04:35.000000000 +0300
+++ xvidcore/src/image/x86_asm/interpolate8x8_3dne.asm	2009-05-29 20:04:58.000000000 +0300
@@ -565,6 +565,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/interpolate8x8_mmx.asm xvidcore/src/image/x86_asm/interpolate8x8_mmx.asm
--- xvidcore.orig/src/image/x86_asm/interpolate8x8_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/interpolate8x8_mmx.asm	2009-05-29 20:03:19.000000000 +0300
@@ -1330,6 +1330,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/interpolate8x8_xmm.asm xvidcore/src/image/x86_asm/interpolate8x8_xmm.asm
--- xvidcore.orig/src/image/x86_asm/interpolate8x8_xmm.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/interpolate8x8_xmm.asm	2009-05-29 20:04:25.000000000 +0300
@@ -821,6 +821,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/postprocessing_mmx.asm xvidcore/src/image/x86_asm/postprocessing_mmx.asm
--- xvidcore.orig/src/image/x86_asm/postprocessing_mmx.asm	2009-05-28 18:04:35.000000000 +0300
+++ xvidcore/src/image/x86_asm/postprocessing_mmx.asm	2009-05-29 20:02:08.000000000 +0300
@@ -115,6 +115,6 @@
 ;//////////////////////////////////////////////////////////////////////
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/postprocessing_sse2.asm xvidcore/src/image/x86_asm/postprocessing_sse2.asm
--- xvidcore.orig/src/image/x86_asm/postprocessing_sse2.asm	2009-05-28 18:04:35.000000000 +0300
+++ xvidcore/src/image/x86_asm/postprocessing_sse2.asm	2009-05-29 20:00:53.000000000 +0300
@@ -131,6 +131,6 @@
 ;//////////////////////////////////////////////////////////////////////
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/qpel_mmx.asm xvidcore/src/image/x86_asm/qpel_mmx.asm
--- xvidcore.orig/src/image/x86_asm/qpel_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/qpel_mmx.asm	2009-05-29 20:05:34.000000000 +0300
@@ -1150,6 +1150,6 @@
 %undef DST_PTR
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/image/x86_asm/reduced_mmx.asm xvidcore/src/image/x86_asm/reduced_mmx.asm
--- xvidcore.orig/src/image/x86_asm/reduced_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/image/x86_asm/reduced_mmx.asm	2009-05-29 20:06:46.000000000 +0300
@@ -987,6 +987,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/motion/x86_asm/sad_3dn.asm xvidcore/src/motion/x86_asm/sad_3dn.asm
--- xvidcore.orig/src/motion/x86_asm/sad_3dn.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/motion/x86_asm/sad_3dn.asm	2009-05-29 20:06:14.000000000 +0300
@@ -218,6 +218,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/motion/x86_asm/sad_3dne.asm xvidcore/src/motion/x86_asm/sad_3dne.asm
--- xvidcore.orig/src/motion/x86_asm/sad_3dne.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/motion/x86_asm/sad_3dne.asm	2009-05-29 20:12:08.000000000 +0300
@@ -492,6 +492,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/motion/x86_asm/sad_mmx.asm xvidcore/src/motion/x86_asm/sad_mmx.asm
--- xvidcore.orig/src/motion/x86_asm/sad_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/motion/x86_asm/sad_mmx.asm	2009-05-29 20:11:01.000000000 +0300
@@ -733,6 +733,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/motion/x86_asm/sad_sse2.asm xvidcore/src/motion/x86_asm/sad_sse2.asm
--- xvidcore.orig/src/motion/x86_asm/sad_sse2.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/motion/x86_asm/sad_sse2.asm	2009-05-29 20:11:44.000000000 +0300
@@ -178,6 +178,6 @@
 
  
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/motion/x86_asm/sad_xmm.asm xvidcore/src/motion/x86_asm/sad_xmm.asm
--- xvidcore.orig/src/motion/x86_asm/sad_xmm.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/motion/x86_asm/sad_xmm.asm	2009-05-29 20:11:22.000000000 +0300
@@ -438,6 +438,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/plugins/x86_asm/plugin_ssim-a.asm xvidcore/src/plugins/x86_asm/plugin_ssim-a.asm
--- xvidcore.orig/src/plugins/x86_asm/plugin_ssim-a.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/plugins/x86_asm/plugin_ssim-a.asm	2009-05-29 20:01:51.000000000 +0300
@@ -274,5 +274,5 @@
 ENDFUNC
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
diff -ur xvidcore.orig/src/quant/x86_asm/quantize_h263_3dne.asm xvidcore/src/quant/x86_asm/quantize_h263_3dne.asm
--- xvidcore.orig/src/quant/x86_asm/quantize_h263_3dne.asm	2009-05-28 18:04:35.000000000 +0300
+++ xvidcore/src/quant/x86_asm/quantize_h263_3dne.asm	2009-05-29 20:09:29.000000000 +0300
@@ -932,6 +932,6 @@
 ENDFUNC
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/quant/x86_asm/quantize_h263_mmx.asm xvidcore/src/quant/x86_asm/quantize_h263_mmx.asm
--- xvidcore.orig/src/quant/x86_asm/quantize_h263_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/quant/x86_asm/quantize_h263_mmx.asm	2009-05-29 20:09:55.000000000 +0300
@@ -1072,6 +1072,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/quant/x86_asm/quantize_mpeg_mmx.asm xvidcore/src/quant/x86_asm/quantize_mpeg_mmx.asm
--- xvidcore.orig/src/quant/x86_asm/quantize_mpeg_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/quant/x86_asm/quantize_mpeg_mmx.asm	2009-05-29 20:09:07.000000000 +0300
@@ -659,6 +659,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/quant/x86_asm/quantize_mpeg_xmm.asm xvidcore/src/quant/x86_asm/quantize_mpeg_xmm.asm
--- xvidcore.orig/src/quant/x86_asm/quantize_mpeg_xmm.asm	2009-05-28 18:04:35.000000000 +0300
+++ xvidcore/src/quant/x86_asm/quantize_mpeg_xmm.asm	2009-05-29 20:10:35.000000000 +0300
@@ -620,6 +620,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/utils/x86_asm/cpuid.asm xvidcore/src/utils/x86_asm/cpuid.asm
--- xvidcore.orig/src/utils/x86_asm/cpuid.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/utils/x86_asm/cpuid.asm	2009-05-29 20:12:32.000000000 +0300
@@ -241,6 +241,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/utils/x86_asm/interlacing_mmx.asm xvidcore/src/utils/x86_asm/interlacing_mmx.asm
--- xvidcore.orig/src/utils/x86_asm/interlacing_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/utils/x86_asm/interlacing_mmx.asm	2009-05-29 19:58:57.000000000 +0300
@@ -200,6 +200,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/utils/x86_asm/mem_transfer_3dne.asm xvidcore/src/utils/x86_asm/mem_transfer_3dne.asm
--- xvidcore.orig/src/utils/x86_asm/mem_transfer_3dne.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/utils/x86_asm/mem_transfer_3dne.asm	2009-05-29 19:59:24.000000000 +0300
@@ -467,6 +467,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
diff -ur xvidcore.orig/src/utils/x86_asm/mem_transfer_mmx.asm xvidcore/src/utils/x86_asm/mem_transfer_mmx.asm
--- xvidcore.orig/src/utils/x86_asm/mem_transfer_mmx.asm	2009-05-28 11:42:37.000000000 +0300
+++ xvidcore/src/utils/x86_asm/mem_transfer_mmx.asm	2009-05-29 20:04:06.000000000 +0300
@@ -590,6 +590,6 @@
 
 
 %ifidn __OUTPUT_FORMAT__,elf
-section ".note.GNU-stack" noalloc noexec nowrite progbits
+section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
