http://bugs.gentoo.org/293573

--- openmotif-2.3.2-orig/lib/Mrm/Makefile.am
+++ openmotif-2.3.2/lib/Mrm/Makefile.am
@@ -8,7 +8,7 @@
 
 libMrm_la_LIBADD = ../Xm/libXm.la ${X_LIBS} -lXt ${X_PRE_LIBS} -lX11 ${X_EXTRA_LIBS}
 
-LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
+AM_LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
 
 lib_LTLIBRARIES = libMrm.la
 
--- openmotif-2.3.2-orig/lib/Xm/Makefile.am
+++ openmotif-2.3.2/lib/Xm/Makefile.am
@@ -11,7 +11,7 @@
            ${X_CFLAGS}
 xmdir = $(includedir)/Xm
 
-LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
+AM_LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
 
 lib_LTLIBRARIES = libXm.la
 
