--- openmotif-2.3.1-orig/lib/Xm/I18List.c	2007-09-14 18:29:52.000000000 +0200
+++ openmotif-2.3.1/lib/Xm/I18List.c	2008-07-25 23:33:23.000000000 +0200
@@ -2086,12 +2086,7 @@
     values.graphics_exposures = False;
 
     mask = GCForeground | GCBackground | GCFont | GCGraphicsExposures;
-
-#ifdef FIX_1381
-	smask = mask | GCFillStyle;
-#else
     smask = mask | GCStipple | GCFillStyle;
-#endif
     
     XmI18List_gc(ilist) = XtGetGC(w, mask, &values);
 
@@ -2122,11 +2117,6 @@
     	XmI18List_entry_background_gc(ilist) = XtGetGC(w, mask, &values);
     }
 
-#ifdef FIX_1381
-	/*added for gray insensitive foreground (instead stipple)*/
-    values.foreground=_XmAssignInsensitiveColor(w);
-#endif
-
     XmI18List_stippled_gc(ilist) = XtGetGC(w, smask, &values);
 
     if (XmI18List_entry_background_use(ilist))
