--- /usr/portage/app-text/enchant/enchant-1.4.2.ebuild	2010-04-08 21:36:41.000000000 +0200
+++ ./app-text/enchant/enchant-1.4.2.ebuild	2010-05-16 00:26:31.000000000 +0200
@@ -30,21 +30,18 @@
 	confutils_require_any aspell hunspell zemberek
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	sed -i -e 's:noinst_PROGRAMS:check_PROGRAMS:' tests/Makefile.am \
 		|| die "unable to remove testdefault build"
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	econf $(use_enable aspell) \
 		$(use_enable hunspell myspell) \
 		$(use_enable zemberek) \
 		--disable-ispell \
 		--with-myspell-dir=/usr/share/myspell/
-	emake || die "emake failed"
 }
 
 src_install() {
